{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus genomes dir is /home/heineike/genomes/y1000plus/\n",
      "Importing y1000plus_tools.py\n",
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# %load std_libraries.py\n",
    "#Indicate operating environment and import core modules\n",
    "import os \n",
    "location_input = input(\"what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = other\")\n",
    "location_dict = {'a': \"C:\\\\Users\\\\BMH_work\\\\\", 'b': \"/home/heineike/\",\n",
    "                 'c': \"C:\\\\Users\\\\Ben\\\\Documents\\\\\", 'd':'you need to add your location to the location_dict'}\n",
    "home_dir = location_dict[location_input]\n",
    "print(\"home directory is \" + home_dir)\n",
    "base_dir = home_dir + os.path.normpath('github/y1000plus_tools') + os.sep\n",
    "print(\"y1000plus_tools dir is \" + base_dir )\n",
    "y1000plus_dir = home_dir + os.path.normpath(\"genomes/y1000plus\") + os.sep\n",
    "print(\"y1000plus genomes dir is \" + y1000plus_dir)\n",
    "\n",
    "import sys\n",
    "\n",
    "if sys.path[-1] != base_dir:\n",
    "    sys.path.append(base_dir)\n",
    "    print(\"Added \" + base_dir + \" to path: \" )\n",
    "    print(sys.path)\n",
    "\n",
    "print(\"Importing y1000plus_tools.py\")\n",
    "from core import y1000plus_tools\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Alphabet import generic_dna\n",
    "from Bio import SeqIO\n",
    "from ete3 import Tree\n",
    "#ete3 is not officially supported on windows, and so must be loaded via pip: \n",
    "# pip install -U https://github.com/etetoolkit/ete/archive/qt5.zip\n",
    "# ref: https://groups.google.com/forum/#!topic/etetoolkit/6NblSBPij4o\n",
    "#20181031: got this error message: twisted 18.7.0 requires PyHamcrest>=1.9.0, which is not installed.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "y1000_species_subset = y1000_species[y1000_species['Major clade'].isin(clades_of_interest)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#genes of interest: \n",
    "\n",
    "#NOtice how many of these genes are very close on the genome MPC3, SOL4, ENO1, GND2\n",
    "\n",
    "\n",
    "gois = ['YGR043C','YLR354C', #Start strong Msn2/4 regulation NQM1/TAL1\n",
    "       'YGR256W', 'YHR183W', #GND2/GND1 - Missing genes swapped between expression and stres. \n",
    "        'YMR105C', 'YKL127W', #PGM2/PGM1 - still has STRE but expression goes away\n",
    "        #'YMR285C','YML118W', #NGL2/NGL3 - Poly-A RNA exonuclease - different OGs \n",
    "        'YGR254W','YHR174W', #ENO1/ENO2\n",
    "        'YGR248W','YHR163W', #SOL4/SOL3 - strong PDS signal\n",
    "        'YGR243W','YHR162W', #MPC3/MPC2 - Start Weak/no Msn2/4 effect: \n",
    "        'YJL164C','YKL166C', #TPK1/TPK3   (but GIS1/RPH1 had an effect and there is a strong PDS) \n",
    "        'YDL021W','YOL056W' #GPM2/GPM3 - #Start average Msn2/4 effect: expression data missing ortholog for high, I'm missing ortholog from low \n",
    "        #'YPL230W','YMR182C' #USV1/RGM1 - Different OGs\n",
    "      ]\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['110_821',\n",
       " '110_1753',\n",
       " '110_4266',\n",
       " '110_326',\n",
       " '110_1562',\n",
       " '110_689',\n",
       " '110_757',\n",
       " '110_4425',\n",
       " '110_4019',\n",
       " '110_1058',\n",
       " '110_2707',\n",
       " '110_3574',\n",
       " '110_2527',\n",
       " '110_1854',\n",
       " '110_1259',\n",
       " '110_2716']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y1000_ids = dict(scer_lookup.loc[gois,'y1000_id'])\n",
    "y1000_id_list = [y1000_ids[goi] for goi in gois]\n",
    "y1000_id_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "more than one goi in same orthogroup :OG1193 found_genes: \n",
      "['110_757', '110_4425']\n",
      "more than one goi in same orthogroup :OG1302 found_genes: \n",
      "['110_689', '110_1562']\n",
      "more than one goi in same orthogroup :OG1488 found_genes: \n",
      "['110_326', '110_4266']\n",
      "more than one goi in same orthogroup :OG1667 found_genes: \n",
      "['110_1753', '110_821']\n",
      "more than one goi in same orthogroup :OG1859 found_genes: \n",
      "['110_2707', '110_3574']\n",
      "more than one goi in same orthogroup :OG3991 found_genes: \n",
      "['110_1058', '110_4019']\n",
      "more than one goi in same orthogroup :OG4796 found_genes: \n",
      "['110_1854', '110_2527']\n",
      "more than one goi in same orthogroup :OG6005 found_genes: \n",
      "['110_2716', '110_1259']\n"
     ]
    }
   ],
   "source": [
    "#Make promoter files for various genes of interest\n",
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "# target_y1000_id = scer_lookup.loc[goi,'y1000_id']\n",
    "\n",
    "y1000_ids = dict(scer_lookup.loc[gois,'y1000_id'])\n",
    "y1000_id_list = [y1000_ids[goi] for goi in gois]\n",
    "y1000_id_list\n",
    "\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OG1667',\n",
       " 'OG1667',\n",
       " 'OG1488',\n",
       " 'OG1488',\n",
       " 'OG1302',\n",
       " 'OG1302',\n",
       " 'OG1193',\n",
       " 'OG1193',\n",
       " 'OG3991',\n",
       " 'OG3991',\n",
       " 'OG1859',\n",
       " 'OG1859',\n",
       " 'OG4796',\n",
       " 'OG4796',\n",
       " 'OG6005',\n",
       " 'OG6005']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[goi_og_lookup[y1000_ids[goi]] for goi in gois]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus genomes dir is /home/heineike/genomes/y1000plus/\n",
      "TPK1_TPK3\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035703_cyberlindnera_americana_160613 Gene = Seq_4055, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034623_pichia_norvegensis_160519 Gene = Seq_641, L_prom = 1000\n",
      "scaffold ends at the end of the gene, genome_name = yHMPu5000034623_pichia_norvegensis_160519 Gene = Seq_641\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034627_pichia_heedii_160519 Gene = Seq_3722, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035318_hyphopichia_heimii_160928 Gene = Seq_422, L_prom = 1000\n",
      "TPK1_TPK3 complete\n",
      "NQM1_TAL1\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034963_hanseniaspora_clermontiae_160519 Gene = Seq_729, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034952_citeromyces_siamensis_160519 Gene = Seq_5207, L_prom = 1000\n",
      "scaffold ends at the end of the gene, genome_name = yHMPu5000034952_citeromyces_siamensis_160519 Gene = Seq_5207\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519 Gene = Seq_3034, L_prom = 1000\n",
      "NQM1_TAL1 complete\n",
      "PGM2_PGM1\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035689_candida_mycetangii_160613 Gene = Seq_4844, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034877_tetrapisispora_namnaonensis_160519 Gene = Seq_5282, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034875_tetrapisispora_fleetii_160519 Gene = Seq_561, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034709_kluyveromyces_aestuarii_160519 Gene = Seq_2561, L_prom = 1000\n",
      "PGM2_PGM1 complete\n",
      "MPC3_MPC2\n",
      "promoter region extends past the scaffold, genome_name = metschnikowia_matae_maris Gene = Seq_6053, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000026142_citeromyces_matritensis_160519 Gene = Seq_7350, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034952_citeromyces_siamensis_160519 Gene = Seq_2637, L_prom = 1000\n",
      "MPC3_MPC2 complete\n",
      "SOL4_SOL3\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035673_candida_orba_160613 Gene = Seq_827, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034952_citeromyces_siamensis_160519 Gene = Seq_10680, L_prom = 1000\n",
      "SOL4_SOL3 complete\n",
      "ENO1_ENO2\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035018_candida_canberraensis_160519 Gene = Seq_412, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034901_ogataea_methylivora_160519 Gene = Seq_5239, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = vanderwaltozyma_polyspora Gene = Seq_4158, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = spathaspora_girioi Gene = Seq_4428, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = spathaspora_girioi Gene = Seq_2691, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034607_saccharomycopsis_capsularis_160519 Gene = Seq_1806, L_prom = 1000\n",
      "ENO1_ENO2 complete\n",
      "GPM2_GPM3\n",
      "GPM2_GPM3 complete\n",
      "GND2_GND1\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034967_candida_freyschussii_160519 Gene = Seq_4966, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034918_nakazawaea_holstii_160519 Gene = Seq_4244, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHAB159_kazachstania_solicola_160519 Gene = Seq_680, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = spathaspora_hagerdaliae Gene = Seq_6862, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHAB155_kazachstania_spencerorum_160519 Gene = Seq_5608, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000035684_kloeckera_hatyaiensis_160519 Gene = Seq_4508, L_prom = 1000\n",
      "scaffold ends at the end of the gene, genome_name = yHMPu5000035684_kloeckera_hatyaiensis_160519 Gene = Seq_4508\n",
      "GND2_GND1 complete\n"
     ]
    }
   ],
   "source": [
    "#Consolidate pairs with orthogroups\n",
    "goi_pair_og_lookup =  {'NQM1_TAL1': 'OG1667', \n",
    "                     'GND2_GND1': 'OG1488', \n",
    "                     'PGM2_PGM1': 'OG1302', \n",
    "                     'ENO1_ENO2': 'OG1193', \n",
    "                     'SOL4_SOL3': 'OG3991', \n",
    "                     'MPC3_MPC2': 'OG1859', \n",
    "                     'TPK1_TPK3': 'OG4796',\n",
    "                     'GPM2_GPM3': 'OG6005'\n",
    "                    }\n",
    "\n",
    "\n",
    "L_prom = 1000\n",
    "\n",
    "for goi_pair, og in goi_pair_og_lookup.items(): \n",
    "    print(goi_pair)\n",
    "    og_genes =  og_genes_lookup[og]\n",
    "    \n",
    "    #For a given orthogroup extract promoters\n",
    "\n",
    "    y1000plus_tools.extract_promoters(L_prom, og, og_genes, y1000_species_subset, fname_string=goi_pair)\n",
    "    print(goi_pair + ' complete')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'100_3933',\n",
       " '101_2082',\n",
       " '103_1124',\n",
       " '103_3315',\n",
       " '106_4826',\n",
       " '107_4755',\n",
       " '107_918',\n",
       " '108_10',\n",
       " '109_4426',\n",
       " '110_166',\n",
       " '110_1854',\n",
       " '110_2527',\n",
       " '112_2232',\n",
       " '113_139',\n",
       " '113_2286',\n",
       " '114_4172',\n",
       " '115_1142',\n",
       " '116_3145',\n",
       " '119_4686',\n",
       " '120_5607',\n",
       " '125_4524',\n",
       " '12_3594',\n",
       " '12_5377',\n",
       " '132_3818',\n",
       " '133_1594',\n",
       " '135_3540',\n",
       " '136_1926',\n",
       " '139_3466',\n",
       " '140_4099',\n",
       " '145_115',\n",
       " '146_1440',\n",
       " '147_4836',\n",
       " '148_2977',\n",
       " '148_4275',\n",
       " '14_1754',\n",
       " '14_4864',\n",
       " '150_3479',\n",
       " '151_2554',\n",
       " '152_1171',\n",
       " '152_3110',\n",
       " '153_1307',\n",
       " '154_2313',\n",
       " '155_3876',\n",
       " '156_3910',\n",
       " '157_2140',\n",
       " '158_2596',\n",
       " '158_3650',\n",
       " '159_581',\n",
       " '15_837',\n",
       " '162_3999',\n",
       " '164_620',\n",
       " '167_3699',\n",
       " '16_2105',\n",
       " '172_4189',\n",
       " '173_790',\n",
       " '174_1524',\n",
       " '175_2697',\n",
       " '176_4449',\n",
       " '176_727',\n",
       " '177_3773',\n",
       " '178_1816',\n",
       " '178_3743',\n",
       " '179_1738',\n",
       " '17_3615',\n",
       " '17_545',\n",
       " '180_936',\n",
       " '181_640',\n",
       " '182_2519',\n",
       " '183_1039',\n",
       " '184_3721',\n",
       " '185_426',\n",
       " '187_3525',\n",
       " '189_2952',\n",
       " '18_3994',\n",
       " '191_1356',\n",
       " '191_3147',\n",
       " '193_3692',\n",
       " '198_4197',\n",
       " '19_3202',\n",
       " '218_1119',\n",
       " '220_887',\n",
       " '222_5300',\n",
       " '223_735',\n",
       " '224_746',\n",
       " '225_405',\n",
       " '227_4894',\n",
       " '227_599',\n",
       " '229_3567',\n",
       " '22_808',\n",
       " '230_2390',\n",
       " '231_5496',\n",
       " '234_2951',\n",
       " '236_415',\n",
       " '237_5012',\n",
       " '242_629',\n",
       " '246_11149',\n",
       " '247_3923',\n",
       " '247_614',\n",
       " '248_188',\n",
       " '24_694',\n",
       " '250_274',\n",
       " '255_933',\n",
       " '256_1894',\n",
       " '258_5542',\n",
       " '25_2508',\n",
       " '260_3901',\n",
       " '262_4372',\n",
       " '263_3389',\n",
       " '263_4102',\n",
       " '264_4176',\n",
       " '265_2681',\n",
       " '276_2124',\n",
       " '276_2701',\n",
       " '279_5052',\n",
       " '286_4138',\n",
       " '287_2357',\n",
       " '288_421',\n",
       " '289_938',\n",
       " '290_4449',\n",
       " '291_1247',\n",
       " '291_4074',\n",
       " '2_2906',\n",
       " '303_2139',\n",
       " '303_452',\n",
       " '304_3769',\n",
       " '306_849',\n",
       " '307_1317',\n",
       " '308_893',\n",
       " '30_5719',\n",
       " '310_2610',\n",
       " '311_1532',\n",
       " '311_3589',\n",
       " '312_5931',\n",
       " '312_760',\n",
       " '313_3963',\n",
       " '315_1740',\n",
       " '316_3559',\n",
       " '316_663',\n",
       " '319_1855',\n",
       " '319_4074',\n",
       " '321_1630',\n",
       " '323_894',\n",
       " '324_4931',\n",
       " '325_3150',\n",
       " '326_4054',\n",
       " '327_4931',\n",
       " '331_2724',\n",
       " '331_3941',\n",
       " '337_1760',\n",
       " '339_3733',\n",
       " '341_3964',\n",
       " '34_519',\n",
       " '35_5333',\n",
       " '35_5435',\n",
       " '39_4406',\n",
       " '44_2740',\n",
       " '45_1166',\n",
       " '45_2640',\n",
       " '46_1267',\n",
       " '52_4427',\n",
       " '55_4868',\n",
       " '56_3401',\n",
       " '58_4583',\n",
       " '59_2316',\n",
       " '5_6929',\n",
       " '60_3358',\n",
       " '62_2265',\n",
       " '63_23',\n",
       " '63_308',\n",
       " '64_1929',\n",
       " '65_3404',\n",
       " '69_5567',\n",
       " '6_6159',\n",
       " '71_1948',\n",
       " '72_2470',\n",
       " '72_4445',\n",
       " '73_3519',\n",
       " '74_973',\n",
       " '75_2004',\n",
       " '75_4050',\n",
       " '76_54',\n",
       " '77_2923',\n",
       " '78_5964',\n",
       " '79_1825',\n",
       " '80_2071',\n",
       " '81_30',\n",
       " '82_1157',\n",
       " '83_5721',\n",
       " '84_3402',\n",
       " '85_4920',\n",
       " '86_1741',\n",
       " '87_3889',\n",
       " '88_2938',\n",
       " '90_548',\n",
       " '96_1096',\n",
       " '98_1441',\n",
       " '98_3701',\n",
       " '99_2010',\n",
       " '9_2568',\n",
       " '9_3458'}"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "og_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import gffutils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "promoter region extends past the scaffold, genome_name = yHMPu5000035703_cyberlindnera_americana_160613 Gene = Seq_4055, L_prom = 1000\n",
      "promoter region extends past the scaffold, genome_name = yHMPu5000034623_pichia_norvegensis_160519 Gene = Seq_641, L_prom = 1000\n",
      "scaffold ends at the end of the gene, genome_name = yHMPu5000034623_pichia_norvegensis_160519 Gene = Seq_641\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "Scaffold not assigned for Seq_2939 in metschnikowia_santaceciliae, example scaffold: sce+_s99",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-91-7a853e949146>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     85\u001b[0m                         \u001b[0mscaffold_assigned\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 87\u001b[0;31m             \u001b[0;32massert\u001b[0m \u001b[0mscaffold_assigned\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Scaffold not assigned for '\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mgene_id\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m' in '\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mgenome_name\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m', example scaffold: '\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mseq_record\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     88\u001b[0m             \u001b[0;31m#if strand is negative, check to see if promoter coordinates are at the end of the scaffold\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     89\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAssertionError\u001b[0m: Scaffold not assigned for Seq_2939 in metschnikowia_santaceciliae, example scaffold: sce+_s99"
     ]
    }
   ],
   "source": [
    "missing_specs = {'saccharomyces_cerevisiae', 'candida_albicans'}  #outgroups not included (e.g. \n",
    "scaffold_name_change =  {'metschnikowia_dekortum':'+', \n",
    "                         'metschnikowia_borealis': '+', \n",
    "                         'metschnikowia_continentalis': '+', \n",
    "                         'metschnikowia_kamakouana': '+', \n",
    "                         'metschnikowia_hibisci': '+'}\n",
    "scaffold_name_change_specs = scaffold_name_change.keys()\n",
    "#Table to look up species by number: ''\n",
    "genome_name_lookup = dict(zip(y1000_species_subset['spec_og_id'],y1000_species_subset['original_genome_id']))\n",
    "\n",
    "\n",
    "#for a given set of genes, \n",
    "# group by species\n",
    "og_genes_specs = {spec_gene.split('_')[0]: [] for spec_gene in og_genes}\n",
    "for y1000_id in og_genes: \n",
    "    spec_og_id = y1000_id.split('_')[0]\n",
    "    og_genes_specs[spec_og_id].append(y1000_id)\n",
    "\n",
    "#for each species in the group, gather promoters.  Extract promoter, print promoter as a line in a fasta file\n",
    "#>spec y1000_id gene_id gene_full chrm strand start end L\n",
    "for spec_og_id, genes in og_genes_specs.items(): \n",
    "    genome_name = genome_name_lookup[int(spec_og_id)]\n",
    "    #if S.Cer skip finding promoter, \n",
    "    if not(genome_name in missing_specs):    \n",
    "        #load gene_id map based on the species\n",
    "        gene_lookup_spec_fname = y1000plus_dir + \"id_lookups/\" + genome_name + '.csv'\n",
    "        gene_lookup_spec = pd.read_csv(gene_lookup_spec_fname, index_col='y1000_id')\n",
    "\n",
    "        # Load GTF for given sequence: \n",
    "        gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "        db_fname = gtf_dir + 'gffutils_dbs/' + genome_name + '.db'\n",
    "\n",
    "        gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "        #Extract sequences from genome\n",
    "        genome_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_assemblies/\"\n",
    "\n",
    "        genome_fname = genome_dir + genome_name + '.fas'\n",
    "        \n",
    "        #For each gene, extract all related features, and then output chromosome and coordinates for the promoter\n",
    "        for y1000_id in genes: \n",
    "            gene_full = gene_lookup_spec.loc[y1000_id,'gene_full']\n",
    "            gene_id = gene_lookup_spec.loc[y1000_id,'gene_id']\n",
    "\n",
    "\n",
    "            cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "            all_features = cursor.fetchall()\n",
    "            if len(all_features) == 0:\n",
    "                print('No features found ' + gene_full + ' ' +genome_name)\n",
    "\n",
    "            strand = all_features[0]['strand']\n",
    "            chrom = all_features[0]['seqid']\n",
    "\n",
    "            starts = []\n",
    "            ends = []\n",
    "            for feature in all_features: \n",
    "                starts.append(feature['start'])\n",
    "                ends.append(feature['end'])\n",
    "\n",
    "            #Adjust coordinates to get L_prom \"promoter\" sequences\n",
    "            if strand == '-': \n",
    "                prom_end = max(ends) + 1\n",
    "                prom_start = prom_end + L_prom   #should do min of this and the total length of the scaffold, \n",
    "            elif strand == '+': \n",
    "                prom_end = min(starts) - 1\n",
    "                prom_start = max(0,prom_end - L_prom)\n",
    "\n",
    "            #extract sequences from genome\n",
    "            #Need to reload the iterator each time\n",
    "            seq_records = SeqIO.parse(genome_fname, \"fasta\")\n",
    "            scaffold_assigned = False\n",
    "            \n",
    "            if genome_name in scaffold_name_change_specs:\n",
    "                extra_char = scaffold_name_change[genome_name]\n",
    "                for seq_record in seq_records:\n",
    "                    test_scaffold = ''.join(seq_record.id.split(extra_char))\n",
    "                    if test_scaffold == chrom:\n",
    "                        scaffold = seq_record\n",
    "                        scaffold_assigned=True\n",
    "            else:     \n",
    "                for seq_record in seq_records:\n",
    "                    test_scaffold = seq_record.id\n",
    "                    if test_scaffold == chrom:\n",
    "                        scaffold = seq_record\n",
    "                        scaffold_assigned=True\n",
    "            \n",
    "            assert scaffold_assigned, 'Scaffold not assigned for ' + gene_id + ' in ' + genome_name + ', example scaffold: ' + seq_record.id\n",
    "            #if strand is negative, check to see if promoter coordinates are at the end of the scaffold\n",
    "\n",
    "            L_scaffold = len(scaffold)\n",
    "\n",
    "            if strand == '-': \n",
    "                if prom_start > L_scaffold: \n",
    "                    print('promoter region extends past the scaffold, genome_name = ' + genome_name + ' Gene = ' + gene_id + ', L_prom = ' + str(L_prom))\n",
    "                    prom_start = L_scaffold\n",
    "                if prom_end > L_scaffold: \n",
    "                    print('scaffold ends at the end of the gene, genome_name = ' + genome_name + ' Gene = ' + gene_id)\n",
    "                    prom_end = L_scaffold\n",
    "\n",
    "                promoter = scaffold.seq[prom_end:prom_start].reverse_complement()\n",
    "            elif strand == '+': \n",
    "                promoter = scaffold.seq[prom_start:prom_end]\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'dek_s17'"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scaffold_assigned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'dek_s99'"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dek+_C14301\n",
      "dek+_C14303\n",
      "dek+_C14317\n",
      "dek+_C14383\n",
      "dek+_C14389\n",
      "dek+_C14395\n",
      "dek+_C14427\n",
      "dek+_C14453\n",
      "dek+_C14543\n",
      "dek+_C14617\n",
      "dek+_C14701\n",
      "dek+_C14871\n",
      "dek+_C14941\n",
      "dek+_C14959\n",
      "dek+_C15015\n",
      "dek+_C15203\n",
      "dek+_C15281\n",
      "dek+_C15301\n",
      "dek+_C16103\n",
      "dek+_s1\n",
      "dek+_s10\n",
      "dek+_s100\n",
      "dek+_s101\n",
      "dek+_s102\n",
      "dek+_s103\n",
      "dek+_s104\n",
      "dek+_s105\n",
      "dek+_s107\n",
      "dek+_s108\n",
      "dek+_s109\n",
      "dek+_s11\n",
      "dek+_s111\n",
      "dek+_s113\n",
      "dek+_s114\n",
      "dek+_s115\n",
      "dek+_s116\n",
      "dek+_s117\n",
      "dek+_s118\n",
      "dek+_s119\n",
      "dek+_s12\n",
      "dek+_s120\n",
      "dek+_s121\n",
      "dek+_s122\n",
      "dek+_s123\n",
      "dek+_s124\n",
      "dek+_s125\n",
      "dek+_s128\n",
      "dek+_s129\n",
      "dek+_s13\n",
      "dek+_s131\n",
      "dek+_s133\n",
      "dek+_s134\n",
      "dek+_s135\n",
      "dek+_s136\n",
      "dek+_s137\n",
      "dek+_s138\n",
      "dek+_s139\n",
      "dek+_s14\n",
      "dek+_s140\n",
      "dek+_s141\n",
      "dek+_s142\n",
      "dek+_s143\n",
      "dek+_s144\n",
      "dek+_s145\n",
      "dek+_s146\n",
      "dek+_s147\n",
      "dek+_s148\n",
      "dek+_s149\n",
      "dek+_s15\n",
      "dek+_s150\n",
      "dek+_s151\n",
      "dek+_s152\n",
      "dek+_s153\n",
      "dek+_s154\n",
      "dek+_s155\n",
      "dek+_s17\n",
      "dek+_s18\n",
      "dek+_s19\n",
      "dek+_s2\n",
      "dek+_s20\n",
      "dek+_s21\n",
      "dek+_s22\n",
      "dek+_s23\n",
      "dek+_s24\n",
      "dek+_s25\n",
      "dek+_s26\n",
      "dek+_s27\n",
      "dek+_s28\n",
      "dek+_s29\n",
      "dek+_s3\n",
      "dek+_s31\n",
      "dek+_s32\n",
      "dek+_s33\n",
      "dek+_s34\n",
      "dek+_s35\n",
      "dek+_s36\n",
      "dek+_s37\n",
      "dek+_s38\n",
      "dek+_s39\n",
      "dek+_s4\n",
      "dek+_s40\n",
      "dek+_s41\n",
      "dek+_s42\n",
      "dek+_s43\n",
      "dek+_s44\n",
      "dek+_s45\n",
      "dek+_s46\n",
      "dek+_s47\n",
      "dek+_s48\n",
      "dek+_s49\n",
      "dek+_s5\n",
      "dek+_s50\n",
      "dek+_s51\n",
      "dek+_s52\n",
      "dek+_s53\n",
      "dek+_s54\n",
      "dek+_s55\n",
      "dek+_s56\n",
      "dek+_s57\n",
      "dek+_s58\n",
      "dek+_s59\n",
      "dek+_s6\n",
      "dek+_s60\n",
      "dek+_s61\n",
      "dek+_s63\n",
      "dek+_s64\n",
      "dek+_s65\n",
      "dek+_s66\n",
      "dek+_s67\n",
      "dek+_s68\n",
      "dek+_s69\n",
      "dek+_s71\n",
      "dek+_s72\n",
      "dek+_s74\n",
      "dek+_s75\n",
      "dek+_s76\n",
      "dek+_s77\n",
      "dek+_s78\n",
      "dek+_s79\n",
      "dek+_s8\n",
      "dek+_s80\n",
      "dek+_s81\n",
      "dek+_s83\n",
      "dek+_s84\n",
      "dek+_s85\n",
      "dek+_s86\n",
      "dek+_s87\n",
      "dek+_s88\n",
      "dek+_s89\n",
      "dek+_s9\n",
      "dek+_s91\n",
      "dek+_s92\n",
      "dek+_s93\n",
      "dek+_s94\n",
      "dek+_s95\n",
      "dek+_s96\n",
      "dek+_s97\n",
      "dek+_s98\n",
      "dek+_s99\n"
     ]
    }
   ],
   "source": [
    "seq_records = SeqIO.parse(genome_fname, \"fasta\")\n",
    "for seq_record in seq_records:\n",
    "    print(seq_record.id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: dek+_s99\n",
      "Name: dek+_s99\n",
      "Description: dek+_s99\n",
      "Number of features: 0\n",
      "Seq('CAATTTGGCTGACTACGAACACCGCCTCTCCAAATTGAAAGACGACTTCGACAT...AAG', SingleLetterAlphabet())\n"
     ]
    }
   ],
   "source": [
    "print(seq_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make phylogenetic trees for various genes of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "seqanalysis",
   "language": "python",
   "name": "seqanalysis"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
