{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus genomes dir is /home/heineike/genomes/y1000plus/\n",
      "Added /home/heineike/github/y1000plus_tools/ to path: \n",
      "['', '/home/lab/envs/seqanalysis/lib/python35.zip', '/home/lab/envs/seqanalysis/lib/python3.5', '/home/lab/envs/seqanalysis/lib/python3.5/plat-linux', '/home/lab/envs/seqanalysis/lib/python3.5/lib-dynload', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/argh-0.26.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/glob2-0.4.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/mmtf_python-1.0.2-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/IPython/extensions', '/home/heineike/.ipython', '/home/heineike/github/y1000plus_tools/']\n",
      "Importing y1000plus_tools.py\n",
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus genomes dir is /home/heineike/genomes/y1000plus/\n"
     ]
    }
   ],
   "source": [
    "# %load std_libraries.py\n",
    "#Indicate operating environment and import core modules\n",
    "import os \n",
    "location_input = input(\"what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = other\")\n",
    "location_dict = {'a': \"C:\\\\Users\\\\BMH_work\\\\\", 'b': \"/home/heineike/\",\n",
    "                 'c': \"C:\\\\Users\\\\Ben\\\\Documents\\\\\", 'd':'you need to add your location to the location_dict'}\n",
    "home_dir = location_dict[location_input]\n",
    "print(\"home directory is \" + home_dir)\n",
    "base_dir = home_dir + os.path.normpath('github/y1000plus_tools') + os.sep\n",
    "print(\"y1000plus_tools dir is \" + base_dir )\n",
    "y1000plus_dir = home_dir + os.path.normpath(\"genomes/y1000plus\") + os.sep\n",
    "print(\"y1000plus genomes dir is \" + y1000plus_dir)\n",
    "\n",
    "import sys\n",
    "\n",
    "if sys.path[-1] != base_dir:\n",
    "    sys.path.append(base_dir)\n",
    "    print(\"Added \" + base_dir + \" to path: \" )\n",
    "    print(sys.path)\n",
    "\n",
    "print(\"Importing y1000plus_tools.py\")\n",
    "from core import y1000plus_tools\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Alphabet import generic_dna\n",
    "from Bio import SeqIO\n",
    "from ete3 import Tree\n",
    "#ete3 is not officially supported on windows, and so must be loaded via pip: \n",
    "# pip install -U https://github.com/etetoolkit/ete/archive/qt5.zip\n",
    "# ref: https://groups.google.com/forum/#!topic/etetoolkit/6NblSBPij4o\n",
    "#20181031: got this error message: twisted 18.7.0 requires PyHamcrest>=1.9.0, which is not installed.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Only need to build it once afterwards can load from file\n",
    "#y1000_species = y1000plus_tools.build_y1000_species_table()\n",
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "##If you only need a subset of species use something like this\n",
    "#y1000_species_subset = y1000_species[y1000_species['Major clade']!='Saccharomycetaceae']\n",
    "\n",
    "##generated lookup tables for the rest of the species - this only needs to be done once. \n",
    "#y1000plus_tools.id_lookup_generate(y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus genomes dir is /home/heineike/genomes/y1000plus/\n"
     ]
    }
   ],
   "source": [
    "# Get target_y1000_id and orthogroup\n",
    "\n",
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "target_y1000_id = scer_lookup.loc['YEL009C','y1000_id']\n",
    "\n",
    "y1000_id_list = [target_y1000_id]\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'OG2986': {'0_1759',\n",
       "  '100_1482',\n",
       "  '101_1451',\n",
       "  '103_1815',\n",
       "  '104_3544',\n",
       "  '105_1916',\n",
       "  '106_2698',\n",
       "  '107_1006',\n",
       "  '108_2582',\n",
       "  '109_917',\n",
       "  '110_4180',\n",
       "  '111_4823',\n",
       "  '112_1988',\n",
       "  '113_3047',\n",
       "  '114_1814',\n",
       "  '115_75',\n",
       "  '116_5279',\n",
       "  '118_697',\n",
       "  '119_2707',\n",
       "  '11_3384',\n",
       "  '120_2694',\n",
       "  '123_2454',\n",
       "  '123_357',\n",
       "  '124_3599',\n",
       "  '125_3936',\n",
       "  '126_1969',\n",
       "  '126_3560',\n",
       "  '127_3120',\n",
       "  '128_955',\n",
       "  '12_5552',\n",
       "  '130_983',\n",
       "  '131_1015',\n",
       "  '132_489',\n",
       "  '133_4621',\n",
       "  '135_2037',\n",
       "  '136_2744',\n",
       "  '137_3042',\n",
       "  '138_331',\n",
       "  '139_824',\n",
       "  '13_2319',\n",
       "  '140_3316',\n",
       "  '142_1060',\n",
       "  '143_187',\n",
       "  '144_4347',\n",
       "  '145_1300',\n",
       "  '146_3752',\n",
       "  '147_5026',\n",
       "  '148_4493',\n",
       "  '149_2274',\n",
       "  '14_1846',\n",
       "  '150_2883',\n",
       "  '151_1494',\n",
       "  '152_5173',\n",
       "  '153_1100',\n",
       "  '154_4519',\n",
       "  '155_4690',\n",
       "  '156_4691',\n",
       "  '157_758',\n",
       "  '158_676',\n",
       "  '159_2475',\n",
       "  '15_4715',\n",
       "  '160_3309',\n",
       "  '161_5846',\n",
       "  '162_4672',\n",
       "  '163_3152',\n",
       "  '165_3483',\n",
       "  '166_4013',\n",
       "  '167_1002',\n",
       "  '168_3217',\n",
       "  '169_4977',\n",
       "  '16_2095',\n",
       "  '170_1878',\n",
       "  '171_432',\n",
       "  '172_1256',\n",
       "  '173_2292',\n",
       "  '174_3757',\n",
       "  '175_3984',\n",
       "  '176_1820',\n",
       "  '177_3596',\n",
       "  '179_3855',\n",
       "  '17_2942',\n",
       "  '180_2761',\n",
       "  '181_2393',\n",
       "  '181_3868',\n",
       "  '182_3511',\n",
       "  '183_3713',\n",
       "  '184_996',\n",
       "  '185_4130',\n",
       "  '186_12134',\n",
       "  '187_1580',\n",
       "  '188_5989',\n",
       "  '189_3885',\n",
       "  '18_5493',\n",
       "  '190_4037',\n",
       "  '191_2414',\n",
       "  '192_4583',\n",
       "  '193_2051',\n",
       "  '194_3815',\n",
       "  '195_9557',\n",
       "  '196_1596',\n",
       "  '197_2062',\n",
       "  '198_5645',\n",
       "  '199_72',\n",
       "  '19_3013',\n",
       "  '1_4449',\n",
       "  '200_2969',\n",
       "  '201_4715',\n",
       "  '202_1420',\n",
       "  '203_5888',\n",
       "  '204_2320',\n",
       "  '205_3527',\n",
       "  '206_2786',\n",
       "  '207_4489',\n",
       "  '208_1372',\n",
       "  '209_659',\n",
       "  '20_3584',\n",
       "  '210_4154',\n",
       "  '211_2759',\n",
       "  '212_3241',\n",
       "  '213_440',\n",
       "  '214_2020',\n",
       "  '215_6421',\n",
       "  '216_7794',\n",
       "  '217_847',\n",
       "  '218_1320',\n",
       "  '219_2313',\n",
       "  '21_3190',\n",
       "  '220_2889',\n",
       "  '221_4319',\n",
       "  '222_2994',\n",
       "  '223_2075',\n",
       "  '223_4968',\n",
       "  '224_5888',\n",
       "  '225_1411',\n",
       "  '226_3656',\n",
       "  '227_4843',\n",
       "  '228_933',\n",
       "  '229_3220',\n",
       "  '22_5392',\n",
       "  '230_2924',\n",
       "  '231_3717',\n",
       "  '232_1228',\n",
       "  '233_4971',\n",
       "  '234_827',\n",
       "  '235_483',\n",
       "  '236_1132',\n",
       "  '237_352',\n",
       "  '238_464',\n",
       "  '239_1251',\n",
       "  '23_1245',\n",
       "  '240_5477',\n",
       "  '241_996',\n",
       "  '242_307',\n",
       "  '243_3535',\n",
       "  '244_5970',\n",
       "  '245_930',\n",
       "  '246_8371',\n",
       "  '247_1137',\n",
       "  '249_3659',\n",
       "  '24_3635',\n",
       "  '250_4680',\n",
       "  '251_3622',\n",
       "  '252_1095',\n",
       "  '253_5557',\n",
       "  '254_2803',\n",
       "  '255_1388',\n",
       "  '256_3696',\n",
       "  '257_3857',\n",
       "  '258_5027',\n",
       "  '259_2804',\n",
       "  '25_3132',\n",
       "  '260_1669',\n",
       "  '261_597',\n",
       "  '262_4316',\n",
       "  '263_3169',\n",
       "  '264_3062',\n",
       "  '265_1489',\n",
       "  '266_2110',\n",
       "  '267_5674',\n",
       "  '268_4752',\n",
       "  '269_4976',\n",
       "  '26_2042',\n",
       "  '270_10',\n",
       "  '271_4754',\n",
       "  '272_3500',\n",
       "  '273_4716',\n",
       "  '274_1803',\n",
       "  '275_449',\n",
       "  '276_175',\n",
       "  '277_1332',\n",
       "  '278_6209',\n",
       "  '279_5',\n",
       "  '27_1997',\n",
       "  '280_244',\n",
       "  '281_2193',\n",
       "  '282_1139',\n",
       "  '283_3845',\n",
       "  '284_4410',\n",
       "  '285_4334',\n",
       "  '286_455',\n",
       "  '287_1822',\n",
       "  '288_1515',\n",
       "  '289_5624',\n",
       "  '28_4482',\n",
       "  '290_909',\n",
       "  '291_5322',\n",
       "  '292_3002',\n",
       "  '293_3858',\n",
       "  '294_4427',\n",
       "  '295_3017',\n",
       "  '296_2299',\n",
       "  '297_4177',\n",
       "  '298_3901',\n",
       "  '299_268',\n",
       "  '29_657',\n",
       "  '2_5414',\n",
       "  '300_206',\n",
       "  '302_1588',\n",
       "  '303_3188',\n",
       "  '304_1612',\n",
       "  '305_4399',\n",
       "  '305_4778',\n",
       "  '306_149',\n",
       "  '307_660',\n",
       "  '308_124',\n",
       "  '309_3738',\n",
       "  '30_1693',\n",
       "  '310_4617',\n",
       "  '311_4358',\n",
       "  '312_4',\n",
       "  '312_4339',\n",
       "  '313_4128',\n",
       "  '314_2150',\n",
       "  '315_1791',\n",
       "  '317_6086',\n",
       "  '318_1818',\n",
       "  '319_5489',\n",
       "  '31_1147',\n",
       "  '320_3170',\n",
       "  '321_4325',\n",
       "  '322_4148',\n",
       "  '325_2468',\n",
       "  '326_2457',\n",
       "  '327_2513',\n",
       "  '328_5426',\n",
       "  '329_1565',\n",
       "  '32_4660',\n",
       "  '330_3222',\n",
       "  '331_2221',\n",
       "  '332_2538',\n",
       "  '332_4483',\n",
       "  '333_1891',\n",
       "  '334_2910',\n",
       "  '335_5316',\n",
       "  '336_2285',\n",
       "  '337_3779',\n",
       "  '338_3955',\n",
       "  '339_145',\n",
       "  '340_1262',\n",
       "  '341_604',\n",
       "  '342_3629',\n",
       "  '34_3290',\n",
       "  '35_378',\n",
       "  '36_3461',\n",
       "  '37_5293',\n",
       "  '38_2732',\n",
       "  '39_2323',\n",
       "  '3_10453',\n",
       "  '40_224',\n",
       "  '41_2850',\n",
       "  '43_1775',\n",
       "  '43_5567',\n",
       "  '46_3365',\n",
       "  '47_4053',\n",
       "  '48_1647',\n",
       "  '49_870',\n",
       "  '4_4496',\n",
       "  '50_299',\n",
       "  '51_2233',\n",
       "  '52_3879',\n",
       "  '53_4787',\n",
       "  '54_2608',\n",
       "  '55_675',\n",
       "  '57_3174',\n",
       "  '58_3758',\n",
       "  '61_2638',\n",
       "  '62_4349',\n",
       "  '63_3775',\n",
       "  '64_1832',\n",
       "  '65_1249',\n",
       "  '65_5054',\n",
       "  '66_174',\n",
       "  '67_4078',\n",
       "  '68_2758',\n",
       "  '69_26',\n",
       "  '70_3410',\n",
       "  '71_3409',\n",
       "  '72_3801',\n",
       "  '73_5885',\n",
       "  '74_692',\n",
       "  '75_2363',\n",
       "  '76_3232',\n",
       "  '77_1306',\n",
       "  '78_393',\n",
       "  '79_607',\n",
       "  '80_4545',\n",
       "  '81_5447',\n",
       "  '82_1387',\n",
       "  '83_4148',\n",
       "  '84_731',\n",
       "  '85_7256',\n",
       "  '86_2902',\n",
       "  '87_4750',\n",
       "  '88_464',\n",
       "  '89_1904',\n",
       "  '8_3336',\n",
       "  '90_1775',\n",
       "  '91_4857',\n",
       "  '92_4328',\n",
       "  '93_1420',\n",
       "  '94_4081',\n",
       "  '95_385',\n",
       "  '96_530',\n",
       "  '97_223',\n",
       "  '98_2443',\n",
       "  '99_3741',\n",
       "  '9_1580'}}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#goi_og_lookup \n",
    "og_genes_lookup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCN4 is given by the y1000_id: 110_4180\n",
      "It is in orthogroup: OG2986\n",
      "which has the genes: {'260_1669', '174_3757', '251_3622', '279_5', '128_955', '13_2319', '214_2020', '40_224', '101_1451', '112_1988', '192_4583', '199_72', '21_3190', '161_5846', '240_5477', '133_4621', '37_5293', '71_3409', '254_2803', '257_3857', '48_1647', '72_3801', '126_1969', '148_4493', '171_432', '70_3410', '194_3815', '313_4128', '175_3984', '182_3511', '65_5054', '86_2902', '73_5885', '330_3222', '292_3002', '322_4148', '131_1015', '158_676', '38_2732', '135_2037', '109_917', '147_5026', '8_3336', '35_378', '67_4078', '258_5027', '242_307', '227_4843', '305_4778', '200_2969', '291_5322', '191_2414', '184_996', '24_3635', '79_607', '176_1820', '334_2910', '328_5426', '308_124', '331_2221', '93_1420', '223_2075', '317_6086', '4_4496', '250_4680', '312_4339', '314_2150', '273_4716', '219_2313', '232_1228', '14_1846', '288_1515', '85_7256', '329_1565', '168_3217', '1_4449', '321_4325', '90_1775', '105_1916', '25_3132', '172_1256', '68_2758', '230_2924', '181_3868', '229_3220', '187_1580', '339_145', '215_6421', '276_175', '28_4482', '234_827', '118_697', '3_10453', '221_4319', '285_4334', '195_9557', '55_675', '186_12134', '2_5414', '61_2638', '113_3047', '125_3936', '107_1006', '225_1411', '103_1815', '162_4672', '163_3152', '63_3775', '54_2608', '89_1904', '64_1832', '179_3855', '245_930', '319_5489', '74_692', '226_3656', '307_660', '286_455', '267_5674', '92_4328', '49_870', '111_4823', '235_483', '265_1489', '297_4177', '203_5888', '138_331', '222_2994', '75_2363', '238_464', '99_3741', '57_3174', '268_4752', '36_3461', '77_1306', '167_1002', '294_4427', '259_2804', '151_1494', '62_4349', '43_1775', '137_3042', '17_2942', '281_2193', '31_1147', '336_2285', '212_3241', '160_3309', '173_2292', '218_1320', '335_5316', '193_2051', '278_6209', '76_3232', '340_1262', '110_4180', '116_5279', '274_1803', '208_1372', '202_1420', '333_1891', '34_3290', '275_449', '157_758', '196_1596', '303_3188', '84_731', '306_149', '263_3169', '342_3629', '207_4489', '305_4399', '26_2042', '243_3535', '81_5447', '95_385', '88_464', '197_2062', '153_1100', '217_847', '80_4545', '115_75', '155_4690', '150_2883', '261_597', '204_2320', '269_4976', '41_2850', '210_4154', '170_1878', '272_3500', '318_1818', '249_3659', '143_187', '27_1997', '295_3017', '30_1693', '23_1245', '159_2475', '211_2759', '337_3779', '223_4968', '78_393', '123_2454', '156_4691', '237_352', '296_2299', '136_2744', '139_824', '198_5645', '146_3752', '309_3738', '104_3544', '283_3845', '39_2323', '290_909', '256_3696', '338_3955', '97_223', '166_4013', '253_5557', '29_657', '177_3596', '69_26', '247_1137', '327_2513', '91_4857', '32_4660', '132_489', '206_2786', '271_4754', '19_3013', '43_5567', '325_2468', '66_174', '94_4081', '152_5173', '53_4787', '51_2233', '126_3560', '332_4483', '293_3858', '165_3483', '239_1251', '119_2707', '181_2393', '16_2095', '205_3527', '289_5624', '82_1387', '9_1580', '220_2889', '233_4971', '302_1588', '252_1095', '106_2698', '300_206', '244_5970', '315_1791', '46_3365', '83_4148', '255_1388', '188_5989', '224_5888', '287_1822', '216_7794', '20_3584', '299_268', '326_2457', '127_3120', '65_1249', '183_3713', '241_996', '332_2538', '12_5552', '87_4750', '213_440', '120_2694', '15_4715', '18_5493', '140_3316', '320_3170', '123_357', '154_4519', '124_3599', '185_4130', '145_1300', '310_4617', '236_1132', '312_4', '262_4316', '98_2443', '209_659', '142_1060', '277_1332', '144_4347', '231_3717', '108_2582', '246_8371', '50_299', '284_4410', '282_1139', '114_1814', '270_10', '304_1612', '52_3879', '96_530', '228_933', '100_1482', '180_2761', '58_3758', '201_4715', '266_2110', '189_3885', '311_4358', '149_2274', '130_983', '190_4037', '0_1759', '11_3384', '280_244', '341_604', '169_4977', '298_3901', '47_4053', '22_5392', '264_3062'}\n"
     ]
    }
   ],
   "source": [
    "print('GCN4 is given by the y1000_id: '  + target_y1000_id)\n",
    "#110_4180\n",
    "    \n",
    "og = goi_og_lookup[target_y1000_id]\n",
    "print('It is in orthogroup: ' + og)\n",
    "print('which has the genes: ' + str(og_genes_lookup[og]))\n",
    "\n",
    "\n",
    "# OG2986: 0_1759 100_1482 101_1451 103_1815 104_3544 105_1916 106_2698 108_2582 109_917 110_4180 111_4823 112_1988 113_3047 114_1814 115_75 116_5279 119_2707 124_3599 130_983 131_1015 133_4621 135_2037 136_2744 137_3042 138_331 139_824 140_3316 142_1060 143_187 144_4347 145_1300 146_3752 147_5026 148_4493 150_2883 154_4519 155_4690 156_4691 157_758 158_676 159_2475 15_4715 161_5846 162_4672 166_4013 167_1002 168_3217 170_1878 171_432 173_2292 17_2942 186_12134 187_1580 188_5989 189_3885 191_2414 192_4583 193_2051 195_9557 196_1596 197_2062 199_72 200_2969 201_4715 202_1420 203_5888 204_2320 205_3527 206_2786 207_4489 208_1372 209_659 20_3584 211_2759 212_3241 214_2020 215_6421 216_7794 217_847 21_3190 221_4319 223_2075 223_4968 224_5888 225_1411 230_2924 231_3717 235_483 237_352 239_1251 23_1245 242_307 243_3535 245_930 246_8371 249_3659 251_3622 252_1095 253_5557 257_3857 258_5027 259_2804 260_1669 261_597 265_1489 268_4752 269_4976 270_10 271_4754 272_3500 273_4716 274_1803 275_449 276_175 277_1332 278_6209 279_5 27_1997 280_244 281_2193 282_1139 283_3845 285_4334 287_1822 288_1515 289_5624 292_3002 293_3858 294_4427 295_3017 296_2299 297_4177 298_3901 299_268 29_657 300_206 302_1588 304_1612 305_4399 306_149 307_660 308_124 309_3738 310_4617 311_4358 312_4339 314_2150 317_6086 318_1818 319_5489 31_1147 320_3170 321_4325 322_4148 325_2468 326_2457 329_1565 332_4483 333_1891 334_2910 335_5316 339_145 340_1262 34_3290 35_378 36_3461 37_5293 39_2323 40_224 41_2850 43_5567 47_4053 49_870 4_4496 50_299 52_3879 54_2608 55_675 65_1249 66_174 67_4078 68_2758 70_3410 71_3409 74_692 79_607 82_1387 84_731 85_7256 91_4857 92_4328 94_4081 95_385 96_530 99_3741 9_1580 11_3384 123_2454 127_3120 132_489 149_2274 151_1494 152_5173 153_1100 160_3309 163_3152 172_1256 194_3815 19_3013 1_4449 218_1320 220_2889 222_2994 227_4843 229_3220 22_5392 232_1228 233_4971 240_5477 241_996 244_5970 247_1137 250_4680 254_2803 262_4316 263_3169 264_3062 267_5674 284_4410 290_909 291_5322 2_5414 30_1693 315_1791 328_5426 32_4660 330_3222 331_2221 336_2285 337_3779 338_3955 341_604 342_3629 38_2732 48_1647 51_2233 53_4787 57_3174 58_3758 61_2638 62_4349 63_3775 64_1832 69_26 72_3801 73_5885 75_2363 76_3232 77_1306 78_393 80_4545 81_5447 83_4148 86_2902 87_4750 88_464 89_1904 90_1775 97_223 98_2443 125_3936 126_1969 174_3757 177_3596 179_3855 18_5493 198_5645 219_2313 226_3656 266_2110 28_4482 313_4128 327_2513 46_3365 175_3984 213_440 8_3336 107_1006 120_2694 14_1846 165_3483 176_1820 180_2761 181_2393 182_3511 184_996 190_4037 228_933 234_827 236_1132 238_464 255_1388 26_2042 286_455 303_3188 43_1775 126_3560 169_4977 16_2095 185_4130 256_3696 13_2319 181_3868 183_3713 123_357 25_3132 65_5054 128_955 332_2538 210_4154 93_1420 118_697 12_5552 24_3635 3_10453 312_4 305_4778\n",
    "    \n",
    "# For some reason I didn't have the arthrobotrys_oligospora or the aspergillus_nidulans annotated peptide file. \n",
    "\n",
    "# Also you will have to add the S.Cerevisiae and C.Albicans because those weren't in the same form as the other genomes so I haven't\n",
    "# added them into my code yet\n",
    "\n",
    "\n",
    "#     333,arthrobotrys_oligospora,Arthrobotrys_oligospora,Arthrobotrys_oligospora,#050f07,333,Arthrobotrys oligospora,outgroup,0,outgroup,outgroup,Leu,3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yHAB134_kazachstania_turicensis_160519\n",
      "yHAB159_kazachstania_solicola_160519\n",
      "yHMPu5000034655_botryozyma_nematodophila_160519\n",
      "yHMPu5000041855_candida_ascalaphidarum_160928\n",
      "yHAB158_kazachstania_viticola_160519\n",
      "nakaseomyces_bracarensis\n",
      "priceomyces_haplophilus\n",
      "yHMPu5000034654_aciculoconidium_aculeatum_160519\n",
      "lipomyces_starkeyi\n",
      "candida_sojae\n",
      "yHMPu5000034613_saturnispora_serradocipensis_160519\n",
      "metschnikowia_kamakouana\n",
      "candida_parapsilosis\n",
      "yHMPu5000034709_kluyveromyces_aestuarii_160519\n",
      "yHMPu5000034632_candida_athensensis_160519\n",
      "yHMPu5000035699_cyberlindnera_maclurae_160613\n",
      "yHMPu5000034673_blastobotrys_nivea_160519\n",
      "candida_auris\n",
      "yHMPu5000034594_starmera_quercuum_160519\n",
      "yHMPu5000034886_ogataea_trehalophila_160519\n",
      "yHMPu5000034998_cephaloascus_albidus_160519\n",
      "clavispora_lusitaniae\n",
      "yHMPu5000034624_pichia_nakasei_160519\n",
      "candida_versatilis\n",
      "yHMPu5000034610_saturnispora_hagleri_160519\n",
      "yHMPu5000035688_cyberlindnera_xylosilytica_160519\n",
      "yHMPu5000034758_lipomyces_japonicus_160519\n",
      "yHMPu5000034660_diddensiella_caesifluorescens_160519\n",
      "yHMPu5000034885_ogataea_zsoltii_160519\n",
      "yHMPu5000034897_ogataea_glucozyma_160519\n",
      "yHMPu5000034950_citeromyces_hawaiiensis_160519\n",
      "yarrowia_keelungensis\n",
      "ambrosiozyma_kashinagacola\n",
      "kazachstania_naganishii\n",
      "metschnikowia_shivogae\n",
      "yHMPu5000035040_ambrosiozyma_maleeae_160519\n",
      "yHMPu5000034648_candida_restingae_160519\n",
      "scheffersomyces_stipitis\n",
      "yHMPu5000035703_cyberlindnera_americana_160613\n",
      "yHMPu5000034933_kuraishia_molischiana_160519\n",
      "yHMPu5000035665_middelhovenomyces_tepae_160613\n",
      "cyberlindnera_jadinii\n",
      "candida_intermedia\n",
      "yHMPu5000034615_saturnispora_zaruensis_160519\n",
      "metschnikowia_cerradonensis\n",
      "yHMPu5000035674_kregervanrija_delftensis_160613\n",
      "spathaspora_hagerdaliae\n",
      "metschnikowia_matae\n",
      "metschnikowia_mauinuiana\n",
      "yHAB153_kazachstania_rosinii_160519\n",
      "millerozyma_acaciae\n",
      "yHMPu5000035282_trigonopsis_vinaria_160928\n",
      "metschnikowia_santaceciliae\n",
      "yHMPu5000034890_ogataea_ramenticola_160519\n",
      "metschnikowia_ipomoeae\n",
      "yHMPu5000034887_ogataea_trehaloabstinens_160519\n",
      "kazachstania_africana\n",
      "yHMPu5000041840_candida_wancherniae_160928\n",
      "saccharomyces_cerevisiae\n",
      "yHMPu5000026197_brettanomyces_custersianus_160519\n",
      "yHMPu5000035325_cyberlindnera_petersonii_160928\n",
      "yHMPu5000034612_saturnispora_saitoi_160519\n",
      "yHMPu5000041863_candida_gorgasii_160928\n",
      "metschnikowia_proteae\n",
      "saccharomycopsis_malanga\n",
      "yHMPu5000026137_ambrosiozyma_ambrosiae_160519\n",
      "lachancea_thermotolerans\n",
      "yHAB148_kazachstania_intestinalis_160519\n",
      "yHMPu5000034604_sporopachydermia_lactativora_160519\n",
      "yHMPu5000035643_yarrowia_bubula_160519\n",
      "metschnikowia_lockheadii\n",
      "yHMPu5000035659_saturnispora_dispora_160519\n",
      "yHMPu5000034876_tetrapisispora_iriomotensis_160519\n",
      "yHMPu5000035302_candida_fragi_160928\n",
      "arthrobotrys_oligospora\n",
      "alloascoidea_hylecoeti\n",
      "yHMPu5000034635_nadsonia_fulvescens_160519\n",
      "metschnikowia_hawaiiensis\n",
      "yHMPu5000034605_spencermartinsiella_europaea_160519\n",
      "candida_homilentoma\n",
      "kluyveromyces_lactis\n",
      "metschnikowia_hamakuensis\n",
      "yHMPu5000035242_zygoascus_ofunaensis_160928\n",
      "yHMPu5000035650_trigonopsis_variabilis_160613\n",
      "metschnikowia_hibisci\n",
      "debaryomyces_hansenii\n",
      "yHAB136_kazachstania_bromeliacearum_160519\n",
      "yHMPu5000035301_pichia_terricola_160928\n",
      "yHMPu5000034902_ogataea_minuta_160519\n",
      "wickerhamomyces_anomalus\n",
      "metschnikowia_borealis\n",
      "yHMPu5000041818_magnusiomyces_tetrasperma_160928\n",
      "candida_infanticola\n",
      "pichia_membranifaciens\n",
      "yHMPu5000034946_ambrosiozyma_oregonensis_160519\n",
      "candida_boidinii_JCM9604\n",
      "yHMPu5000034881_torulaspora_pretoriensis_160519\n",
      "yHMPu5000026274_komagataella_populi_160519\n",
      "yHMPu5000035018_candida_canberraensis_160519\n",
      "yHMPu5000035047_barnettozyma_pratensis_160519\n",
      "lachancea_fermentati\n",
      "yHMPu5000026256_zygotorulaspora_mrakii_161215\n",
      "yHMPu5000035671_phaffomyces_antillensis_160613\n",
      "yarrowia_lipolytica\n",
      "yHMPu5000034883_peterozyma_xylosa_160519\n",
      "yHMPu5000035658_starmera_amethionina_160613\n",
      "metschnikowia_bowlesiae\n",
      "lachancea_dasiensis\n",
      "yHMPu5000035041_ambrosiozyma_pseudovanderkliftii_160519\n",
      "torulaspora_delbrueckii\n",
      "yHMPu5000035048_barnettozyma_salicaria_160519\n",
      "metschnikowia_aberdeeniae\n",
      "yHMPu5000034637_ogataea_populiabae_160519\n",
      "yHMPu5000034646_wickerhamiella_cacticola_160519\n",
      "yHMPu5000035667_kurtzmaniella_cleridarum_160928\n",
      "nakaseomyces_castellii\n",
      "yHMPu5000035318_hyphopichia_heimii_160928\n",
      "wickerhamiella_domercqiae\n",
      "ogataea_parapolymorpha\n",
      "yHMPu5000034979_cyberlindnera_misumaiensis_160519\n",
      "spathaspora_arborariae\n",
      "brettanomyces_anomalus\n",
      "kluyveromyces_marxianus\n",
      "yHMPu5000041693_debaryomyces_nepalensis_160928\n",
      "candida_tanzawaensis\n",
      "yHMPu5000034988_candida_fructus_160519\n",
      "saccharomyces_mikatae\n",
      "yarrowia_deformans\n",
      "yHMPu5000035681_candida_gotoi_160519\n",
      "yHMPu5000035011_candida_pyralidae_160519\n",
      "yHMPu5000035268_wickerhamomyces_hampshirensis_160928\n",
      "yHMPu5000035244_candida_incommunis_160928\n",
      "yHMPu5000034669_blastobotrys_raffinofermentans_160519\n",
      "yHMPu5000035271_wickerhamomyces_bovis_160928\n",
      "dekkera_bruxellensis\n",
      "geotrichum_candidum\n",
      "yHMPu5000035046_barnettozyma_populi_160613\n",
      "nadsonia_fulvescens_var_elongata\n",
      "yHMPu5000034710_kluyveromyces_dobzhanskii_160519\n",
      "yHMPu5000034967_candida_freyschussii_160519\n",
      "yHMPu5000035044_barnettozyma_californica_160519\n",
      "yHMPu5000035252_yamadazyma_nakazawae_160928\n",
      "candida_apicola\n",
      "starmerella_bombicola_JCM9596\n",
      "nakaseomyces_bacillisporus\n",
      "yHMPu5000035639_wickerhamomyces_canadensis_160613\n",
      "ogataea_polymorpha\n",
      "yHMPu5000035286_candida_azyma_160928\n",
      "yHMPu5000034899_ogataea_kodamae_160519\n",
      "yHMPu5000035672_phaffomyces_thermotolerans_160613\n",
      "sporopachydermia_quercuum\n",
      "metschnikowia_kipukae\n",
      "ogataea_methanolica\n",
      "wickerhamomyces_ciferrii\n",
      "yHMPu5000034862_zygotorulaspora_florentina_160519\n",
      "tetrapisispora_phaffii\n",
      "yHMPu5000034884_peterozyma_toletana_160519\n",
      "metschnikowia_continentalis\n",
      "yHMPu5000034742_lipomyces_suomiensis_160519\n",
      "yHMPu5000034891_ogataea_pini_160519\n",
      "metschnikowia_bicuspidata\n",
      "saccharomyces_uvarum\n",
      "yHMPu5000035629_yueomyces_sinensis_160519\n",
      "naumovozyma_dairenensis\n",
      "scheffersomyces_lignosus\n",
      "yHMPu5000035274_wickerhamomyces_alni_160928\n",
      "metschnikowia_matae_maris\n",
      "yHMPu5000035031_candida_kruisii_160519\n",
      "zygosaccharomyces_rouxii\n",
      "yHMPu5000035679_candida_rhagii_160613\n",
      "yHMPu5000034978_cyberlindnera_mrakii_160519\n",
      "yHMPu5000034661_dipodascus_albidus_160519\n",
      "yHMPu5000035243_zygoascus_meyerae_160928\n",
      "yHMPu5000034749_lipomyces_mesembrius_160519\n",
      "yHMPu5000034754_lipomyces_arxii_160519\n",
      "yHMPu5000034757_lipomyces_doorenjongii_160519\n",
      "hanseniaspora_vinae\n",
      "yHMPu5000034760_lipomyces_kononenkoae_160519\n",
      "yHMPu5000034670_blastobotrys_proliferans_160519\n",
      "yHMPu5000035707_candida_heveicola_160519\n",
      "yHMPu5000034904_ogataea_nonfermentans_160519\n",
      "yHMPu5000034681_blastobotrys_americana_160519\n",
      "yHMPu5000035335_candida_blattae_160928\n",
      "metschnikowia_dekortum\n",
      "yHMPu5000034999_cephaloascus_fragrans_160519\n",
      "yHMPu5000034952_citeromyces_siamensis_160519\n",
      "yHAB132_kazachstania_martiniae_160519\n",
      "kuraishia_capsulata\n",
      "yHMPu5000035640_yamadazyma_philogaea_160519\n",
      "babjeviella_inositovora\n",
      "candida_albicans\n",
      "yHMPu5000034665_kodamaea_laetipori_160519\n",
      "hyphopichia_burtonii\n",
      "yHAB160_kazachstania_kunashirensis_160519\n",
      "zygosaccharomyces_bailii\n",
      "komagataella_pastoris\n",
      "eremothecium_sinecaudum\n",
      "yHMPu5000034748_lipomyces_oligophaga_160519\n",
      "yHMPu5000041862_candida_golubevii_160928\n",
      "yHMPu5000035296_priceomyces_carsonii_160928\n",
      "yHMPu5000034643_candida_schatavii_160519\n",
      "yHMPu5000035297_priceomyces_castillae_160928\n",
      "candida_glabrata\n",
      "vanderwaltozyma_polyspora\n",
      "candida_succiphila\n",
      "yHMPu5000026145_ambrosiozyma_vanderkliftii_160519\n",
      "yHMPu5000034932_kuraishia_ogatae_160519\n",
      "yHMPu5000035037_candida_montana_160519\n",
      "metschnikowia_drakensbergensis\n",
      "yHAB154_kazachstania_transvaalensis_160519\n",
      "lachancea_nothofagi\n",
      "yHAB143_kazachstania_siamensis_160519\n",
      "yHMPu5000035652_torulaspora_maleeae_160613\n",
      "meyerozyma_guilliermondii\n",
      "yHMPu5000034947_ambrosiozyma_philentoma_160519\n",
      "yHMPu5000035045_barnettozyma_hawaiiensis_160613\n",
      "yHMPu5000026124_ogataea_henricii_160519\n",
      "yHMPu5000035691_candida_salmanticensis_160519\n",
      "yHMPu5000034671_blastobotrys_peoriensis_160519\n",
      "yHMPu5000034631_martiniozyma_abiesophila_161215\n",
      "yHMPu5000035677_kodamaea_ohmeri_160519\n",
      "candida_dubliniensis\n",
      "yHMPu5000034939_komagataella_pseudopastoris_160519\n",
      "yHMPu5000041824_debaryomyces_subglobosus_160928\n",
      "yHMPu5000034918_nakazawaea_holstii_160519\n",
      "yHMPu5000034627_pichia_heedii_160519\n",
      "candida_sorboxylosa\n",
      "yHMPu5000034625_pichia_kudriavzevii_160519\n",
      "yHMPu5000035662_meyerozyma_caribbica_160613\n",
      "candida_orthopsilosis\n",
      "saccharomyces_arboricola\n",
      "yHMPu5000034674_blastobotrys_muscicola_160519\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yHMPu5000035687_cyberlindnera_suaveolens_160613\n",
      "metschnikowia_similis\n",
      "yHMPu5000035690_candida_vartiovaarae_160613\n",
      "metschnikowia_arizonensis\n",
      "yHMPu5000035033_candida_cretensis_160519\n",
      "yHMPu5000041743_candida_hawaiiana_160928\n",
      "yHMPu5000035261_candida_ponderosae_160928\n",
      "nakazawaea_peltata\n",
      "saccharomyces_eubayanus\n",
      "yHMPu5000035670_phaffomyces_opuntiae_160613\n",
      "yHMPu5000041833_candida_tammaniensis_160928\n",
      "yHMPu5000035675_kregervanrija_fluxuum_160613\n",
      "yHMPu5000034866_zygosaccharomyces_bisporus_160519\n",
      "yHMPu5000034622_pichia_occidentalis_160519\n",
      "spathaspora_gorwiae\n",
      "yHMPu5000034597_candida_stellimalicola_160519\n",
      "candida_arabinofermentans\n",
      "lachancea_quebecensis\n",
      "yHMPu5000034611_saturnispora_mendoncae_160519\n",
      "nakaseomyces_delphensis\n",
      "yHMPu5000034667_blastobotrys_serpentis_160519\n",
      "nakaseomyces_nivariensis\n",
      "yHAB133_kazachstania_unispora_160519\n",
      "yHMPu5000034957_hanseniaspora_osmophila_160519\n",
      "yHMPu5000035686_cyberlindnera_saturnus_160519\n",
      "yHMPu5000034893_ogataea_philodendra_160519\n",
      "yHMPu5000041829_debaryomyces_fabryi_160928\n",
      "yHMPu5000035651_torulaspora_microellipsoides_160519\n",
      "pachysolen_tannophilus\n",
      "yHMPu5000034973_danielozyma_ontarioensis_160519\n",
      "arxula_adeninivorans\n",
      "yHMPu5000026142_citeromyces_matritensis_160519\n",
      "yHMPu5000035641_yamadazyma_scolyti_160613\n",
      "eremothecium_gossypii\n",
      "naumovozyma_castellii\n",
      "yHMPu5000034629_pichia_exigua_160519\n",
      "yHMPu5000041678_debaryomyces_prosopidis_160928\n",
      "yHMPu5000034636_ogataea_nitratoaversa_160519\n",
      "eremothecium_cymbalariae\n",
      "candida_carpophila\n",
      "yHMPu5000034606_priceomyces_medius_160519\n",
      "lachancea_cidri\n",
      "candida_tropicalis\n",
      "saccharomyces_paradoxus\n",
      "yHMPu5000035673_candida_orba_160613\n",
      "yHMPu5000034990_candida_corydali_160519\n",
      "spathaspora_passalidarum\n",
      "yHMPu5000041713_debaryomyces_maramus_160928\n",
      "lachancea_kluyveri\n",
      "cyberlindnera_fabianii_JCM3601\n",
      "lodderomyces_elongisporus\n",
      "yHMPu5000034675_blastobotrys_mokoenaii_160519\n",
      "saprochaete_clavata\n",
      "yHMPu5000034761_lipomyces_lipofer_160519\n",
      "yHMPu5000035022_candida_emberorum_160519\n",
      "tetrapisispora_blattae\n",
      "yHAB155_kazachstania_spencerorum_160519\n",
      "yHMPu5000035645_yarrowia_divulgata_160519\n",
      "ambrosiozyma_monospora\n",
      "yHAB166_kazachstania_yakushimaensis_160519\n",
      "yHMPu5000035032_candida_gatunensis_160519\n",
      "spathaspora_girioi\n",
      "yHMPu5000034901_ogataea_methylivora_160519\n",
      "yHMPu5000034875_tetrapisispora_fleetii_160519\n",
      "aspergillus_nidulans\n",
      "yHMPu5000034607_saccharomycopsis_capsularis_160519\n",
      "yHMPu5000034974_deakozyma_indianensis_160519\n",
      "sugiyamaella_lignohabitans\n",
      "wickerhamia_fluorescens\n",
      "saccharomyces_kudriavzevii\n",
      "yHMPu5000034877_tetrapisispora_namnaonensis_160519\n",
      "yHMPu5000034986_candida_oregonensis_160519\n",
      "candida_tenuis\n",
      "yHMPu5000035279_tortispora_starmeri_160928\n",
      "lachancea_mirantina\n",
      "yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519\n",
      "yHAB164_kazachstania_aerobia_160519\n",
      "yHMPu5000035633_candida_hispaniensis_160519\n",
      "yHMPu5000034903_ogataea_naganishii_160519\n",
      "eremothecium_coryli\n",
      "yHMPu5000034892_ogataea_pilisensis_160519\n",
      "lachancea_waltii\n",
      "yHAB147_kazachstania_taianensis_160519\n",
      "yHMPu5000035689_candida_mycetangii_160613\n",
      "yHMPu5000034623_pichia_norvegensis_160519\n",
      "yHMPu5000041822_dipodascus_geniculatus_160928\n"
     ]
    }
   ],
   "source": [
    "#Coding Sequences or Peptide sequences from big file\n",
    "\n",
    "#Table to look up species by number (this is also used for promoters): \n",
    "genome_name_lookup = dict(zip(y1000_species['spec_og_id'],y1000_species['original_genome_id']))\n",
    "\n",
    "#For a given S.C. Paralog Pair, \n",
    "\n",
    "#check that orthogroup is shared between paralogs\n",
    "\n",
    "#Extract orthogroup  given orthogroup\n",
    "\n",
    "# og = \"OG2986\"       #\"OG1299\" # CDC19/PYK2     #\"OG3790\"  # CYC1/CYC7\n",
    "og = goi_og_lookup[target_y1000_id]\n",
    "\n",
    "og_genes = list(og_genes_lookup[og])\n",
    "\n",
    "\n",
    "# #dict to lookup S.Cer genes - this only captures one of the genes since the original dict is 2:1\n",
    "# og_goi_lookup = {og : goi for goi, og in goi_og_lookup.items()}\n",
    "\n",
    "#For each gene in orthogroup, group by species, open PEP file, extract sequence and save as fasta\n",
    "\n",
    "# sc_common_names = ohnologs_goi_og_genes.loc[((ohnologs_goi_og_genes['OG_index_high']== og_goi_lookup[og]) | (ohnologs_goi_og_genes['OG_index_low']==og_goi_lookup[og])) , ['SC_common_name_low', 'SC_common_name_high']] \n",
    "# sc_common_names_str = '_'.join(sc_common_names.values[0])\n",
    "\n",
    "proteins_og_fname = \"/home/heineike/github/expression_broad_data/expression_data/\" + os.path.normpath('promoter_phylogeny/proteins_og/' + og + '_GCN4.fasta')\n",
    "\n",
    "with open(proteins_og_fname,'w') as f: \n",
    "    # group by species\n",
    "    og_genes_specs = {spec_gene.split('_')[0]: [] for spec_gene in og_genes}\n",
    "    for y1000_id in og_genes: \n",
    "        spec_og_id = y1000_id.split('_')[0]\n",
    "        og_genes_specs[spec_og_id].append(y1000_id)\n",
    "\n",
    "    for spec_og_id, genes in og_genes_specs.items(): \n",
    "        genome_name = genome_name_lookup[int(spec_og_id)]\n",
    "        print(genome_name)\n",
    "        #if S.Cer skip finding protein sequences, \n",
    "        \n",
    "        ## Should add protein sequences in case it is s.cer\n",
    "        \n",
    "        if not(genome_name in {'saccharomyces_cerevisiae', 'arthrobotrys_oligospora', 'candida_albicans', 'aspergillus_nidulans'}):    \n",
    "            #load gene_id map based on the species\n",
    "            gene_lookup_spec_fname = y1000plus_dir + \"id_lookups/\" + genome_name + '.csv'\n",
    "            gene_lookup_spec = pd.read_csv(gene_lookup_spec_fname, index_col='y1000_id')\n",
    "\n",
    "            #Extract peptide sequences from peptide fasta from genome\n",
    "            protein_dir = \"/home/heineike/genomes/y1000plus/0_332yeast_genomes/332_genome_annotations/pep/\"\n",
    "\n",
    "            protein_fname = protein_dir + genome_name + '.max.pep'\n",
    "\n",
    "            seq_records = SeqIO.parse(protein_fname, \"fasta\")\n",
    "            \n",
    "            genes_lookup = gene_lookup_spec.loc[genes]\n",
    "            genes_lookup_set = set(gene_lookup_spec.loc[genes,'gene_full'].values)\n",
    "            \n",
    "            for seq_record in seq_records:\n",
    "                #gene_full = 'augustus_masked-Deha2C-processed-gene-4.36'\n",
    "                gene_full = seq_record.description.split()[1].split('=')[1]\n",
    "                #print(gene_full)\n",
    "                if (gene_full in genes_lookup_set):\n",
    "                    #find which y1000_id was matched\n",
    "                    y1000_rlookup = genes_lookup['gene_full'] == gene_full\n",
    "                    for gene, tf in y1000_rlookup.items(): \n",
    "                        if tf:\n",
    "                            y1000_id=gene\n",
    "                    gene_id = genes_lookup.loc[y1000_id, 'gene_id']\n",
    "                    protein_seq = seq_record.seq\n",
    "                    f.write('>' + genome_name + '_' + gene_id + ' y1000_id=' + y1000_id + ' gene_full=' + gene_full +\n",
    "                            ' ' + seq_record.description.split()[2] + #this adds in the cds from the original description\n",
    "                            '\\n')\n",
    "                    f.write(str(protein_seq) + '\\n')  #I wonder why some of the bases were in lower case\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Sort into species\n",
    "#For each species, go into CDS file and extract\n",
    "#also could go into PEP file and extract (which one is better for MSA?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Provide an SCer genename, output a list of orthologs\n",
    "\n",
    "\n",
    "\n",
    "#for a list of orthologs, output a file of protein sequences\n",
    "#provide command for protein alignment\n",
    "#provide command for phylogenetic tree\n",
    "\n",
    "#for a list of orthologs, output a file of promoters\n",
    "\n",
    "#output a protein sequence alignment\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "seqanalysis",
   "language": "python",
   "name": "seqanalysis"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
