{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = othera\n",
      "home directory is C:\\Users\\BMH_work\\\n",
      "y1000plus_tools dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\\n",
      "y1000plus data dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\genomes\\y1000plus\\\n",
      "Importing y1000plus_tools.py\n",
      "sys.path : \n",
      "\n",
      "['C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\python37.zip', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\DLLs', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env', '', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\BMH_work\\\\.ipython', 'C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\BMH_work\\\\github\\\\yeast_esr_expression_analysis\\\\']\n",
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "#clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "clades_of_interest = {'Saccharomycetaceae', 'Saccharomycodaceae'}\n",
    "\n",
    "outgroups = [ 'Wickerhamomyces anomalus','Cyberlindnera jadinii', 'Ascoidea rubescens','Komagataella pastoris','Debaryomyces hansenii', 'Candida albicans', 'Yarrowia lipolytica']\n",
    "#here is a longer curated list of outgroups\n",
    "#nonsacc_visualization_subset = pd.read_csv(y1000plus_dir + 'species_visualization_subset.csv')\n",
    "#outgroup = 'Yarrowia lipolytica'\n",
    "\n",
    "y1000_species_subset = y1000_species[(y1000_species['Major clade'].isin(clades_of_interest))  |  (y1000_species['Species name'].isin(outgroups))]\n",
    "\n",
    "\n",
    "#outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "#y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n",
    "#y1000_species_subset\n",
    "\n",
    "genome_name_lookup = dict(zip(y1000_species_subset['species_names_fig2'], y1000_species_subset['original_genome_id']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nonsacc_visualization_subset = pd.read_csv(y1000plus_dir + 'species_visualization_subset.csv')\n",
    "# species_subset = ( set(nonsacc_visualization_subset['original_genome_id']) | \\\n",
    "#                    set(y1000_species[y1000_species['Major clade']=='Saccharomycetaceae']['original_genome_id']) ) # | \\\n",
    "#                    #set(y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'])\n",
    "#                  #)\n",
    "# y1000_species_subset = y1000_species[(y1000_species['original_genome_id'].isin(species_subset))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load Ohnolog data and make ancestor look up table that links the common name of the high gene to the ancestor. \n",
    "\n",
    "ohnologs_expression_sorted = pd.read_csv(yeast_esr_exp_path + os.path.normpath(\"expression_data/kl_sc_PKA_as_m24_r1g1_20181017/20200607_ohnologs_expression_sorted.csv\"), index_col=0)\n",
    "\n",
    "anc_lookup = dict(zip(ohnologs_expression_sorted['SC_common_name_high'], ohnologs_expression_sorted.index))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "C:\\Users\\BMH_work\\github\\yeast_esr_expression_analysis\\expression_data\\ortholog_files_YGOB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Load GOI ohnolog data\n",
    "# fname = io_library.data_processing_dir + os.sep + os.path.normpath('kl_sc_PKA_as_m24_r1g1_20181017/gois_20181205.pkl')\n",
    "\n",
    "# with open(fname, 'rb') as file:\n",
    "#     [ohnologs_goi,ohnologs_expression_sorted,goi_criteria] = pickle.load(file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'ohnologs_goi' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-6-b073bdaf2974>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mgoi_pair\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'EGO4_EGO2'\u001b[0m \u001b[1;31m#'GPM2_GPM3'  #GND2_GND1 #'EGO4_EGO2'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m \u001b[0manc_lookup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mzip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mohnologs_goi\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'SC_common_name_high'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mohnologs_goi\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'Ancestor'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m \u001b[0mfocus_anc\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0manc_lookup\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mgene\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'ohnologs_goi' is not defined"
     ]
    }
   ],
   "source": [
    "#get surrounding input genenames from YGOB site\n",
    "gene = 'EGO4' #'GND2'  #'YNR034W-A'  #EGO4\n",
    "goi_pair = 'EGO4_EGO2' #'GPM2_GPM3'  #GND2_GND1 #'EGO4_EGO2'\n",
    "\n",
    "anc_lookup = dict(zip(ohnologs_goi['SC_common_name_high'], ohnologs_goi['Ancestor']))\n",
    "focus_anc = anc_lookup[gene]\n",
    "\n",
    "#Extract files using url\n",
    "window = '8'\n",
    "\n",
    "#gene_synteny_YGOB = os.path.normpath(data_processing_dir + \"ortholog_files_YGOB/YGOB_pillars_goi.txt\")\n",
    "\n",
    "#with open(pillars_goi_fname,'w') as f:\n",
    "#    for anc in ohnologs_goi['Ancestor']:\n",
    "\n",
    "YGOB_query_url = 'http://ygob.ucd.ie/cgi/browser/tab.pl?ver=Latest&win=' + window + '&gene=' + focus_anc\n",
    "YGOB_query_response = requests.get(YGOB_query_url)\n",
    "\n",
    "soup = BeautifulSoup(YGOB_query_response.text)\n",
    "table = soup.find(\"table\")\n",
    "\n",
    "output_rows = []\n",
    "for table_row in table.findAll('tr'):\n",
    "    columns = table_row.findAll('td')\n",
    "    output_row = []\n",
    "    for column in columns:\n",
    "        output_row.append(column.text)\n",
    "    output_rows.append(output_row)\n",
    "\n",
    "synteny_window = pd.DataFrame(output_rows[2:],columns = output_rows[1])\n",
    "\n",
    "\n",
    "#synteny_row = synteny_window[synteny_window['Ancestor']==anc]\n",
    "# print(\"\\t\".join(synteny_row.iloc[0,:].values) + '\\n')\n",
    "# f.write(\"\\t\".join(synteny_row.iloc[0,:].values) + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make K.lactis map from Kl genename to ygob name\n",
    "spec = 'Klac'\n",
    "orth_dir = os.path.normpath(io_library.data_processing_dir + 'ortholog_files_y1000'  ) + os.sep\n",
    "\n",
    "klac_YGOB_to_y1000 = io_library.read_orth_lookup_table(spec + '_YGOB', spec + '_y1000', orth_dir)\n",
    "\n",
    "#Make og lookups for S.cer genes\n",
    "klac_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/kluyveromyces_lactis.csv')\n",
    "klac_lookup = pd.read_csv(klac_lookup_fname, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No sequence match for KLLA0C04004g\n"
     ]
    }
   ],
   "source": [
    "kl_y1000ogid_lookup = {}\n",
    "for kl_genename in list(set(synteny_window['K.lactis'])-{'---'}):\n",
    "    kl_genename_y1000_options = klac_YGOB_to_y1000[kl_genename]\n",
    "    if len(kl_genename_y1000_options)==1:\n",
    "        if kl_genename_y1000_options[0] != 'NONE':\n",
    "            y1000_no = kl_genename_y1000_options[0]\n",
    "            kl_y1000ogid_lookup[kl_genename] = klac_lookup.loc[y1000_no,'y1000_id']\n",
    "        else: \n",
    "            print('No sequence match for ' + kl_genename)\n",
    "            kl_y1000ogid_lookup[kl_genename] = 'NONE'\n",
    "    else: \n",
    "        kl_y1000ogid_lookup[kl_genename] = 'AMBIGUOUS'\n",
    "        print('More than one sequence match for ' + kl_genename)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "more than one goi in same orthogroup :OG1137 found_genes: \n",
      "['110_5400', '110_1909']\n",
      "more than one goi in same orthogroup :OG1338 found_genes: \n",
      "['110_5348', '110_5369', '50_2849']\n",
      "more than one goi in same orthogroup :OG1943 found_genes: \n",
      "['110_1913', '50_1961']\n",
      "more than one goi in same orthogroup :OG2637 found_genes: \n",
      "['50_4209', '110_1069']\n",
      "more than one goi in same orthogroup :OG3565 found_genes: \n",
      "['110_4211', '50_2615']\n",
      "more than one goi in same orthogroup :OG3811 found_genes: \n",
      "['50_2242', '110_2238']\n",
      "more than one goi in same orthogroup :OG6005 found_genes: \n",
      "['110_1259', '110_2716', '50_4867']\n"
     ]
    }
   ],
   "source": [
    "gois_sc = list((set(synteny_window['S.cerevisiae A']) | set(synteny_window['S.cerevisiae B']))-{'---'})\n",
    "\n",
    "#Make og lookups for S.cer genes\n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "#target_y1000_id = scer_lookup.loc[gois_sc,'y1000_id']\n",
    "y1000_ids = dict(scer_lookup.loc[gois_sc,'y1000_id'])\n",
    "#y1000_id_list = [y1000_ids[goi] for goi in gois]\n",
    "\n",
    "gois = list((set(kl_y1000ogid_lookup.values()) | set(y1000_ids.values())) - {'NONE'})\n",
    "\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(gois, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "OrderedDict([('Anc_3.169', ['OG1137']),\n",
       "             ('Anc_3.170', ['OG1943']),\n",
       "             ('Anc_3.171', ['OG3565']),\n",
       "             ('Anc_3.172', ['OG6005']),\n",
       "             ('Anc_3.173', ['OG27905', 'OG8256']),\n",
       "             ('Anc_3.174', ['OG3811']),\n",
       "             ('Anc_3.175', ['OG1338']),\n",
       "             ('Anc_3.176', ['OG2637'])])"
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spec_og_search = ['S.cerevisiae A', 'S.cerevisiae B', 'K.lactis']\n",
    "\n",
    "anc_genes = []\n",
    "for anc in synteny_window['Ancestor']: \n",
    "    if anc != '---':\n",
    "        anc_genes.append((anc, dict(synteny_window.loc[synteny_window['Ancestor']==anc, spec_og_search].iloc[0,:])))\n",
    "# For Ancestor Column, Make list of tuple: species, genename.  \n",
    "\n",
    "anc_genes_ogs = OrderedDict()\n",
    "anc_genes_ogs_flat = OrderedDict()\n",
    "for anc, gene_dict in anc_genes: \n",
    "    anc_og_dict = {}\n",
    "    for spec in spec_og_search: \n",
    "        gene = gene_dict[spec]\n",
    "        \n",
    "        og = ''\n",
    "        \n",
    "        if gene !='---': \n",
    "            if 'S.cerevisiae' in spec: \n",
    "                y1000_id = y1000_ids[gene]\n",
    "                og = goi_og_lookup[y1000_id]\n",
    "            elif spec == 'K.lactis':\n",
    "                y1000_id = kl_y1000ogid_lookup[gene]\n",
    "                if not(y1000_id in {'NONE', 'AMBIGUOUS'}):\n",
    "                    og = goi_og_lookup[y1000_id]\n",
    "        anc_og_dict[spec] = og\n",
    "    anc_genes_ogs[anc]  = anc_og_dict\n",
    "    anc_genes_ogs_flat[anc] = list((set(anc_og_dict.values()) - {''}))\n",
    "    \n",
    "anc_genes_ogs_flat\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Yueomyces_sinensis\n",
      "Tetrapisispora_blattae\n",
      "Vanderwaltozyma_polyspora\n",
      "Tetrapisispora_iriomotensis\n",
      "Tetrapisispora_phaffii\n",
      "Tetrapisispora_fleetii\n",
      "Tetrapisispora_namnaonensis\n",
      "Candida_castellii\n",
      "Nakaseomyces_bacillisporus\n",
      "Candida_glabrata\n",
      "Candida_bracarensis\n",
      "Nakaseomyces_delphensis\n",
      "Candida_nivariensis\n",
      "Saccharomyces_eubayanus\n",
      "Saccharomyces_uvarum\n",
      "More than one CDS - introns?.  N of CDS: 8\n",
      "Saccharomyces_arboricola\n",
      "Saccharomyces_kudriavzevii\n",
      "Saccharomyces_mikatae\n",
      "More than one CDS - introns?.  N of CDS: 10\n",
      "Saccharomyces_paradoxus\n",
      "Naumovozyma_castellii\n",
      "Naumovozyma_dairenensis\n",
      "Kazachstania_viticola\n",
      "Kazachstania_africana\n",
      "More than one CDS - introns?.  N of CDS: 5\n",
      "Kazachstania_rosinii\n",
      "Kazachstania_spencerorum\n",
      "Kazachstania_kunashirensis\n",
      "Kazachstania_turicensis\n",
      "Kazachstania_martiniae\n",
      "Kazachstania_intestinalis\n",
      "Kazachstania_bromeliacearum\n",
      "Kazachstania_naganishii\n",
      "Kazachstania_taianensis\n",
      "Kazachstania_yakushimaensis\n",
      "Kazachstania_transvaalensis\n",
      "Kazachstania_siamensis\n",
      "Kazachstania_unispora\n",
      "Kazachstania_aerobia\n",
      "Kazachstania_solicola\n"
     ]
    }
   ],
   "source": [
    "#Make list of post-WGH species\n",
    "#set up data matrix based on time-calibrated species tree, and take subset of species based on species subset\n",
    "tree_fname = y1000plus_dir + '332_2408OGs_time-calibrated_phylogeny_species-names_updated.newick'\n",
    "t = Tree(tree_fname)\n",
    "t.ladderize()\n",
    "\n",
    "post_wgh_specs = t.get_common_ancestor(['Vanderwaltozyma_polyspora', 'Saccharomyces_cerevisiae']).get_leaves()\n",
    "post_wgh_specs_names = [node.name for node in post_wgh_specs]\n",
    "post_wgh_specs_names.remove('Saccharomyces_cerevisiae')\n",
    "#For each species, load gtf, get surrounding genes, compare surrounding genes with orthologs (from ogs) of surrounding genes\n",
    "bp_range = 20000  #Base pair range - goal is to get at least 8 CDS features\n",
    "\n",
    "\n",
    "synteny_window_new_cols = synteny_window\n",
    "\n",
    "for spec_fig2 in post_wgh_specs_names: \n",
    "    print(spec_fig2)\n",
    "    focus_spec = genome_name_lookup[spec_fig2]\n",
    "    #focus_spec = 'saccharomyces_arboricola'\n",
    "    \n",
    "    #'saccharomyces_uvarum',\n",
    "    #  'saccharomyces_eubayanus',\n",
    "    #  'saccharomyces_kudriavzevii',\n",
    "    #  'saccharomyces_arboricola'\n",
    "\n",
    "    #Get focus species genes that are present in focus ogs\n",
    "    spec_og_id_lookup = dict(zip( y1000_species_subset['original_genome_id'], y1000_species_subset['spec_og_id']))\n",
    "    spec_og_id = spec_og_id_lookup[focus_spec]\n",
    "    focus_ogs = anc_genes_ogs_flat[focus_anc]\n",
    "\n",
    "    og_genes = set()\n",
    "    for og in focus_ogs: \n",
    "        og_genes = og_genes | og_genes_lookup[og]\n",
    "\n",
    "    focus_spec_genes = []\n",
    "    for og_gene in list(og_genes): \n",
    "        spec_og_id_test = int(og_gene.split('_')[0])\n",
    "        if spec_og_id == spec_og_id_test:\n",
    "            focus_spec_genes.append(og_gene)\n",
    "\n",
    "    #Make full_gene lookup for focus spec\n",
    "    focus_spec_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/' + focus_spec + '.csv')\n",
    "    focus_spec_lookup = pd.read_csv(focus_spec_lookup_fname, index_col=0)\n",
    "    y1000_id_gene_full_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup['gene_full']))        \n",
    "    gene_full_y1000_id_lookup = dict(zip(focus_spec_lookup['gene_full'],focus_spec_lookup['y1000_id']))    \n",
    "    y1000_id_gene_id_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup.index))    \n",
    "\n",
    "    # Load GTF for given sequence: \n",
    "    gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "    db_fname = gtf_dir + 'gffutils_dbs/' + focus_spec + '.db'\n",
    "\n",
    "    gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "\n",
    "    # Make a dictionary of scaffold sizes\n",
    "    genome_assembly_fname = y1000plus_dir + os.path.normpath('0_332yeast_genomes/332_genome_assemblies/' + focus_spec + '.fas')\n",
    "\n",
    "    scaffold_sizes = {}\n",
    "    for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "        scaffold_sizes[seq.id] = len(seq.seq)\n",
    "\n",
    "\n",
    "    #For each of these genes, find the surrounding genes\n",
    "    for og_gene in focus_spec_genes: \n",
    "        #og_gene = '109_2086'\n",
    "        focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "        gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "        cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "        all_features = cursor.fetchall()\n",
    "        if len(all_features) == 0:\n",
    "            print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "        CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "        if len(CDS_list)==1: \n",
    "            start = CDS_list[0]['start']\n",
    "            end = CDS_list[0]['end']\n",
    "        elif len(CDS_list)>1: \n",
    "            print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "            start = min([CDS['start'] for CDS in CDS_list])\n",
    "            end = max([CDS['end'] for CDS in CDS_list])\n",
    "        else: \n",
    "            raise AssertionError('No CDS found' + og_gene)\n",
    "                  \n",
    "        CDS = CDS_list[0]\n",
    "        strand = CDS['strand']\n",
    "        scaffold = CDS['seqid']\n",
    "\n",
    "        # if strand=='+': \n",
    "        #     lower = start\n",
    "        #     higher = end\n",
    "        # elif strand == '-': \n",
    "        #     lower = end\n",
    "        #     higher = start\n",
    "\n",
    "        low = max(0,start-bp_range/2)\n",
    "        high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "        surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "        #surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "        surrounding_CDS_data = []\n",
    "        for CDS_surr in surrounding_CDS:\n",
    "            surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "        #Find OG and line that each gene is in\n",
    "        anc_focus_spec_genes = OrderedDict([(anc,[]) for anc in anc_genes_ogs_flat.keys()])\n",
    "        for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "            y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "            for anc,og_list in anc_genes_ogs_flat.items():\n",
    "                for og in og_list: \n",
    "                    og_genes = og_genes_lookup[og]\n",
    "                    if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                        anc_focus_spec_genes[anc].append(y1000_id_gene_id_lookup[y1000_id])\n",
    "                        #could make triple dashes if len(anc_focus_spec_genes[anc]) == 0:\n",
    "                #remove redundancies created by genes with introns\n",
    "                anc_focus_spec_genes[anc]=list(set(anc_focus_spec_genes[anc]))\n",
    "        #Merge into synteny window dataframe\n",
    "\n",
    "        focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "        synteny_window_new_cols = synteny_window_new_cols.merge(focus_gene_column, left_on='Ancestor', right_index = True, how='left')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Missing ingredients: \n",
    "\n",
    "#synteny_window"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#For a given example pair, find synteny of genes surrounding the ortholog in y1000 species not contained in YGOB.  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'synteny_window' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-5-bc687aa8d564>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 14\u001b[1;33m \u001b[0msynteny_window_new_cols\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msynteny_window\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'synteny_window' is not defined"
     ]
    }
   ],
   "source": [
    "#Make list of post-WGH species\n",
    "#set up data matrix based on time-calibrated species tree, and take subset of species based on species subset\n",
    "tree_fname = y1000plus_dir + '332_2408OGs_time-calibrated_phylogeny_species-names_updated.newick'\n",
    "t = Tree(tree_fname)\n",
    "t.ladderize()\n",
    "\n",
    "post_wgh_specs = t.get_common_ancestor(['Vanderwaltozyma_polyspora', 'Saccharomyces_cerevisiae']).get_leaves()\n",
    "post_wgh_specs_names = [node.name for node in post_wgh_specs]\n",
    "post_wgh_specs_names.remove('Saccharomyces_cerevisiae')\n",
    "#For each species, load gtf, get surrounding genes, compare surrounding genes with orthologs (from ogs) of surrounding genes\n",
    "bp_range = 20000  #Base pair range - goal is to get at least 8 CDS features\n",
    "\n",
    "\n",
    "#synteny_window_new_cols = synteny_window\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Given a species, find genes surrounding ortholog\n",
    "for spec_fig2 in post_wgh_specs_names: \n",
    "    print(spec_fig2)\n",
    "    focus_spec = genome_name_lookup[spec_fig2]\n",
    "    #focus_spec = 'saccharomyces_arboricola'\n",
    "    \n",
    "    #'saccharomyces_uvarum',\n",
    "    #  'saccharomyces_eubayanus',\n",
    "    #  'saccharomyces_kudriavzevii',\n",
    "    #  'saccharomyces_arboricola'\n",
    "\n",
    "    #Get focus species genes that are present in focus ogs\n",
    "    spec_og_id_lookup = dict(zip( y1000_species_subset['original_genome_id'], y1000_species_subset['spec_og_id']))\n",
    "    spec_og_id = spec_og_id_lookup[focus_spec]\n",
    "    focus_ogs = anc_genes_ogs_flat[focus_anc]\n",
    "\n",
    "    og_genes = set()\n",
    "    for og in focus_ogs: \n",
    "        og_genes = og_genes | og_genes_lookup[og]\n",
    "\n",
    "    focus_spec_genes = []\n",
    "    for og_gene in list(og_genes): \n",
    "        spec_og_id_test = int(og_gene.split('_')[0])\n",
    "        if spec_og_id == spec_og_id_test:\n",
    "            focus_spec_genes.append(og_gene)\n",
    "\n",
    "    #Make full_gene lookup for focus spec\n",
    "    focus_spec_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/' + focus_spec + '.csv')\n",
    "    focus_spec_lookup = pd.read_csv(focus_spec_lookup_fname, index_col=0)\n",
    "    y1000_id_gene_full_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup['gene_full']))        \n",
    "    gene_full_y1000_id_lookup = dict(zip(focus_spec_lookup['gene_full'],focus_spec_lookup['y1000_id']))    \n",
    "    y1000_id_gene_id_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup.index))    \n",
    "\n",
    "    # Load GTF for given sequence: \n",
    "    gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "    db_fname = gtf_dir + 'gffutils_dbs/' + focus_spec + '.db'\n",
    "\n",
    "    gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "\n",
    "    # Make a dictionary of scaffold sizes\n",
    "    genome_assembly_fname = y1000plus_dir + os.path.normpath('0_332yeast_genomes/332_genome_assemblies/' + focus_spec + '.fas')\n",
    "\n",
    "    scaffold_sizes = {}\n",
    "    for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "        scaffold_sizes[seq.id] = len(seq.seq)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "og_gene = '109_2086'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "og_gene, y1000_gene_id_id_lookup, y1000_id_gene_full_lookup, gtf_db, bp_range, scaffold_sizes, anc_genes_ogs_flat, focus_spec_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for a given gene find the surrounding genes: \n",
    "#def y1000_find_surrounding_genes(og_gene, y1000_gene_id_id_lookup, y1000_id_gene_full_lookup, gtf_db, bp_range, scaffold_sizes, anc_genes_ogs_flat, focus_spec_genes):\n",
    "\n",
    "\n",
    "focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "all_features = cursor.fetchall()\n",
    "if len(all_features) == 0:\n",
    "    print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "if len(CDS_list)==1: \n",
    "    start = CDS_list[0]['start']\n",
    "    end = CDS_list[0]['end']\n",
    "elif len(CDS_list)>1: \n",
    "    print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "    start = min([CDS['start'] for CDS in CDS_list])\n",
    "    end = max([CDS['end'] for CDS in CDS_list])\n",
    "else: \n",
    "    raise AssertionError('No CDS found' + og_gene)\n",
    "\n",
    "CDS = CDS_list[0]\n",
    "strand = CDS['strand']\n",
    "scaffold = CDS['seqid']\n",
    "\n",
    "# if strand=='+': \n",
    "#     lower = start\n",
    "#     higher = end\n",
    "# elif strand == '-': \n",
    "#     lower = end\n",
    "#     higher = start\n",
    "\n",
    "low = max(0,start-bp_range/2)\n",
    "high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "#surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "surrounding_CDS_data = []\n",
    "for CDS_surr in surrounding_CDS:\n",
    "    surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "#Find OG and line that each gene is in\n",
    "anc_focus_spec_genes = OrderedDict([(anc,[]) for anc in anc_genes_ogs_flat.keys()])\n",
    "for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "    y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "    for anc,og_list in anc_genes_ogs_flat.items():\n",
    "        for og in og_list: \n",
    "            og_genes = og_genes_lookup[og]\n",
    "            if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                anc_focus_spec_genes[anc].append(y1000_id_gene_id_lookup[y1000_id])\n",
    "                #could make triple dashes if len(anc_focus_spec_genes[anc]) == 0:\n",
    "        #remove redundancies created by genes with introns\n",
    "        anc_focus_spec_genes[anc]=list(set(anc_focus_spec_genes[anc]))\n",
    "#Merge into synteny window dataframe\n",
    "\n",
    "focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "\n",
    "\n",
    "#Not sure if I will need this in the function\n",
    "#synteny_window_new_cols = synteny_window_new_cols.merge(focus_gene_column, left_on='Ancestor', right_index = True, how='left')\n",
    "\n",
    "#return focus_gene_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>V.polyspora B</th>\n",
       "      <th>T.phaffii B</th>\n",
       "      <th>T.blattae B</th>\n",
       "      <th>N.dairenensis B</th>\n",
       "      <th>N.castellii B</th>\n",
       "      <th>K.naganishii B</th>\n",
       "      <th>K.africana B</th>\n",
       "      <th>C.glabrata B</th>\n",
       "      <th>S.uvarum B</th>\n",
       "      <th>S.kudriavzevii B</th>\n",
       "      <th>...</th>\n",
       "      <th>yHAB153_kazachstania_rosinii_160519_Seq_2148</th>\n",
       "      <th>yHAB134_kazachstania_turicensis_160519_Seq_1990</th>\n",
       "      <th>yHAB132_kazachstania_martiniae_160519_Seq_4635</th>\n",
       "      <th>yHAB147_kazachstania_taianensis_160519_Seq_2501</th>\n",
       "      <th>yHAB166_kazachstania_yakushimaensis_160519_Seq_4897</th>\n",
       "      <th>yHAB154_kazachstania_transvaalensis_160519_Seq_2083</th>\n",
       "      <th>yHAB143_kazachstania_siamensis_160519_Seq_1903</th>\n",
       "      <th>yHAB133_kazachstania_unispora_160519_Seq_2374</th>\n",
       "      <th>yHAB164_kazachstania_aerobia_160519_Seq_2028</th>\n",
       "      <th>yHAB159_kazachstania_solicola_160519_Seq_4608</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NDAI0G02950</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NDAI0G02960</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Kpol_1045.48</td>\n",
       "      <td>TPHA0P00620</td>\n",
       "      <td>TBLA0F00800</td>\n",
       "      <td>NDAI0G02970</td>\n",
       "      <td>NCAS0C03620</td>\n",
       "      <td>KNAG0K01170</td>\n",
       "      <td>KAFR0C01000</td>\n",
       "      <td>CAGL0C05137g</td>\n",
       "      <td>Suva_15.106</td>\n",
       "      <td>Skud_15.94</td>\n",
       "      <td>...</td>\n",
       "      <td>[Seq_1994]</td>\n",
       "      <td>[Seq_416]</td>\n",
       "      <td>[Seq_4557]</td>\n",
       "      <td>[Seq_1317]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_4975]</td>\n",
       "      <td>[Seq_4773]</td>\n",
       "      <td>[Seq_4978]</td>\n",
       "      <td>[Seq_1405]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>TBLA0C05710</td>\n",
       "      <td>NDAI0G02980</td>\n",
       "      <td>NCAS0C03630</td>\n",
       "      <td>KNAG0K01180</td>\n",
       "      <td>KAFR0C00990</td>\n",
       "      <td>CAGL0C05115g</td>\n",
       "      <td>Suva_15.107</td>\n",
       "      <td>Skud_15.95</td>\n",
       "      <td>...</td>\n",
       "      <td>[Seq_2057]</td>\n",
       "      <td>[Seq_722]</td>\n",
       "      <td>[Seq_2588]</td>\n",
       "      <td>[Seq_800]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_2775]</td>\n",
       "      <td>[Seq_5094]</td>\n",
       "      <td>[Seq_122]</td>\n",
       "      <td>[Seq_4853]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Kpol_1045.49</td>\n",
       "      <td>TPHA0P00610</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NCAS0C03640</td>\n",
       "      <td>KNAG0K01190</td>\n",
       "      <td>KAFR0C00980</td>\n",
       "      <td>CAGL0C05093g</td>\n",
       "      <td>Suva_15.108</td>\n",
       "      <td>Skud_15.96</td>\n",
       "      <td>...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_3134]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_2761]</td>\n",
       "      <td>[Seq_4063]</td>\n",
       "      <td>[Seq_5631]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Kpol_1045.50</td>\n",
       "      <td>TPHA0P00600</td>\n",
       "      <td>TBLA0G00870</td>\n",
       "      <td>NDAI0G02990</td>\n",
       "      <td>NCAS0C03650</td>\n",
       "      <td>KNAG0K01200</td>\n",
       "      <td>KAFR0C00970</td>\n",
       "      <td>---</td>\n",
       "      <td>Suva_15.109</td>\n",
       "      <td>Skud_15.97</td>\n",
       "      <td>...</td>\n",
       "      <td>[Seq_2148]</td>\n",
       "      <td>[Seq_1990]</td>\n",
       "      <td>[Seq_4635]</td>\n",
       "      <td>[Seq_2501]</td>\n",
       "      <td>[Seq_4897]</td>\n",
       "      <td>[Seq_2083]</td>\n",
       "      <td>[Seq_1903]</td>\n",
       "      <td>[Seq_2374]</td>\n",
       "      <td>[Seq_2028]</td>\n",
       "      <td>[Seq_4608]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>TBLA0G00860</td>\n",
       "      <td>NDAI0G03000</td>\n",
       "      <td>NCAS0C03660</td>\n",
       "      <td>KNAG0K01210</td>\n",
       "      <td>KAFR0C00960</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Kpol_1045.51</td>\n",
       "      <td>TPHA0P00590</td>\n",
       "      <td>TBLA0G00850</td>\n",
       "      <td>NDAI0G03010</td>\n",
       "      <td>NCAS0C03670</td>\n",
       "      <td>KNAG0K01220</td>\n",
       "      <td>KAFR0C00950</td>\n",
       "      <td>CAGL0C05071g</td>\n",
       "      <td>Suva_15.110</td>\n",
       "      <td>Skud_15.98</td>\n",
       "      <td>...</td>\n",
       "      <td>[Seq_3792]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_2534]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[Seq_5247]</td>\n",
       "      <td>[Seq_2749]</td>\n",
       "      <td>[Seq_4213]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Kpol_1045.52</td>\n",
       "      <td>TPHA0P00580</td>\n",
       "      <td>TBLA0G00840</td>\n",
       "      <td>NDAI0G03020</td>\n",
       "      <td>NCAS0C03680</td>\n",
       "      <td>KNAG0K01230</td>\n",
       "      <td>KAFR0C00940</td>\n",
       "      <td>CAGL0C05049g</td>\n",
       "      <td>Suva_1.110</td>\n",
       "      <td>Skud_1.91</td>\n",
       "      <td>...</td>\n",
       "      <td>[Seq_839]</td>\n",
       "      <td>[Seq_3664]</td>\n",
       "      <td>[Seq_3574]</td>\n",
       "      <td>[Seq_5024]</td>\n",
       "      <td>[Seq_677]</td>\n",
       "      <td>[Seq_2783]</td>\n",
       "      <td>[Seq_2919]</td>\n",
       "      <td>[Seq_3661]</td>\n",
       "      <td>[Seq_2798]</td>\n",
       "      <td>[Seq_2789]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Suva_1.109</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Kpol_1045.53</td>\n",
       "      <td>TPHA0P00570</td>\n",
       "      <td>TBLA0G00830</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>CAGL0C05027g</td>\n",
       "      <td>Suva_1.108</td>\n",
       "      <td>Skud_1.89</td>\n",
       "      <td>...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17 rows Ã— 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   V.polyspora B  T.phaffii B  T.blattae B N.dairenensis B N.castellii B  \\\n",
       "0            ---          ---          ---     NDAI0G02950           ---   \n",
       "1            ---          ---          ---     NDAI0G02960           ---   \n",
       "2            ---          ---          ---             ---           ---   \n",
       "3   Kpol_1045.48  TPHA0P00620  TBLA0F00800     NDAI0G02970   NCAS0C03620   \n",
       "4            ---          ---          ---             ---           ---   \n",
       "5            ---          ---          ---             ---           ---   \n",
       "6            ---          ---  TBLA0C05710     NDAI0G02980   NCAS0C03630   \n",
       "7   Kpol_1045.49  TPHA0P00610          ---             ---   NCAS0C03640   \n",
       "8   Kpol_1045.50  TPHA0P00600  TBLA0G00870     NDAI0G02990   NCAS0C03650   \n",
       "9            ---          ---  TBLA0G00860     NDAI0G03000   NCAS0C03660   \n",
       "10  Kpol_1045.51  TPHA0P00590  TBLA0G00850     NDAI0G03010   NCAS0C03670   \n",
       "11           ---          ---          ---             ---           ---   \n",
       "12  Kpol_1045.52  TPHA0P00580  TBLA0G00840     NDAI0G03020   NCAS0C03680   \n",
       "13           ---          ---          ---             ---           ---   \n",
       "14  Kpol_1045.53  TPHA0P00570  TBLA0G00830             ---           ---   \n",
       "15           ---          ---          ---             ---           ---   \n",
       "16           ---          ---          ---             ---           ---   \n",
       "\n",
       "   K.naganishii B K.africana B  C.glabrata B   S.uvarum B S.kudriavzevii B  \\\n",
       "0             ---          ---           ---          ---              ---   \n",
       "1             ---          ---           ---          ---              ---   \n",
       "2             ---          ---           ---          ---              ---   \n",
       "3     KNAG0K01170  KAFR0C01000  CAGL0C05137g  Suva_15.106       Skud_15.94   \n",
       "4             ---          ---           ---          ---              ---   \n",
       "5             ---          ---           ---          ---              ---   \n",
       "6     KNAG0K01180  KAFR0C00990  CAGL0C05115g  Suva_15.107       Skud_15.95   \n",
       "7     KNAG0K01190  KAFR0C00980  CAGL0C05093g  Suva_15.108       Skud_15.96   \n",
       "8     KNAG0K01200  KAFR0C00970           ---  Suva_15.109       Skud_15.97   \n",
       "9     KNAG0K01210  KAFR0C00960           ---          ---              ---   \n",
       "10    KNAG0K01220  KAFR0C00950  CAGL0C05071g  Suva_15.110       Skud_15.98   \n",
       "11            ---          ---           ---          ---              ---   \n",
       "12    KNAG0K01230  KAFR0C00940  CAGL0C05049g   Suva_1.110        Skud_1.91   \n",
       "13            ---          ---           ---   Suva_1.109              ---   \n",
       "14            ---          ---  CAGL0C05027g   Suva_1.108        Skud_1.89   \n",
       "15            ---          ---           ---          ---              ---   \n",
       "16            ---          ---           ---          ---              ---   \n",
       "\n",
       "                        ...                       \\\n",
       "0                       ...                        \n",
       "1                       ...                        \n",
       "2                       ...                        \n",
       "3                       ...                        \n",
       "4                       ...                        \n",
       "5                       ...                        \n",
       "6                       ...                        \n",
       "7                       ...                        \n",
       "8                       ...                        \n",
       "9                       ...                        \n",
       "10                      ...                        \n",
       "11                      ...                        \n",
       "12                      ...                        \n",
       "13                      ...                        \n",
       "14                      ...                        \n",
       "15                      ...                        \n",
       "16                      ...                        \n",
       "\n",
       "   yHAB153_kazachstania_rosinii_160519_Seq_2148  \\\n",
       "0                                           NaN   \n",
       "1                                           NaN   \n",
       "2                                           NaN   \n",
       "3                                    [Seq_1994]   \n",
       "4                                           NaN   \n",
       "5                                           NaN   \n",
       "6                                    [Seq_2057]   \n",
       "7                                            []   \n",
       "8                                    [Seq_2148]   \n",
       "9                                            []   \n",
       "10                                   [Seq_3792]   \n",
       "11                                          NaN   \n",
       "12                                    [Seq_839]   \n",
       "13                                          NaN   \n",
       "14                                           []   \n",
       "15                                          NaN   \n",
       "16                                          NaN   \n",
       "\n",
       "   yHAB134_kazachstania_turicensis_160519_Seq_1990  \\\n",
       "0                                              NaN   \n",
       "1                                              NaN   \n",
       "2                                              NaN   \n",
       "3                                        [Seq_416]   \n",
       "4                                              NaN   \n",
       "5                                              NaN   \n",
       "6                                        [Seq_722]   \n",
       "7                                               []   \n",
       "8                                       [Seq_1990]   \n",
       "9                                               []   \n",
       "10                                              []   \n",
       "11                                             NaN   \n",
       "12                                      [Seq_3664]   \n",
       "13                                             NaN   \n",
       "14                                              []   \n",
       "15                                             NaN   \n",
       "16                                             NaN   \n",
       "\n",
       "   yHAB132_kazachstania_martiniae_160519_Seq_4635  \\\n",
       "0                                             NaN   \n",
       "1                                             NaN   \n",
       "2                                             NaN   \n",
       "3                                      [Seq_4557]   \n",
       "4                                             NaN   \n",
       "5                                             NaN   \n",
       "6                                      [Seq_2588]   \n",
       "7                                              []   \n",
       "8                                      [Seq_4635]   \n",
       "9                                              []   \n",
       "10                                     [Seq_2534]   \n",
       "11                                            NaN   \n",
       "12                                     [Seq_3574]   \n",
       "13                                            NaN   \n",
       "14                                             []   \n",
       "15                                            NaN   \n",
       "16                                            NaN   \n",
       "\n",
       "   yHAB147_kazachstania_taianensis_160519_Seq_2501  \\\n",
       "0                                              NaN   \n",
       "1                                              NaN   \n",
       "2                                              NaN   \n",
       "3                                       [Seq_1317]   \n",
       "4                                              NaN   \n",
       "5                                              NaN   \n",
       "6                                        [Seq_800]   \n",
       "7                                       [Seq_3134]   \n",
       "8                                       [Seq_2501]   \n",
       "9                                               []   \n",
       "10                                              []   \n",
       "11                                             NaN   \n",
       "12                                      [Seq_5024]   \n",
       "13                                             NaN   \n",
       "14                                              []   \n",
       "15                                             NaN   \n",
       "16                                             NaN   \n",
       "\n",
       "   yHAB166_kazachstania_yakushimaensis_160519_Seq_4897  \\\n",
       "0                                                 NaN    \n",
       "1                                                 NaN    \n",
       "2                                                 NaN    \n",
       "3                                                  []    \n",
       "4                                                 NaN    \n",
       "5                                                 NaN    \n",
       "6                                                  []    \n",
       "7                                                  []    \n",
       "8                                          [Seq_4897]    \n",
       "9                                                  []    \n",
       "10                                                 []    \n",
       "11                                                NaN    \n",
       "12                                          [Seq_677]    \n",
       "13                                                NaN    \n",
       "14                                                 []    \n",
       "15                                                NaN    \n",
       "16                                                NaN    \n",
       "\n",
       "   yHAB154_kazachstania_transvaalensis_160519_Seq_2083  \\\n",
       "0                                                 NaN    \n",
       "1                                                 NaN    \n",
       "2                                                 NaN    \n",
       "3                                                  []    \n",
       "4                                                 NaN    \n",
       "5                                                 NaN    \n",
       "6                                                  []    \n",
       "7                                                  []    \n",
       "8                                          [Seq_2083]    \n",
       "9                                                  []    \n",
       "10                                                 []    \n",
       "11                                                NaN    \n",
       "12                                         [Seq_2783]    \n",
       "13                                                NaN    \n",
       "14                                                 []    \n",
       "15                                                NaN    \n",
       "16                                                NaN    \n",
       "\n",
       "   yHAB143_kazachstania_siamensis_160519_Seq_1903  \\\n",
       "0                                             NaN   \n",
       "1                                             NaN   \n",
       "2                                             NaN   \n",
       "3                                      [Seq_4975]   \n",
       "4                                             NaN   \n",
       "5                                             NaN   \n",
       "6                                      [Seq_2775]   \n",
       "7                                              []   \n",
       "8                                      [Seq_1903]   \n",
       "9                                              []   \n",
       "10                                             []   \n",
       "11                                            NaN   \n",
       "12                                     [Seq_2919]   \n",
       "13                                            NaN   \n",
       "14                                             []   \n",
       "15                                            NaN   \n",
       "16                                            NaN   \n",
       "\n",
       "   yHAB133_kazachstania_unispora_160519_Seq_2374  \\\n",
       "0                                            NaN   \n",
       "1                                            NaN   \n",
       "2                                            NaN   \n",
       "3                                     [Seq_4773]   \n",
       "4                                            NaN   \n",
       "5                                            NaN   \n",
       "6                                     [Seq_5094]   \n",
       "7                                     [Seq_2761]   \n",
       "8                                     [Seq_2374]   \n",
       "9                                             []   \n",
       "10                                    [Seq_5247]   \n",
       "11                                           NaN   \n",
       "12                                    [Seq_3661]   \n",
       "13                                           NaN   \n",
       "14                                            []   \n",
       "15                                           NaN   \n",
       "16                                           NaN   \n",
       "\n",
       "   yHAB164_kazachstania_aerobia_160519_Seq_2028  \\\n",
       "0                                           NaN   \n",
       "1                                           NaN   \n",
       "2                                           NaN   \n",
       "3                                    [Seq_4978]   \n",
       "4                                           NaN   \n",
       "5                                           NaN   \n",
       "6                                     [Seq_122]   \n",
       "7                                    [Seq_4063]   \n",
       "8                                    [Seq_2028]   \n",
       "9                                            []   \n",
       "10                                   [Seq_2749]   \n",
       "11                                          NaN   \n",
       "12                                   [Seq_2798]   \n",
       "13                                          NaN   \n",
       "14                                           []   \n",
       "15                                          NaN   \n",
       "16                                          NaN   \n",
       "\n",
       "   yHAB159_kazachstania_solicola_160519_Seq_4608  \n",
       "0                                            NaN  \n",
       "1                                            NaN  \n",
       "2                                            NaN  \n",
       "3                                     [Seq_1405]  \n",
       "4                                            NaN  \n",
       "5                                            NaN  \n",
       "6                                     [Seq_4853]  \n",
       "7                                     [Seq_5631]  \n",
       "8                                     [Seq_4608]  \n",
       "9                                             []  \n",
       "10                                    [Seq_4213]  \n",
       "11                                           NaN  \n",
       "12                                    [Seq_2789]  \n",
       "13                                           NaN  \n",
       "14                                            []  \n",
       "15                                           NaN  \n",
       "16                                           NaN  \n",
       "\n",
       "[17 rows x 70 columns]"
      ]
     },
     "execution_count": 297,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synteny_window_new_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [],
   "source": [
    "synteny_window_new_cols.to_csv(io_library.data_processing_dir + os.path.normpath('example_synteny/'+goi_pair + '.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S.cerevisiae A</th>\n",
       "      <th>S.cerevisiae B</th>\n",
       "      <th>saccharomyces_arboricola_Seq_2087</th>\n",
       "      <th>saccharomyces_arboricola_Seq_2204</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR032C-A</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR033W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YCR073W-A</td>\n",
       "      <td>YNR034W</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>YCR075C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>YCR075W-A</td>\n",
       "      <td>YNR034W-A</td>\n",
       "      <td>Seq_2087</td>\n",
       "      <td>Seq_2204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>YCR076C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YCR077C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR035C</td>\n",
       "      <td>Seq_3527</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR036C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YCR079W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR037C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR038W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   S.cerevisiae A S.cerevisiae B saccharomyces_arboricola_Seq_2087  \\\n",
       "0             ---      YNR032C-A                               ---   \n",
       "1             ---        YNR033W                               ---   \n",
       "2             ---            ---                               NaN   \n",
       "3       YCR073W-A        YNR034W                               ---   \n",
       "4         YCR075C            ---                               ---   \n",
       "5             ---            ---                               NaN   \n",
       "6             ---            ---                               ---   \n",
       "7             ---            ---                               NaN   \n",
       "8       YCR075W-A      YNR034W-A                          Seq_2087   \n",
       "9         YCR076C            ---                               ---   \n",
       "10        YCR077C            ---                               ---   \n",
       "11            ---        YNR035C                          Seq_3527   \n",
       "12            ---            ---                               NaN   \n",
       "13            ---        YNR036C                               ---   \n",
       "14        YCR079W            ---                               ---   \n",
       "15            ---        YNR037C                               ---   \n",
       "16            ---        YNR038W                               ---   \n",
       "\n",
       "   saccharomyces_arboricola_Seq_2204  \n",
       "0                                ---  \n",
       "1                                ---  \n",
       "2                                NaN  \n",
       "3                           Seq_3505  \n",
       "4                           Seq_3359  \n",
       "5                                NaN  \n",
       "6                                ---  \n",
       "7                                NaN  \n",
       "8                           Seq_2204  \n",
       "9                                ---  \n",
       "10                           Seq_921  \n",
       "11                               ---  \n",
       "12                               NaN  \n",
       "13                               ---  \n",
       "14                          Seq_3876  \n",
       "15                               ---  \n",
       "16                               ---  "
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synteny_window_new_cols.loc[:,['S.cerevisiae A', 'S.cerevisiae B','saccharomyces_arboricola_Seq_2087','saccharomyces_arboricola_Seq_2204']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Naumovozyma_castellii\n"
     ]
    }
   ],
   "source": [
    "##Guts of for loop\n",
    "\n",
    "spec_fig2 = 'Naumovozyma_castellii'\n",
    "print(spec_fig2)\n",
    "focus_spec = genome_name_lookup[spec_fig2]\n",
    "#focus_spec = 'saccharomyces_arboricola'\n",
    "\n",
    "#'saccharomyces_uvarum',\n",
    "#  'saccharomyces_eubayanus',\n",
    "#  'saccharomyces_kudriavzevii',\n",
    "#  'saccharomyces_arboricola'\n",
    "\n",
    "#Get focus species genes that are present in focus ogs\n",
    "spec_og_id_lookup = dict(zip( y1000_species_subset['original_genome_id'], y1000_species_subset['spec_og_id']))\n",
    "spec_og_id = spec_og_id_lookup[focus_spec]\n",
    "focus_ogs = anc_genes_ogs_flat[focus_anc]\n",
    "\n",
    "og_genes = set()\n",
    "for og in focus_ogs: \n",
    "    og_genes = og_genes | og_genes_lookup[og]\n",
    "\n",
    "focus_spec_genes = []\n",
    "for og_gene in list(og_genes): \n",
    "    spec_og_id_test = int(og_gene.split('_')[0])\n",
    "    if spec_og_id == spec_og_id_test:\n",
    "        focus_spec_genes.append(og_gene)\n",
    "\n",
    "#Make full_gene lookup for focus spec\n",
    "focus_spec_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/' + focus_spec + '.csv')\n",
    "focus_spec_lookup = pd.read_csv(focus_spec_lookup_fname, index_col=0)\n",
    "y1000_id_gene_full_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup['gene_full']))        \n",
    "gene_full_y1000_id_lookup = dict(zip(focus_spec_lookup['gene_full'],focus_spec_lookup['y1000_id']))    \n",
    "y1000_id_gene_id_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup.index))    \n",
    "\n",
    "# Load GTF for given sequence: \n",
    "gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "db_fname = gtf_dir + 'gffutils_dbs/' + focus_spec + '.db'\n",
    "\n",
    "gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "\n",
    "# Make a dictionary of scaffold sizes\n",
    "genome_assembly_fname = y1000plus_dir + os.path.normpath('0_332yeast_genomes/332_genome_assemblies/' + focus_spec + '.fas')\n",
    "\n",
    "scaffold_sizes = {}\n",
    "for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "    scaffold_sizes[seq.id] = len(seq.seq)\n",
    "\n",
    "\n",
    "#For each of these genes, find the surrounding genes\n",
    "for og_gene in focus_spec_genes: \n",
    "    #og_gene = '109_2086'\n",
    "    focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "    gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "    cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "    all_features = cursor.fetchall()\n",
    "    if len(all_features) == 0:\n",
    "        print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "    CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "    if len(CDS_list)==1: \n",
    "        start = CDS_list[0]['start']\n",
    "        end = CDS_list[0]['end']\n",
    "    elif len(CDS_list)>1: \n",
    "        print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "        start = min([CDS['start'] for CDS in CDS_list])\n",
    "        end = max([CDS['end'] for CDS in CDS_list])\n",
    "    else: \n",
    "        raise AssertionError('No CDS found' + og_gene)\n",
    "\n",
    "    CDS = CDS_list[0]\n",
    "    strand = CDS['strand']\n",
    "    scaffold = CDS['seqid']\n",
    "\n",
    "    # if strand=='+': \n",
    "    #     lower = start\n",
    "    #     higher = end\n",
    "    # elif strand == '-': \n",
    "    #     lower = end\n",
    "    #     higher = start\n",
    "\n",
    "    low = max(0,start-bp_range/2)\n",
    "    high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "    surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "    #surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "    surrounding_CDS_data = []\n",
    "    for CDS_surr in surrounding_CDS:\n",
    "        surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "    #Find OG and line that each gene is in\n",
    "    anc_focus_spec_genes = OrderedDict([(anc,'---') for anc in anc_genes_ogs_flat.keys()])\n",
    "    for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "        y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "        for anc,og_list in anc_genes_ogs_flat.items():\n",
    "            for og in og_list: \n",
    "                og_genes = og_genes_lookup[og]\n",
    "                if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                    anc_focus_spec_genes[anc] = y1000_id_gene_id_lookup[y1000_id]\n",
    "\n",
    "    #Merge into synteny window dataframe\n",
    "\n",
    "    focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>naumovozyma_castellii_Seq_246</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Anc_6.343</th>\n",
       "      <td>[Seq_4618]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.344</th>\n",
       "      <td>[Seq_4183]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.345</th>\n",
       "      <td>[Seq_992]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.346</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.347</th>\n",
       "      <td>[Seq_4790, Seq_4790]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.348</th>\n",
       "      <td>[Seq_246, Seq_527]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.349</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.350</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.351</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.352</th>\n",
       "      <td>[Seq_3737, Seq_3737]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.353</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.354</th>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anc_6.355</th>\n",
       "      <td>[Seq_1834]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          naumovozyma_castellii_Seq_246\n",
       "Anc_6.343                    [Seq_4618]\n",
       "Anc_6.344                    [Seq_4183]\n",
       "Anc_6.345                     [Seq_992]\n",
       "Anc_6.346                            []\n",
       "Anc_6.347          [Seq_4790, Seq_4790]\n",
       "Anc_6.348            [Seq_246, Seq_527]\n",
       "Anc_6.349                            []\n",
       "Anc_6.350                            []\n",
       "Anc_6.351                            []\n",
       "Anc_6.352          [Seq_3737, Seq_3737]\n",
       "Anc_6.353                            []\n",
       "Anc_6.354                            []\n",
       "Anc_6.355                    [Seq_1834]"
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "og_gene = '100_245'#'109_2086'\n",
    "focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "all_features = cursor.fetchall()\n",
    "if len(all_features) == 0:\n",
    "    print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "if len(CDS_list)==1: \n",
    "    start = CDS_list[0]['start']\n",
    "    end = CDS_list[0]['end']\n",
    "elif len(CDS_list)>1: \n",
    "    print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "    start = min([CDS['start'] for CDS in CDS_list])\n",
    "    end = max([CDS['end'] for CDS in CDS_list])\n",
    "else: \n",
    "    raise AssertionError('No CDS found' + og_gene)\n",
    "\n",
    "CDS = CDS_list[0]\n",
    "strand = CDS['strand']\n",
    "scaffold = CDS['seqid']\n",
    "\n",
    "# if strand=='+': \n",
    "#     lower = start\n",
    "#     higher = end\n",
    "# elif strand == '-': \n",
    "#     lower = end\n",
    "#     higher = start\n",
    "\n",
    "low = max(0,start-bp_range/2)\n",
    "high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "#surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "surrounding_CDS_data = []\n",
    "for CDS_surr in surrounding_CDS:\n",
    "    surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "#Find OG and line that each gene is in\n",
    "anc_focus_spec_genes = OrderedDict([(anc,[]) for anc in anc_genes_ogs_flat.keys()])\n",
    "for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "    y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "    for anc,og_list in anc_genes_ogs_flat.items():\n",
    "        for og in og_list: \n",
    "            og_genes = og_genes_lookup[og]\n",
    "            if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                anc_focus_spec_genes[anc].append(y1000_id_gene_id_lookup[y1000_id])\n",
    "                #could make triple dashes if len(anc_focus_spec_genes[anc]) == 0:\n",
    "\n",
    "#Merge into synteny window dataframe\n",
    "\n",
    "focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "focus_gene_column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Add these in\n",
    "\n",
    "naumovozyma_castellii_Seq_2431 and naumovozyma_dairenensis_Seq_3886"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
