{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "home directory is unset\n",
      "y1000plus_tools dir is unset\n",
      "y1000plus data dir is unset\n",
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = othera\n",
      "home directory is C:\\Users\\BMH_work\\\n",
      "y1000plus_tools dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\\n",
      "y1000plus data dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\genomes\\y1000plus\\\n",
      "Importing y1000plus_tools.py\n",
      "importing yeast_esr_exp\n",
      "['C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\python37.zip', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\DLLs', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env', '', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\BMH_work\\\\.ipython', 'C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\BMH_work\\\\github\\\\yeast_esr_expression_analysis\\\\']\n",
      "sys.path : \n",
      "\n",
      "['C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\python37.zip', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\DLLs', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env', '', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\BMH_work\\\\.ipython', 'C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\BMH_work\\\\github\\\\yeast_esr_expression_analysis\\\\']\n",
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "#clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "clades_of_interest = {'Saccharomycetaceae', 'Saccharomycodaceae'}\n",
    "\n",
    "outgroups = [ 'Wickerhamomyces anomalus','Cyberlindnera jadinii', 'Ascoidea rubescens','Komagataella pastoris','Debaryomyces hansenii', 'Candida albicans', 'Yarrowia lipolytica']\n",
    "#here is a longer curated list of outgroups\n",
    "#nonsacc_visualization_subset = pd.read_csv(y1000plus_dir + 'species_visualization_subset.csv')\n",
    "#outgroup = 'Yarrowia lipolytica'\n",
    "\n",
    "y1000_species_subset = y1000_species[(y1000_species['Major clade'].isin(clades_of_interest))  |  (y1000_species['Species name'].isin(outgroups))]\n",
    "\n",
    "\n",
    "#outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "#y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n",
    "#y1000_species_subset\n",
    "\n",
    "genome_name_lookup = dict(zip(y1000_species_subset['species_names_fig2'], y1000_species_subset['original_genome_id']))\n",
    "\n",
    "\n",
    "\n",
    "#Load Ohnolog data and make ancestor look up table that links the common name of the high gene to the ancestor. \n",
    "ohnologs_expression_sorted = pd.read_csv(yeast_esr_exp_path + os.path.normpath(\"expression_data/kl_sc_PKA_as_m24_r1g1_20181017/20200607_ohnologs_expression_sorted.csv\"), index_col=0)\n",
    "anc_lookup = dict(zip(ohnologs_expression_sorted['SC_common_name_high'], ohnologs_expression_sorted.index))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nonsacc_visualization_subset = pd.read_csv(y1000plus_dir + 'species_visualization_subset.csv')\n",
    "# species_subset = ( set(nonsacc_visualization_subset['original_genome_id']) | \\\n",
    "#                    set(y1000_species[y1000_species['Major clade']=='Saccharomycetaceae']['original_genome_id']) ) # | \\\n",
    "#                    #set(y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'])\n",
    "#                  #)\n",
    "# y1000_species_subset = y1000_species[(y1000_species['original_genome_id'].isin(species_subset))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#For input goi pair, get surrounding genenames from YGOB site.  \n",
    "#Convention for this script is to put the high LFC gene name first \n",
    "\n",
    "goi_pair = 'EGO4_EGO2' #'GPM2_GPM3'  #GND2_GND1 #'EGO4_EGO2'\n",
    "focus_gene = goi_pair.split('_')[0]\n",
    "\n",
    "focus_anc = anc_lookup[focus_gene]\n",
    "\n",
    "#Extract files using url\n",
    "window = '8'\n",
    "\n",
    "load_saved_synteny_window = True\n",
    "\n",
    "if load_saved_synteny_window:\n",
    "    synteny_window = pd.read_csv(yeast_esr_exp_path + os.path.normpath('expression_data/example_synteny/'+goi_pair + '_YGOB.csv'), index_col = 0)\n",
    "\n",
    "else: \n",
    "    YGOB_query_url = 'http://ygob.ucd.ie/cgi/browser/tab.pl?ver=Latest&win=' + window + '&gene=' + focus_anc\n",
    "    YGOB_query_response = requests.get(YGOB_query_url)\n",
    "\n",
    "    soup = BeautifulSoup(YGOB_query_response.text)\n",
    "    table = soup.find(\"table\")\n",
    "\n",
    "    output_rows = []\n",
    "    for table_row in table.findAll('tr'):\n",
    "        columns = table_row.findAll('td')\n",
    "        output_row = []\n",
    "        for column in columns:\n",
    "            output_row.append(column.text)\n",
    "        output_rows.append(output_row)\n",
    "\n",
    "    synteny_window = pd.DataFrame(output_rows[2:],columns = output_rows[1])\n",
    "\n",
    "    synteny_window.to_csv(yeast_esr_exp_path + os.path.normpath('expression_data/example_synteny/'+goi_pair + '_YGOB.csv'))\n",
    "\n",
    "\n",
    "#synteny_row = synteny_window[synteny_window['Ancestor']==anc]\n",
    "# print(\"\\t\".join(synteny_row.iloc[0,:].values) + '\\n')\n",
    "# f.write(\"\\t\".join(synteny_row.iloc[0,:].values) + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make K.lactis map from Kl genename to ygob name\n",
    "spec = 'Klac'\n",
    "orth_dir = os.path.normpath(yeast_esr_exp.data_processing_dir + 'ortholog_files_y1000'  ) + os.sep\n",
    "\n",
    "klac_YGOB_to_y1000 = yeast_esr_exp.read_orth_lookup_table(spec + '_YGOB', spec + '_y1000', orth_dir)\n",
    "\n",
    "#Make og lookups for S.cer genes\n",
    "klac_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/kluyveromyces_lactis.csv')\n",
    "klac_lookup = pd.read_csv(klac_lookup_fname, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No sequence match for KLLA0C08437g\n"
     ]
    }
   ],
   "source": [
    "kl_y1000ogid_lookup = {}\n",
    "for kl_genename in list(set(synteny_window['K.lactis'])-{'---'}):\n",
    "    kl_genename_y1000_options = klac_YGOB_to_y1000[kl_genename]\n",
    "    if len(kl_genename_y1000_options)==1:\n",
    "        if kl_genename_y1000_options[0] != 'NONE':\n",
    "            y1000_no = kl_genename_y1000_options[0]\n",
    "            kl_y1000ogid_lookup[kl_genename] = klac_lookup.loc[y1000_no,'y1000_id']\n",
    "        else: \n",
    "            print('No sequence match for ' + kl_genename)\n",
    "            kl_y1000ogid_lookup[kl_genename] = 'NONE'\n",
    "    else: \n",
    "        kl_y1000ogid_lookup[kl_genename] = 'AMBIGUOUS'\n",
    "        print('More than one sequence match for ' + kl_genename)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "more than one goi in same orthogroup :OG2262 found_genes: \n",
      "['50_1817', '110_1608']\n",
      "more than one goi in same orthogroup :OG2411 found_genes: \n",
      "['110_3243', '110_1459', '50_1567']\n",
      "more than one goi in same orthogroup :OG2918 found_genes: \n",
      "['110_2274', '50_4899']\n",
      "more than one goi in same orthogroup :OG3003 found_genes: \n",
      "['50_739', '110_1780']\n",
      "more than one goi in same orthogroup :OG3210 found_genes: \n",
      "['50_4534', '110_4843']\n",
      "more than one goi in same orthogroup :OG3245 found_genes: \n",
      "['50_1876', '110_3759']\n",
      "more than one goi in same orthogroup :OG3448 found_genes: \n",
      "['110_241', '50_4803']\n",
      "more than one goi in same orthogroup :OG3502 found_genes: \n",
      "['110_2258', '50_532']\n",
      "more than one goi in same orthogroup :OG3534 found_genes: \n",
      "['50_80', '110_815']\n",
      "more than one goi in same orthogroup :OG4390 found_genes: \n",
      "['50_2219', '110_966']\n",
      "more than one goi in same orthogroup :OG6958 found_genes: \n",
      "['50_4079', '110_700']\n"
     ]
    }
   ],
   "source": [
    "gois_sc = list((set(synteny_window['S.cerevisiae A']) | set(synteny_window['S.cerevisiae B']))-{'---'})\n",
    "\n",
    "#Make og lookups for S.cer genes\n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "#target_y1000_id = scer_lookup.loc[gois_sc,'y1000_id']\n",
    "y1000_ids = dict(scer_lookup.loc[gois_sc,'y1000_id'])\n",
    "#y1000_id_list = [y1000_ids[goi] for goi in gois]\n",
    "\n",
    "gois = list((set(kl_y1000ogid_lookup.values()) | set(y1000_ids.values())) - {'NONE'})\n",
    "\n",
    "#Add an extra gene's OG - this OG should be included but wasn't in the same OG as S. cerevisiae and K. lactis.  \n",
    "extra_spec_gene = ('naumovozyma_castellii', 'Seq_2431')\n",
    "extra_spec_ids = pd.read_csv(y1000plus_dir + os.path.normpath('id_lookups/' + extra_spec_gene[0] + '.csv'))\n",
    "extra_spec_y1000_lookup = dict(zip(extra_spec_ids['gene_id'], extra_spec_ids['y1000_id']))\n",
    "extra_id = extra_spec_y1000_lookup[extra_spec_gene[1]]\n",
    "\n",
    "gois.append(extra_id)\n",
    "\n",
    "\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(gois, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "OrderedDict([('Anc_6.343', ['OG3448']),\n",
       "             ('Anc_6.344', ['OG2262']),\n",
       "             ('Anc_6.345', ['OG2411']),\n",
       "             ('Anc_6.346', ['OG4390']),\n",
       "             ('Anc_6.347', ['OG6695']),\n",
       "             ('Anc_6.348', ['OG6958', 'OG10444', 'OG38595']),\n",
       "             ('Anc_6.349', ['OG6956', 'OG22604']),\n",
       "             ('Anc_6.350', ['OG2918']),\n",
       "             ('Anc_6.351', ['OG3502']),\n",
       "             ('Anc_6.352', ['OG3210']),\n",
       "             ('Anc_6.353', ['OG3245']),\n",
       "             ('Anc_6.354', ['OG3534']),\n",
       "             ('Anc_6.355', ['OG3003'])])"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spec_og_search = ['S.cerevisiae A', 'S.cerevisiae B', 'K.lactis']\n",
    "\n",
    "\n",
    "#Orthogroups found from S. cerevisiae and K. lactis orthologs of each gene may not be complete - this includes extra orthogroups. \n",
    "#It would be better to search for orthogroups based on genes from each species present in YGOB. Also need to have og present\n",
    "#in og lookup so need to add the gene as a goi above.  \n",
    "\n",
    "additional_ogs = {'EGO4_EGO2': [('Anc_6.348','OG38595')]  #found that EGO4_EGO2 had a small orthogroup containing naumovozyma_castellii_Seq_2431 and naumovozyma_dairenensis_Seq_3886\n",
    "                 }\n",
    "\n",
    "anc_genes = []\n",
    "for anc in synteny_window['Ancestor']: \n",
    "    if anc != '---':\n",
    "        anc_genes.append((anc, dict(synteny_window.loc[synteny_window['Ancestor']==anc, spec_og_search].iloc[0,:])))\n",
    "# For Ancestor Column, Make list of tuple: species, genename.  \n",
    "\n",
    "anc_genes_ogs = OrderedDict()\n",
    "anc_genes_ogs_flat = OrderedDict()\n",
    "for anc, gene_dict in anc_genes: \n",
    "    anc_og_dict = {}\n",
    "    for spec in spec_og_search: \n",
    "        gene = gene_dict[spec]\n",
    "        \n",
    "        og = ''\n",
    "        \n",
    "        if gene !='---': \n",
    "            if 'S.cerevisiae' in spec: \n",
    "                y1000_id = y1000_ids[gene]\n",
    "                og = goi_og_lookup[y1000_id]\n",
    "            elif spec == 'K.lactis':\n",
    "                y1000_id = kl_y1000ogid_lookup[gene]\n",
    "                if not(y1000_id in {'NONE', 'AMBIGUOUS'}):\n",
    "                    og = goi_og_lookup[y1000_id]\n",
    "        anc_og_dict[spec] = og\n",
    "    anc_genes_ogs[anc]  = anc_og_dict\n",
    "    anc_genes_ogs_flat[anc] = list((set(anc_og_dict.values()) - {''}))\n",
    "\n",
    "\n",
    "#Add additional ogs. \n",
    "if goi_pair in additional_ogs.keys():\n",
    "    for anc, additional_og in additional_ogs[goi_pair]:\n",
    "        anc_genes_ogs_flat[anc].append(additional_og)\n",
    "    \n",
    "anc_genes_ogs_flat\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Yueomyces_sinensis\n",
      "Tetrapisispora_blattae\n",
      "Vanderwaltozyma_polyspora\n",
      "Tetrapisispora_iriomotensis\n",
      "Tetrapisispora_phaffii\n",
      "Tetrapisispora_fleetii\n",
      "Tetrapisispora_namnaonensis\n",
      "Candida_castellii\n",
      "Nakaseomyces_bacillisporus\n",
      "Candida_glabrata\n",
      "More than one CDS - introns?.  N of CDS: 2\n",
      "Candida_bracarensis\n",
      "Nakaseomyces_delphensis\n",
      "Candida_nivariensis\n",
      "Saccharomyces_eubayanus\n",
      "Saccharomyces_uvarum\n",
      "Saccharomyces_arboricola\n",
      "Saccharomyces_kudriavzevii\n",
      "Saccharomyces_mikatae\n",
      "Saccharomyces_paradoxus\n",
      "Naumovozyma_castellii\n",
      "Naumovozyma_dairenensis\n",
      "Kazachstania_viticola\n",
      "More than one CDS - introns?.  N of CDS: 2\n",
      "Kazachstania_africana\n",
      "Kazachstania_rosinii\n",
      "Kazachstania_spencerorum\n",
      "Kazachstania_kunashirensis\n",
      "Kazachstania_turicensis\n",
      "Kazachstania_martiniae\n",
      "Kazachstania_intestinalis\n",
      "Kazachstania_bromeliacearum\n",
      "Kazachstania_naganishii\n",
      "Kazachstania_taianensis\n",
      "Kazachstania_yakushimaensis\n",
      "Kazachstania_transvaalensis\n",
      "Kazachstania_siamensis\n",
      "Kazachstania_unispora\n",
      "Kazachstania_aerobia\n",
      "Kazachstania_solicola\n"
     ]
    }
   ],
   "source": [
    "#For a given example pair, find synteny of genes surrounding the ortholog in y1000 species not contained in YGOB.  \n",
    "\n",
    "\n",
    "#Make list of post-WGH species\n",
    "#set up data matrix based on time-calibrated species tree, and take subset of species based on species subset\n",
    "tree_fname = y1000plus_dir + '332_2408OGs_time-calibrated_phylogeny_species-names_updated.newick'\n",
    "t = Tree(tree_fname)\n",
    "t.ladderize()\n",
    "\n",
    "post_wgh_specs = t.get_common_ancestor(['Vanderwaltozyma_polyspora', 'Saccharomyces_cerevisiae']).get_leaves()\n",
    "post_wgh_specs_names = [node.name for node in post_wgh_specs]\n",
    "post_wgh_specs_names.remove('Saccharomyces_cerevisiae')\n",
    "#For each species, load gtf, get surrounding genes, compare surrounding genes with orthologs (from ogs) of surrounding genes\n",
    "bp_range = 20000  #Base pair range - goal is to get at least 8 CDS features\n",
    "\n",
    "synteny_window_new_cols = synteny_window\n",
    "\n",
    "for spec_fig2 in post_wgh_specs_names: \n",
    "    #spec_fig2 = 'Saccharomyces_arboricola'\n",
    "    print(spec_fig2)\n",
    "    focus_spec = genome_name_lookup[spec_fig2]\n",
    "    synteny_window_new_cols = y1000plus_tools.find_spec_synteny(synteny_window_new_cols, focus_spec, focus_anc, genome_name_lookup, y1000_species_subset, anc_genes_ogs_flat, og_genes_lookup, bp_range)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "synteny_window_new_cols.to_csv(yeast_esr_exp_path + os.path.normpath('expression_data/example_synteny/'+goi_pair + '.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S.cerevisiae A</th>\n",
       "      <th>S.cerevisiae B</th>\n",
       "      <th>saccharomyces_arboricola_Seq_2087</th>\n",
       "      <th>saccharomyces_arboricola_Seq_2204</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR032C-A</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR033W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YCR073W-A</td>\n",
       "      <td>YNR034W</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>YCR075C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>YCR075W-A</td>\n",
       "      <td>YNR034W-A</td>\n",
       "      <td>Seq_2087</td>\n",
       "      <td>Seq_2204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>YCR076C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>YCR077C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR035C</td>\n",
       "      <td>Seq_3527</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR036C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>YCR079W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>Seq_3876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR037C</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>---</td>\n",
       "      <td>YNR038W</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   S.cerevisiae A S.cerevisiae B saccharomyces_arboricola_Seq_2087  \\\n",
       "0             ---      YNR032C-A                               ---   \n",
       "1             ---        YNR033W                               ---   \n",
       "2             ---            ---                               NaN   \n",
       "3       YCR073W-A        YNR034W                               ---   \n",
       "4         YCR075C            ---                               ---   \n",
       "5             ---            ---                               NaN   \n",
       "6             ---            ---                               ---   \n",
       "7             ---            ---                               NaN   \n",
       "8       YCR075W-A      YNR034W-A                          Seq_2087   \n",
       "9         YCR076C            ---                               ---   \n",
       "10        YCR077C            ---                               ---   \n",
       "11            ---        YNR035C                          Seq_3527   \n",
       "12            ---            ---                               NaN   \n",
       "13            ---        YNR036C                               ---   \n",
       "14        YCR079W            ---                               ---   \n",
       "15            ---        YNR037C                               ---   \n",
       "16            ---        YNR038W                               ---   \n",
       "\n",
       "   saccharomyces_arboricola_Seq_2204  \n",
       "0                                ---  \n",
       "1                                ---  \n",
       "2                                NaN  \n",
       "3                           Seq_3505  \n",
       "4                           Seq_3359  \n",
       "5                                NaN  \n",
       "6                                ---  \n",
       "7                                NaN  \n",
       "8                           Seq_2204  \n",
       "9                                ---  \n",
       "10                           Seq_921  \n",
       "11                               ---  \n",
       "12                               NaN  \n",
       "13                               ---  \n",
       "14                          Seq_3876  \n",
       "15                               ---  \n",
       "16                               ---  "
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synteny_window_new_cols.loc[:,['S.cerevisiae A', 'S.cerevisiae B','saccharomyces_arboricola_Seq_2087','saccharomyces_arboricola_Seq_2204']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Yueomyces_sinensis\n",
      "Tetrapisispora_blattae\n",
      "Vanderwaltozyma_polyspora\n",
      "Tetrapisispora_iriomotensis\n",
      "Tetrapisispora_phaffii\n",
      "Tetrapisispora_fleetii\n",
      "Tetrapisispora_namnaonensis\n",
      "Candida_castellii\n",
      "Nakaseomyces_bacillisporus\n",
      "Candida_glabrata\n",
      "Candida_bracarensis\n",
      "Nakaseomyces_delphensis\n",
      "Candida_nivariensis\n",
      "Saccharomyces_eubayanus\n",
      "Saccharomyces_uvarum\n",
      "More than one CDS - introns?.  N of CDS: 8\n",
      "Saccharomyces_arboricola\n",
      "Saccharomyces_kudriavzevii\n",
      "Saccharomyces_mikatae\n",
      "More than one CDS - introns?.  N of CDS: 10\n",
      "Saccharomyces_paradoxus\n",
      "Naumovozyma_castellii\n",
      "Naumovozyma_dairenensis\n",
      "Kazachstania_viticola\n",
      "Kazachstania_africana\n",
      "More than one CDS - introns?.  N of CDS: 5\n",
      "Kazachstania_rosinii\n",
      "Kazachstania_spencerorum\n",
      "Kazachstania_kunashirensis\n",
      "Kazachstania_turicensis\n",
      "Kazachstania_martiniae\n",
      "Kazachstania_intestinalis\n",
      "Kazachstania_bromeliacearum\n",
      "Kazachstania_naganishii\n",
      "Kazachstania_taianensis\n",
      "Kazachstania_yakushimaensis\n",
      "Kazachstania_transvaalensis\n",
      "Kazachstania_siamensis\n",
      "Kazachstania_unispora\n",
      "Kazachstania_aerobia\n",
      "Kazachstania_solicola\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#Replaced with functions\n",
    "\n",
    "#Make list of post-WGH species\n",
    "#set up data matrix based on time-calibrated species tree, and take subset of species based on species subset\n",
    "tree_fname = y1000plus_dir + '332_2408OGs_time-calibrated_phylogeny_species-names_updated.newick'\n",
    "t = Tree(tree_fname)\n",
    "t.ladderize()\n",
    "\n",
    "post_wgh_specs = t.get_common_ancestor(['Vanderwaltozyma_polyspora', 'Saccharomyces_cerevisiae']).get_leaves()\n",
    "post_wgh_specs_names = [node.name for node in post_wgh_specs]\n",
    "post_wgh_specs_names.remove('Saccharomyces_cerevisiae')\n",
    "#For each species, load gtf, get surrounding genes, compare surrounding genes with orthologs (from ogs) of surrounding genes\n",
    "bp_range = 20000  #Base pair range - goal is to get at least 8 CDS features\n",
    "\n",
    "\n",
    "synteny_window_new_cols = synteny_window\n",
    "\n",
    "for spec_fig2 in post_wgh_specs_names: \n",
    "    print(spec_fig2)\n",
    "    focus_spec = genome_name_lookup[spec_fig2]\n",
    "    #focus_spec = 'saccharomyces_arboricola'\n",
    "    \n",
    "    #'saccharomyces_uvarum',\n",
    "    #  'saccharomyces_eubayanus',\n",
    "    #  'saccharomyces_kudriavzevii',\n",
    "    #  'saccharomyces_arboricola'\n",
    "\n",
    "    #Get focus species genes that are present in focus ogs\n",
    "    spec_og_id_lookup = dict(zip( y1000_species_subset['original_genome_id'], y1000_species_subset['spec_og_id']))\n",
    "    spec_og_id = spec_og_id_lookup[focus_spec]\n",
    "    focus_ogs = anc_genes_ogs_flat[focus_anc]\n",
    "\n",
    "    og_genes = set()\n",
    "    for og in focus_ogs: \n",
    "        og_genes = og_genes | og_genes_lookup[og]\n",
    "\n",
    "    focus_spec_genes = []\n",
    "    for og_gene in list(og_genes): \n",
    "        spec_og_id_test = int(og_gene.split('_')[0])\n",
    "        if spec_og_id == spec_og_id_test:\n",
    "            focus_spec_genes.append(og_gene)\n",
    "\n",
    "    #Make full_gene lookup for focus spec\n",
    "    focus_spec_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/' + focus_spec + '.csv')\n",
    "    focus_spec_lookup = pd.read_csv(focus_spec_lookup_fname, index_col=0)\n",
    "    y1000_id_gene_full_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup['gene_full']))        \n",
    "    gene_full_y1000_id_lookup = dict(zip(focus_spec_lookup['gene_full'],focus_spec_lookup['y1000_id']))    \n",
    "    y1000_id_gene_id_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup.index))    \n",
    "\n",
    "    # Load GTF for given sequence: \n",
    "    gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "    db_fname = gtf_dir + 'gffutils_dbs/' + focus_spec + '.db'\n",
    "\n",
    "    gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "\n",
    "    # Make a dictionary of scaffold sizes\n",
    "    genome_assembly_fname = y1000plus_dir + os.path.normpath('0_332yeast_genomes/332_genome_assemblies/' + focus_spec + '.fas')\n",
    "\n",
    "    scaffold_sizes = {}\n",
    "    for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "        scaffold_sizes[seq.id] = len(seq.seq)\n",
    "\n",
    "\n",
    "    #For each of these genes, find the surrounding genes\n",
    "    for og_gene in focus_spec_genes: \n",
    "        #og_gene = '109_2086'\n",
    "        focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "        gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "        cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "        all_features = cursor.fetchall()\n",
    "        if len(all_features) == 0:\n",
    "            print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "        CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "        if len(CDS_list)==1: \n",
    "            start = CDS_list[0]['start']\n",
    "            end = CDS_list[0]['end']\n",
    "        elif len(CDS_list)>1: \n",
    "            print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "            start = min([CDS['start'] for CDS in CDS_list])\n",
    "            end = max([CDS['end'] for CDS in CDS_list])\n",
    "        else: \n",
    "            raise AssertionError('No CDS found' + og_gene)\n",
    "                  \n",
    "        CDS = CDS_list[0]\n",
    "        strand = CDS['strand']\n",
    "        scaffold = CDS['seqid']\n",
    "\n",
    "        # if strand=='+': \n",
    "        #     lower = start\n",
    "        #     higher = end\n",
    "        # elif strand == '-': \n",
    "        #     lower = end\n",
    "        #     higher = start\n",
    "\n",
    "        low = max(0,start-bp_range/2)\n",
    "        high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "        surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "        #surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "        surrounding_CDS_data = []\n",
    "        for CDS_surr in surrounding_CDS:\n",
    "            surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "        #Find OG and line that each gene is in\n",
    "        anc_focus_spec_genes = OrderedDict([(anc,[]) for anc in anc_genes_ogs_flat.keys()])\n",
    "        for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "            y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "            for anc,og_list in anc_genes_ogs_flat.items():\n",
    "                for og in og_list: \n",
    "                    og_genes = og_genes_lookup[og]\n",
    "                    if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                        anc_focus_spec_genes[anc].append(y1000_id_gene_id_lookup[y1000_id])\n",
    "                        #could make triple dashes if len(anc_focus_spec_genes[anc]) == 0:\n",
    "                #remove redundancies created by genes with introns\n",
    "                anc_focus_spec_genes[anc]=list(set(anc_focus_spec_genes[anc]))\n",
    "        #Merge into synteny window dataframe\n",
    "\n",
    "        focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "        synteny_window_new_cols = synteny_window_new_cols.merge(focus_gene_column, left_on='Ancestor', right_index = True, how='left')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
