{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#matplotlib notebook\n",
    "\n",
    "y1000_plus_dir_ubuntu = '/home/heineike/GitHub/y1000plus_tools/data/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/y1000_species_table.csv\"), index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "#clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "#clades_of_interest = {'Saccharomycetaceae', 'Saccharomycodaceae'}\n",
    "#outgroups = [ 'Wickerhamomyces anomalus','Cyberlindnera jadinii', 'Ascoidea rubescens','Komagataella pastoris','Debaryomyces hansenii', 'Candida albicans', 'Yarrowia lipolytica']\n",
    "#outgroup = 'Yarrowia lipolytica'\n",
    "\n",
    "#y1000_species_subset = y1000_species[(y1000_species['Major clade'].isin(clades_of_interest))  |  (y1000_species['Species name'].isin(outgroups))]\n",
    "\n",
    "\n",
    "#outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "#y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n",
    "\n",
    "y1000_species_subset = y1000_species[y1000_species['Major clade']!='outgroup']\n",
    "#y1000_species_subset = y1000_speciesâ€‹\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('y1000plus_tools_data/y1000plus/id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "\n",
    "#Met17: 'YLR303W', \n",
    "#Str2:  YJR130C\n",
    "#'YML082W', \n",
    "#Str3: 'YGL184C', \n",
    "#'YLL058W', \n",
    "#Cys3: 'YAL012W'\n",
    "#IRC7: 'YFR055W'\n",
    "\n",
    "\n",
    "target_y1000_ids = scer_lookup.loc[['YLR303W', 'YML082W','YJR130C', 'YFR055W', 'YHR112C', 'YGL184C', 'YLL058W', 'YAL012W'], 'y1000_id']\n",
    "\n",
    "y1000_id_list = list(target_y1000_ids)\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Only need to do once\n",
    "\n",
    "# for protein_fname in ['IRC7', 'YHR112C']:\n",
    "#     target_y1000_id = protein_fnames[protein_fname]\n",
    "#     og_genes = og_genes_lookup[goi_og_lookup[target_y1000_id]]\n",
    "#     y1000plus_tools.extract_protein_seqs(og_genes, protein_fname, y1000_species_subset)\n",
    "#     print(protein_fname + ' protein sequence extraction complete')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "goi = 'YHR112C'  #'YHR112C' #'STR3'  #YHR112C, IRC7\n",
    " \n",
    "protein_fname_base = y1000_plus_dir_ubuntu + 'y1000plus_tools_data/y1000plus/proteins_og/' + goi + '/' + goi\n",
    "\n",
    "#Command:   mafft v 7.487\n",
    "mafft_command = ['mafft', '--genafpair', '--maxiterate', '1000', \n",
    "                protein_fname_base + '.fasta', \n",
    "                '>',\n",
    "                protein_fname_base + '_aln.fasta']\n",
    "\n",
    "\n",
    "print(\" \".join(mafft_command))\n",
    "\n",
    "\n",
    "#Trimming Command: \n",
    "#trimal 1.2r59\n",
    "#Had to install g++ on the linux system to make the program.  \n",
    "trimal_command = ['/home/heineike/trimAl/source/trimal',\n",
    "                  '-in', protein_fname_base + '_aln.fasta', \n",
    "                  '-out', protein_fname_base + '_aln_trimmed.fasta', \n",
    "                  '-gappyout']\n",
    "\n",
    "print(\"\\nTrimal command\\n\")\n",
    "print(\" \".join(trimal_command))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Check to see which major clades / species are present in the orthogroup if \n",
    "#there is no good outgroup\n",
    "og_genes = og_genes_lookup[goi_og_lookup[protein_fnames[goi]]]\n",
    "specs_in_og = []\n",
    "for gene in og_genes: \n",
    "    specs_in_og.append(gene.split('_')[0])\n",
    "\n",
    "set(y1000_species_subset[y1000_species_subset['spec_og_id'].isin(set(specs_in_og))]['Major clade'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#choose outgroup species (important for making protein trees from multiple sequence alignments) \n",
    "outgroup = 'Lipomyces starkeyi'#'Yarrowia lipolytica' #'Hanseniaspora vineae'#'Yarrowia lipolytica' #'Lipomyces starkeyi' # 'Tortispora starmeri' #'Lipomyces starkeyi'   #Lipomyces starkeyi is a good outgroup for all the species\n",
    "\n",
    "outgroup_orig_genome = y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'].values[0]\n",
    "\n",
    "#Test to see if orthogroup has something from outgroup: \n",
    "\n",
    "y1000_id_outgroup = y1000_species_subset[y1000_species_subset['Species name']==outgroup]['spec_og_id'].values[0]\n",
    "\n",
    "outgroup_gene = ''\n",
    "for gene in og_genes:#og_genes: \n",
    "    y1000_spec_id = gene.split('_')[0]\n",
    "    if y1000_spec_id == str(y1000_id_outgroup):\n",
    "        outgroup_gene_id = gene\n",
    "        \n",
    "outgrp_orig_spec_name = y1000_species_subset[y1000_species_subset['Species name']==outgroup]['original_genome_id'].values[0]\n",
    "outgrp_lookup_fname = y1000plus_dir + os.path.normpath('y1000plus_tools_data/y1000plus/id_lookups/' + outgrp_orig_spec_name + '.csv')\n",
    "\n",
    "outgrp_lookup = pd.read_csv(outgrp_lookup_fname, index_col=0)\n",
    "y1000_species_subset[y1000_species_subset['Species name']==outgroup]['original_genome_id'].values[0]\n",
    "outgroup_gene = outgrp_orig_spec_name  + '_' + outgrp_lookup[outgrp_lookup['y1000_id']==outgroup_gene_id].index[0]\n",
    "    \n",
    "\n",
    "if outgroup_gene =='': \n",
    "    print(outgroup + ' has no proteins in og ' + ogs + ' for ' + goi + '. Pick another outgroup')\n",
    "else: \n",
    "    print('outgroup gene id = ' + outgroup_gene_id)\n",
    "    print('outgroup gene = ' + outgroup_gene)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#protein_fname_aln = protein_fname_base + '_aln.fasta'\n",
    "#prot_seqs = SeqIO.parse(protein_fname_aln, \"fasta\")  \n",
    "\n",
    "outgroup_id = outgroup_dict[goi]\n",
    "\n",
    "# #routine for figuring out outgroup_id if we don't use the outgroup_dict\n",
    "# if outgroup_gene == None: \n",
    "#     print('No outgroup gene designated - need to pick outgroup gene')\n",
    "\n",
    "# for seq in prot_seqs: \n",
    "#     y1000_id = seq.description.split(' ')[1].split('=')[1]\n",
    "#     if y1000_id == outgroup_gene:\n",
    "#         outgroup_id = seq.id\n",
    "\n",
    "#iqtree-1.6.12 linux 64 bit\n",
    "iqtree_command = [\"/home/heineike/iqtree-1.6.12-Linux/bin/iqtree\", \n",
    "                  \"-s\" , protein_fname_base + '_aln_trimmed.fasta',\n",
    "                  \"-m\", 'LG+R8',  #The first two were LG+R7 and LG+R8, so used LG+R8 to save time\n",
    "                  \"-nt\", \"AUTO\",  #automatically determines number of threads \n",
    "                  \"-o\", outgroup_id]\n",
    "print(\" \".join(iqtree_command))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#protein_fname_aln = protein_fname_base + '_aln.fasta'\n",
    "#prot_seqs = SeqIO.parse(protein_fname_aln, \"fasta\")  \n",
    "\n",
    "outgroup_id = outgroup_dict[goi]\n",
    "\n",
    "# #routine for figuring out outgroup_id if we don't use the outgroup_dict\n",
    "# if outgroup_gene == None: \n",
    "#     print('No outgroup gene designated - need to pick outgroup gene')\n",
    "\n",
    "# for seq in prot_seqs: \n",
    "#     y1000_id = seq.description.split(' ')[1].split('=')[1]\n",
    "#     if y1000_id == outgroup_gene:\n",
    "#         outgroup_id = seq.id\n",
    "\n",
    "#iqtree-1.6.12 linux 64 bit\n",
    "iqtree_command = [\"/home/heineike/iqtree-1.6.12-Linux/bin/iqtree\", \n",
    "                  \"-s\" , protein_fname_base + '_aln_trimmed.fasta',\n",
    "                  \"-m\", 'LG+R8',  #The first two were LG+R7 and LG+R8, so used LG+R8 to save time\n",
    "                  \"-nt\", \"AUTO\",  #automatically determines number of threads \n",
    "                  \"-o\", outgroup_id]\n",
    "print(\" \".join(iqtree_command))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
