{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New users will need to update the y1000plus_tools, y1000plus_dir, figsave_dir, and yeast_esr_exp_dir (if those libraries are being used)\n",
      "y1000plus_tools dir is C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\\n",
      "y1000plus data dir is C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\\n",
      "Added C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\ to path\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Use yeast_esr_expression functions? (Y/n) Y\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added C:\\Users\\heineib\\Documents\\GitHub\\yeast_esr_expression_analysis\\ to path\n",
      "Importing yeast_esr_exp.  If autoreload, may need to reset base_dir and data_processing dir \n",
      "  yeast_esr_exp.base_dir=base_dir \n",
      " yeast_esr_exp.data_processing_dir = data_processing_dir\n",
      "Importing y1000plus_tools.py\n",
      "y1000plus_tools dir is unset\n",
      "y1000plus data dir is unset\n",
      "importing yeast_esr_exp\n",
      "['C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\python37.zip', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\DLLs', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py', '', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\ete3-3.1.2-py3.7.egg', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\heineib\\\\.ipython', 'C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\yeast_esr_expression_analysis\\\\']\n",
      "sys.path : \n",
      "\n",
      "['C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\python37.zip', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\DLLs', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py', '', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\ete3-3.1.2-py3.7.egg', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\heineib\\\\Anaconda3\\\\envs\\\\sc_seq_py\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\heineib\\\\.ipython', 'C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\heineib\\\\Documents\\\\GitHub\\\\yeast_esr_expression_analysis\\\\']\n"
     ]
    }
   ],
   "source": [
    "%run std_libraries.py\n",
    "#from Bio.Align import MultipleSeqAlignment\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#matplotlib notebook\n",
    "\n",
    "y1000_plus_dir_ubuntu = '/home/heineike/GitHub/y1000plus_tools/data/'\n",
    "metproj_data_dir = os.path.normpath('C:/Users/heineib/Google Drive/Crick_LMS/projects/jason_met_auxotrophy') + os.sep\n",
    "\n",
    "aln_fname_base_win = y1000plus_dir + os.path.normpath('y1000plus_tools_data/y1000plus/proteins_og') + os.sep\n",
    "\n",
    "\n",
    "y1000_species = pd.read_csv(y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/y1000_species_table.csv\"), index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "#clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "#clades_of_interest = {'Saccharomycetaceae', 'Saccharomycodaceae'}\n",
    "#outgroups = [ 'Wickerhamomyces anomalus','Cyberlindnera jadinii', 'Ascoidea rubescens','Komagataella pastoris','Debaryomyces hansenii', 'Candida albicans', 'Yarrowia lipolytica']\n",
    "#outgroup = 'Yarrowia lipolytica'\n",
    "\n",
    "#y1000_species_subset = y1000_species[(y1000_species['Major clade'].isin(clades_of_interest))  |  (y1000_species['Species name'].isin(outgroups))]\n",
    "\n",
    "\n",
    "#outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "#y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n",
    "\n",
    "y1000_species_subset = y1000_species[y1000_species['Major clade']!='outgroup']\n",
    "#y1000_species_subset = y1000_species\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#For now just looking in the vicinity of YLL058W.  In the future would like to look in the vicinity of all of them. \n",
    "\n",
    "#From paper Jason Cited: \n",
    "#['AYT1', 'MHT1', 'MMP1', 'GTT2', 'YLL059C', 'YLL058W', 'YLL057C', 'YLL056C', 'YLL055W', 'YLL054C', 'YLL053C', 'AQY2', 'FRE6']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('y1000plus_tools_data/y1000plus/id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "scer_y1000_id_gene_id_lookup = dict(zip(scer_lookup['y1000_id'], scer_lookup.index))\n",
    "\n",
    "#Met17: 'YLR303W', \n",
    "#Str2:  YJR130C\n",
    "#'YML082W', \n",
    "#Str3: 'YGL184C', \n",
    "#'YLL058W', \n",
    "#Cys3: 'YAL012W'\n",
    "#IRC7: 'YFR055W'\n",
    "\n",
    "# sul_operon = ['YLL051C',\n",
    "# 'YLL052C',\n",
    "# 'YLL053C',\n",
    "# 'YLL054C',\n",
    "# 'YLL055W',\n",
    "# 'YLL056C',\n",
    "# 'YLL057C',\n",
    "# 'YLL058W',\n",
    "# 'YLL059C',\n",
    "# 'YLL060C',\n",
    "# 'YLL061W',\n",
    "# 'YLL062C',\n",
    "# 'YLL063C']\n",
    "\n",
    "# print('The following genes are missing from y1000plus index')\n",
    "# print(set(sul_operon)-set(scer_lookup.index))\n",
    "\n",
    "# target_y1000_ids = scer_lookup.loc[list(set(sul_operon) & set(scer_lookup.index)), 'y1000_id']\n",
    "\n",
    "# y1000_id_list = list(target_y1000_ids)\n",
    "# goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species_subset)\n",
    "\n",
    "# gene_id_gene_full_lookup = dict(zip(spec_lookup.index, spec_lookup['gene_full']))\n",
    "# gene_full_y1000_id_lookup = dict(zip(spec_lookup['gene_full'],spec_lookup['y1000_id']))    \n",
    "# gene_full_gene_id_lookup = dict(zip(spec_lookup['gene_full'],spec_lookup.index)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_eubayanus.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_uvarum.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034862_zygotorulaspora_florentina_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_kudriavzevii.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_arboricola.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_mikatae.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 8\n",
      "more than one goi in same orthogroup :OG2973 found_genes: \n",
      "['113_4472', '113_2871']\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\saccharomyces_paradoxus.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000026256_zygotorulaspora_mrakii_161215.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000035652_torulaspora_maleeae_160613.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000035651_torulaspora_microellipsoides_160519.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 10\n",
      "more than one goi in same orthogroup :OG1058 found_genes: \n",
      "['299_5096', '299_4188']\n",
      "more than one goi in same orthogroup :OG1075 found_genes: \n",
      "['299_4439', '299_1312']\n",
      "more than one goi in same orthogroup :OG1114 found_genes: \n",
      "['299_2222', '299_1781']\n",
      "more than one goi in same orthogroup :OG1219 found_genes: \n",
      "['299_4989', '299_5111']\n",
      "more than one goi in same orthogroup :OG1356 found_genes: \n",
      "['299_2978', '299_1668']\n",
      "more than one goi in same orthogroup :OG1390 found_genes: \n",
      "['299_2165', '299_17']\n",
      "more than one goi in same orthogroup :OG1444 found_genes: \n",
      "['299_2749', '299_3243', '299_4840', '299_367']\n",
      "more than one goi in same orthogroup :OG1530 found_genes: \n",
      "['299_1574', '299_293']\n",
      "more than one goi in same orthogroup :OG1812 found_genes: \n",
      "['299_540', '299_3104']\n",
      "more than one goi in same orthogroup :OG1863 found_genes: \n",
      "['299_1590', '299_2089']\n",
      "more than one goi in same orthogroup :OG1990 found_genes: \n",
      "['299_5180', '299_3450']\n",
      "more than one goi in same orthogroup :OG2006 found_genes: \n",
      "['299_1375', '299_1304']\n",
      "more than one goi in same orthogroup :OG2029 found_genes: \n",
      "['299_4530', '299_3410']\n",
      "more than one goi in same orthogroup :OG2273 found_genes: \n",
      "['299_4502', '299_1646']\n",
      "more than one goi in same orthogroup :OG2348 found_genes: \n",
      "['299_609', '299_4482']\n",
      "more than one goi in same orthogroup :OG2551 found_genes: \n",
      "['299_4255', '299_678']\n",
      "more than one goi in same orthogroup :OG2777 found_genes: \n",
      "['299_1436', '299_3307']\n",
      "more than one goi in same orthogroup :OG2871 found_genes: \n",
      "['299_980', '299_4041']\n",
      "more than one goi in same orthogroup :OG2872 found_genes: \n",
      "['299_2785', '299_3705']\n",
      "more than one goi in same orthogroup :OG3105 found_genes: \n",
      "['299_1155', '299_5265']\n",
      "more than one goi in same orthogroup :OG3136 found_genes: \n",
      "['299_1694', '299_1391']\n",
      "more than one goi in same orthogroup :OG3321 found_genes: \n",
      "['299_2050', '299_93']\n",
      "more than one goi in same orthogroup :OG3330 found_genes: \n",
      "['299_2587', '299_2197']\n",
      "more than one goi in same orthogroup :OG3727 found_genes: \n",
      "['299_2724', '299_1824']\n",
      "more than one goi in same orthogroup :OG3788 found_genes: \n",
      "['299_974', '299_4495']\n",
      "more than one goi in same orthogroup :OG3902 found_genes: \n",
      "['299_4694', '299_2928']\n",
      "more than one goi in same orthogroup :OG4799 found_genes: \n",
      "['299_1826', '299_3574']\n",
      "more than one goi in same orthogroup :OG4919 found_genes: \n",
      "['299_442', '299_3850']\n",
      "more than one goi in same orthogroup :OG5123 found_genes: \n",
      "['299_4278', '299_4289']\n",
      "more than one goi in same orthogroup :OG5190 found_genes: \n",
      "['299_2196', '299_202']\n",
      "more than one goi in same orthogroup :OG5479 found_genes: \n",
      "['299_2772', '299_2245']\n",
      "more than one goi in same orthogroup :OG6544 found_genes: \n",
      "['299_3048', '299_2908']\n",
      "more than one goi in same orthogroup :OG6822 found_genes: \n",
      "['299_599', '299_2011']\n",
      "more than one goi in same orthogroup :OG7816 found_genes: \n",
      "['299_1879', '299_3596']\n",
      "more than one goi in same orthogroup :OG8421 found_genes: \n",
      "['299_303', '299_1656']\n",
      "more than one goi in same orthogroup :OG15087 found_genes: \n",
      "['299_1612', '299_197']\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000035651_torulaspora_microellipsoides_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000035651_torulaspora_microellipsoides_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034881_torulaspora_pretoriensis_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000026152_torulaspora_franciscae_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\torulaspora_delbrueckii.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_cidri.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 8\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_fermentati.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 2\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034862_zygotorulaspora_florentina_160519.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 2\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_nothofagi.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\zygosaccharomyces_bailii.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\yHMPu5000034866_zygosaccharomyces_bisporus_160519.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_waltii.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_quebecensis.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_thermotolerans.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_lanzarotensis.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_fantastica.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_mirantina.db already exists\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_dasiensis.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 4\n",
      "C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs\\lachancea_fantastica.db already exists\n",
      "More than one CDS - introns?.  N of CDS: 11\n",
      "more than one goi in same orthogroup :OG61139 found_genes: \n",
      "['56_2391', '56_864']\n",
      "more than one goi in same orthogroup :OG61147 found_genes: \n",
      "['56_827', '56_4945']\n"
     ]
    }
   ],
   "source": [
    "#For HSU1 orthologs \n",
    "goi = 'STR2_YLL058W' # 'STR2_YLL058W' #'GPM2_GPM3'#'GPM2_GPM3' #\n",
    "#og = goi_pair_og_lookup[goi_pair]\n",
    "\n",
    "#fname_tree = os.path.normpath(y1000plus_dir + 'proteins_og/' + goi_pair + '_' + og + '/tree/run' + N_tree_dict[goi_pair][0] + '/' + goi_pair + '_' + og + '_aln_' + N_tree_dict[goi_pair][1] + 'trimmed.fasta.treefile')\n",
    "fname_tree = y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/proteins_og/\" + goi + '/'+ goi + '_aln_trimmed.fasta.treefile')\n",
    "t = Tree(fname_tree, format=1)  \n",
    "\n",
    "hsu1_mrca = t.get_common_ancestor(['saccharomyces_cerevisiae_YLL058W','lachancea_fantastica_Seq_4073'])\n",
    "hsu1_orths = hsu1_mrca.get_leaf_names()\n",
    "\n",
    "bp_range = 15000\n",
    "\n",
    "gene_synteny = {}\n",
    "\n",
    "for hsu1_orth in hsu1_orths: \n",
    "\n",
    "    #hsu1_orth = 'lachancea_thermotolerans_Seq_2696'# 'saccharomyces_eubayanus_Seq_1484'#'saccharomyces_mikatae_Seq_3298'#'zygosaccharomyces_bailii_Seq_2609' #'lachancea_thermotolerans_Seq_2696'#'saccharomyces_paradoxus_Seq_3252' #'saccharomyces_eubayanus_Seq_1484'\n",
    "\n",
    "    #Except for S. cer\n",
    "    if hsu1_orth != 'saccharomyces_cerevisiae_YLL058W':\n",
    "        \n",
    "        spec = '_'.join(hsu1_orth.split('_')[:-2])\n",
    "    \n",
    "        gene_id = '_'.join(hsu1_orth.split('_')[-2:])\n",
    "\n",
    "        #If necessary, make the gtf databases: \n",
    "\n",
    "        db_fname =  y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/gffutils_dbs/\" + spec + '.db')\n",
    "\n",
    "        if not(os.path.isfile(db_fname)):\n",
    "            y1000plus_tools.make_gtf_dbs([spec])\n",
    "            # db = gffutils.create_db(fn, dbfn=dbfn, force=True, keep_order=True, merge_strategy='merge', sort_attribute_values=True)\n",
    "        else: \n",
    "            print(db_fname + ' already exists')\n",
    "\n",
    "        #Load the genome \n",
    "\n",
    "        #Make full_gene lookup for focus spec\n",
    "        spec_lookup_fname = y1000plus_dir + os.path.normpath('y1000plus_tools_data/y1000plus/id_lookups/' + spec + '.csv')\n",
    "        spec_lookup = pd.read_csv(spec_lookup_fname, index_col=0)\n",
    "\n",
    "        #y1000_id_gene_full_lookup = dict(zip(spec_lookup['y1000_id'],spec_lookup['gene_full']))        \n",
    "        #y1000_id_gene_id_lookup = dict(zip(spec_lookup['y1000_id'],spec_lookup.index))    \n",
    "        gene_id_gene_full_lookup = dict(zip(spec_lookup.index, spec_lookup['gene_full']))\n",
    "        gene_full_y1000_id_lookup = dict(zip(spec_lookup['gene_full'],spec_lookup['y1000_id']))    \n",
    "        gene_full_gene_id_lookup = dict(zip(spec_lookup['gene_full'],spec_lookup.index))    \n",
    "\n",
    "\n",
    "        # Load GTF for given sequence: \n",
    "        gtf_dir = y1000plus_dir + os.path.normpath(\"shen_2018_data/0_332yeast_genomes/332_genome_annotations/gtf\") + os.sep\n",
    "        db_fname =  y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/gffutils_dbs/\" + spec + '.db')\n",
    "\n",
    "        gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "        # Make a dictionary of scaffold sizes\n",
    "        genome_assembly_fname = y1000plus_dir + os.path.normpath('shen_2018_data/0_332yeast_genomes/332_genome_assemblies/' + spec + '.fas')\n",
    "\n",
    "        scaffold_sizes = {}\n",
    "        for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "            scaffold_sizes[seq.id] = len(seq.seq)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "        #Find surrounding genes (what range)\n",
    "\n",
    "        #focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "        gene_full = gene_id_gene_full_lookup[gene_id]\n",
    "\n",
    "        cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "        all_features = cursor.fetchall()\n",
    "        if len(all_features) == 0:\n",
    "            print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "        CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "        if len(CDS_list)==1: \n",
    "            start = CDS_list[0]['start']\n",
    "            end = CDS_list[0]['end']\n",
    "        elif len(CDS_list)>1: \n",
    "            print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "            start = min([CDS['start'] for CDS in CDS_list])\n",
    "            end = max([CDS['end'] for CDS in CDS_list])\n",
    "        else: \n",
    "            raise AssertionError('No CDS found' + og_gene)\n",
    "\n",
    "        CDS = CDS_list[0]\n",
    "        strand = CDS['strand']\n",
    "        scaffold = CDS['seqid']\n",
    "\n",
    "        # if strand=='+': \n",
    "        #     lower = start\n",
    "        #     higher = end\n",
    "        # elif strand == '-': \n",
    "        #     lower = end\n",
    "        #     higher = start\n",
    "\n",
    "        low = max(0,start-bp_range/2)\n",
    "        high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "        surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "        #surrounding_CDS info: (gene_full, CDS_id, strand, start, end)\n",
    "        surrounding_CDS_data = []\n",
    "        for CDS_surr in surrounding_CDS:\n",
    "            gene_full_surr = CDS_surr.attributes['gene_id'][0]\n",
    "            surrounding_CDS_data.append((gene_full_gene_id_lookup[gene_full_surr],\n",
    "                                         gene_full_y1000_id_lookup[gene_full_surr],\n",
    "                                         gene_full_surr,\n",
    "                                         CDS_surr.id,\n",
    "                                         CDS_surr.strand,\n",
    "                                         CDS_surr.start,\n",
    "                                         CDS_surr.end ))\n",
    "\n",
    "\n",
    "        surrounding_CDS_df = pd.DataFrame(surrounding_CDS_data, columns=['gene_id', 'y1000_id', 'gene_full', 'CDS_id', 'strand', 'start','stop'])\n",
    "\n",
    "        #Find genes in S. cer related to surrounding genes\n",
    "\n",
    "        goi_og_lookup_surr, og_genes_lookup_surr = y1000plus_tools.make_og_genes_lookup(surrounding_CDS_df['y1000_id'], y1000_species_subset)\n",
    "\n",
    "        scer_prefix = '110'\n",
    "\n",
    "        sc_genes_dict = {}\n",
    "        N_sc_orths = 0\n",
    "        N_cluster_orths = 0\n",
    "        for goi, og in goi_og_lookup_surr.items():\n",
    "            og_genes = og_genes_lookup_surr[og]\n",
    "            sc_genes = []\n",
    "            for og_gene in list(og_genes): \n",
    "                if (og_gene.split('_')[0]==scer_prefix):                         \n",
    "                    sc_genes.append(scer_y1000_id_gene_id_lookup[og_gene])\n",
    "            \n",
    "            if len(sc_genes)!=0: \n",
    "                N_sc_orths = N_sc_orths + 1\n",
    "                for sc_gene in sc_genes: \n",
    "                    if ((sc_gene[0:3]=='YLL') & (int(sc_gene[3:6]) in range(51,64))):\n",
    "                        N_cluster_orths = N_cluster_orths + 1\n",
    "                \n",
    "            sc_genes_common_names = yeast_esr_exp.SC_common_name_lookup(sc_genes)\n",
    "            sc_genes_names = []\n",
    "            for sc_gene, sc_gene_common_name in zip(sc_genes, sc_genes_common_names):\n",
    "                sc_genes_names.append('{} ({})'.format(sc_gene, sc_gene_common_name))\n",
    "            sc_genes_dict[goi] = sc_genes_names\n",
    "\n",
    "        #Output for each species: (sc_genes_dict, )\n",
    "        \n",
    "        N_total_genes = len(sc_genes_dict.keys())\n",
    "        \n",
    "        gene_synteny[hsu1_orth]= (sc_genes_dict, N_total_genes, N_sc_orths, N_cluster_orths )\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = 'YLL058W'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'saccharomyces_eubayanus_Seq_1484': ({'111_1483': ['YLL058W (YLL058W)',\n",
       "    'YML082W (YML082W)',\n",
       "    'YJR130C (STR2)'],\n",
       "   '111_68': [],\n",
       "   '111_3382': [],\n",
       "   '111_1158': [],\n",
       "   '111_3664': ['YER187W (YER187W)'],\n",
       "   '111_27': []},\n",
       "  6,\n",
       "  2,\n",
       "  1),\n",
       " 'saccharomyces_uvarum_Seq_2777': ({'115_2776': ['YLL058W (YLL058W)',\n",
       "    'YML082W (YML082W)',\n",
       "    'YJR130C (STR2)'],\n",
       "   '115_4480': [],\n",
       "   '115_3000': ['YLL063C (AYT1)'],\n",
       "   '115_4796': [],\n",
       "   '115_1161': ['YLL060C (GTT2)'],\n",
       "   '115_921': []},\n",
       "  6,\n",
       "  3,\n",
       "  3),\n",
       " 'yHMPu5000034862_zygotorulaspora_florentina_160519_Seq_3219': ({'217_1286': ['YLR286C (CTS1)'],\n",
       "   '217_3218': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '217_1177': [],\n",
       "   '217_107': [],\n",
       "   '217_4437': [],\n",
       "   '217_3368': []},\n",
       "  6,\n",
       "  2,\n",
       "  1),\n",
       " 'saccharomyces_kudriavzevii_Seq_59': ({'112_196': ['YLL057C (JLP1)'],\n",
       "   '112_58': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '112_1253': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '112_2991': ['YLL055W (YCT1)'],\n",
       "   '112_3241': ['YLL056C (YLL056C)'],\n",
       "   '112_3827': ['YLL063C (AYT1)'],\n",
       "   '112_3532': [],\n",
       "   '112_3361': [],\n",
       "   '112_3078': [],\n",
       "   '112_2811': [],\n",
       "   '112_1697': [],\n",
       "   '112_2605': [],\n",
       "   '112_2664': [],\n",
       "   '112_3233': []},\n",
       "  14,\n",
       "  6,\n",
       "  6),\n",
       " 'saccharomyces_arboricola_Seq_1887': ({'109_1437': ['YLL057C (JLP1)'],\n",
       "   '109_1886': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '109_251': ['YLL055W (YCT1)'],\n",
       "   '109_4510': ['YLL056C (YLL056C)'],\n",
       "   '109_4916': ['YLL063C (AYT1)'],\n",
       "   '109_589': [],\n",
       "   '109_2884': [],\n",
       "   '109_3780': [],\n",
       "   '109_3299': []},\n",
       "  9,\n",
       "  5,\n",
       "  5),\n",
       " 'saccharomyces_mikatae_Seq_3298': ({'113_2006': ['YKR104W (YKR104W)',\n",
       "    'YDR135C (YCF1)',\n",
       "    'YLL015W (BPT1)',\n",
       "    'YKR103W (NFT1)',\n",
       "    'YHL035C (VMR1)',\n",
       "    'YGR281W (YOR1)',\n",
       "    'YLL048C (YBT1)'],\n",
       "   '113_2404': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '113_4357': ['YNL209W (SSB2)',\n",
       "    'YEL030W (ECM10)',\n",
       "    'YER103W (SSA4)',\n",
       "    'YAL005C (SSA1)',\n",
       "    'YJR045C (SSC1)',\n",
       "    'YLR369W (SSQ1)',\n",
       "    'YBL075C (SSA3)',\n",
       "    'YDL229W (SSB1)',\n",
       "    'YLL024C (SSA2)',\n",
       "    'YJL034W (KAR2)'],\n",
       "   '113_4504': ['YLL026W (HSP104)', 'YDR258C (HSP78)'],\n",
       "   '113_43': ['YLL057C (JLP1)'],\n",
       "   '113_1657': ['YLL029W (FRA1)'],\n",
       "   '113_1784': ['YHL033C (RPL8A)', 'YLL045C (RPL8B)'],\n",
       "   '113_695': ['YLL028W (TPO1)'],\n",
       "   '113_3297': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '113_4551': ['YLL040C (VPS13)'],\n",
       "   '113_4603': ['YLL031C (GPI13)'],\n",
       "   '113_2124': ['YLL034C (RIX7)'],\n",
       "   '113_1864': ['YLL027W (ISA1)'],\n",
       "   '113_4305': ['YLL041C (SDH2)'],\n",
       "   '113_4472': ['YLL051C (FRE6)', 'YLL050C (COF1)'],\n",
       "   '113_2871': ['YLL051C (FRE6)', 'YLL050C (COF1)'],\n",
       "   '113_3897': ['YLL035W (GRC3)'],\n",
       "   '113_1195': ['YLL036C (PRP19)'],\n",
       "   '113_737': ['YLL032C (YLL032C)'],\n",
       "   '113_264': ['YPR192W (AQY1)'],\n",
       "   '113_4256': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '113_142': ['YLL055W (YCT1)'],\n",
       "   '113_2779': ['YLL021W (SPA2)'],\n",
       "   '113_2254': ['YLL023C (POM33)'],\n",
       "   '113_4149': ['YLL043W (FPS1)', 'YFL054C (AQY3)'],\n",
       "   '113_1405': ['YLL054C (YLL054C)', 'YCR106W (RDS1)'],\n",
       "   '113_1579': ['YLL033W (IRC19)'],\n",
       "   '113_1533': ['YLL042C (ATG10)'],\n",
       "   '113_2888': ['YLR037C (PAU23)', 'YLL025W (PAU17)'],\n",
       "   '113_2032': ['YLL049W (LDB18)'],\n",
       "   '113_3378': ['YLL046C (RNP1)'],\n",
       "   '113_4376': ['YLL022C (HIF1)'],\n",
       "   '113_143': [],\n",
       "   '113_4815': []},\n",
       "  34,\n",
       "  32,\n",
       "  8),\n",
       " 'saccharomyces_paradoxus_Seq_3252': ({'114_3719': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '114_4723': ['YLL057C (JLP1)'],\n",
       "   '114_3251': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '114_2792': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '114_1212': ['YLL055W (YCT1)'],\n",
       "   '114_1573': ['YLL056C (YLL056C)'],\n",
       "   '114_637': ['YLL054C (YLL054C)', 'YCR106W (RDS1)'],\n",
       "   '114_3457': ['YLL060C (GTT2)'],\n",
       "   '114_2075': []},\n",
       "  9,\n",
       "  8,\n",
       "  8),\n",
       " 'yHMPu5000026256_zygotorulaspora_mrakii_161215_Seq_3510': ({'166_2322': ['YEL065W (SIT1)',\n",
       "    'YKR106W (GEX2)',\n",
       "    'YCL073C (GEX1)',\n",
       "    'YHL040C (ARN1)',\n",
       "    'YHL047C (ARN2)'],\n",
       "   '166_3509': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '166_3898': ['YOL154W (ZPS1)'],\n",
       "   '166_663': [],\n",
       "   '166_3022': []},\n",
       "  5,\n",
       "  3,\n",
       "  1),\n",
       " 'yHMPu5000035652_torulaspora_maleeae_160613_Seq_388': ({'300_3288': ['YDL247W (MPH2)',\n",
       "    'YJR160C (MPH3)',\n",
       "    'YBR298C (MAL31)',\n",
       "    'YGR289C (MAL11)'],\n",
       "   '300_3585': ['YBR092C (PHO3)',\n",
       "    'YHR215W (PHO12)',\n",
       "    'YDL024C (DIA3)',\n",
       "    'YAR071W (PHO11)',\n",
       "    'YBR093C (PHO5)'],\n",
       "   '300_947': ['YPR022C (SDD4)'],\n",
       "   '300_387': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '300_938': ['YNR053C (NOG2)'],\n",
       "   '300_363': ['YMR283C (RIT1)'],\n",
       "   '300_1371': ['YNR054C (ESF2)']},\n",
       "  7,\n",
       "  7,\n",
       "  1),\n",
       " 'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_2166': ({'299_5096': ['YJR152W (DAL5)'],\n",
       "   '299_4188': ['YJR152W (DAL5)'],\n",
       "   '299_4439': ['YPR181C (SEC23)'],\n",
       "   '299_1312': ['YPR181C (SEC23)'],\n",
       "   '299_2222': ['YLR454W (FMP27)', 'YPR117W (YPR117W)'],\n",
       "   '299_1781': ['YLR454W (FMP27)', 'YPR117W (YPR117W)'],\n",
       "   '299_4989': ['YER175C (TMT1)', 'YHR209W (CRG1)'],\n",
       "   '299_5111': ['YER175C (TMT1)', 'YHR209W (CRG1)'],\n",
       "   '299_2978': ['YKL221W (MCH2)'],\n",
       "   '299_1668': ['YKL221W (MCH2)'],\n",
       "   '299_2165': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '299_17': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '299_2749': ['YPR165W (RHO1)'],\n",
       "   '299_3243': ['YPR165W (RHO1)'],\n",
       "   '299_4840': ['YPR165W (RHO1)'],\n",
       "   '299_367': ['YPR165W (RHO1)'],\n",
       "   '299_1574': ['YPR172W (YPR172W)', 'YLR456W (YLR456W)'],\n",
       "   '299_293': ['YPR172W (YPR172W)', 'YLR456W (YLR456W)'],\n",
       "   '299_540': ['YPR173C (VPS4)'],\n",
       "   '299_3104': ['YPR173C (VPS4)'],\n",
       "   '299_1590': ['YPR175W (DPB2)'],\n",
       "   '299_2089': ['YPR175W (DPB2)'],\n",
       "   '299_5180': ['YPR189W (SKI3)'],\n",
       "   '299_3450': ['YPR189W (SKI3)'],\n",
       "   '299_1375': ['YPR191W (QCR2)'],\n",
       "   '299_1304': ['YPR191W (QCR2)'],\n",
       "   '299_4530': ['YPR190C (RPC82)'],\n",
       "   '299_3410': ['YPR190C (RPC82)'],\n",
       "   '299_4502': ['YPR187W (RPO26)'],\n",
       "   '299_1646': ['YPR187W (RPO26)'],\n",
       "   '299_609': ['YPR184W (GDB1)'],\n",
       "   '299_4482': ['YPR184W (GDB1)'],\n",
       "   '299_4255': ['YPR186C (PZF1)'],\n",
       "   '299_678': ['YPR186C (PZF1)'],\n",
       "   '299_1436': ['YPR176C (BET2)'],\n",
       "   '299_3307': ['YPR176C (BET2)'],\n",
       "   '299_980': ['YPR167C (MET16)'],\n",
       "   '299_4041': ['YPR167C (MET16)'],\n",
       "   '299_2785': ['YLR459W (GAB1)'],\n",
       "   '299_3705': ['YLR459W (GAB1)'],\n",
       "   '299_1155': ['YPR180W (AOS1)'],\n",
       "   '299_5265': ['YPR180W (AOS1)'],\n",
       "   '299_1694': ['YPR179C (HDA3)'],\n",
       "   '299_1391': ['YPR179C (HDA3)'],\n",
       "   '299_2050': ['YPR168W (NUT2)'],\n",
       "   '299_93': ['YPR168W (NUT2)'],\n",
       "   '299_2587': ['YPR178W (PRP4)'],\n",
       "   '299_2197': ['YPR178W (PRP4)'],\n",
       "   '299_2724': ['YPR185W (ATG13)'],\n",
       "   '299_1824': ['YPR185W (ATG13)'],\n",
       "   '299_974': ['YPR169W (JIP5)'],\n",
       "   '299_4495': ['YPR169W (JIP5)'],\n",
       "   '299_4694': ['YPR183W (DPM1)'],\n",
       "   '299_2928': ['YPR183W (DPM1)'],\n",
       "   '299_1826': [],\n",
       "   '299_3574': [],\n",
       "   '299_442': [],\n",
       "   '299_3850': [],\n",
       "   '299_4278': ['YPR188C (MLC2)'],\n",
       "   '299_4289': ['YPR188C (MLC2)'],\n",
       "   '299_2196': [],\n",
       "   '299_202': [],\n",
       "   '299_4012': ['YLL056C (YLL056C)'],\n",
       "   '299_2772': ['YLR455W (PDP3)'],\n",
       "   '299_2245': ['YLR455W (PDP3)'],\n",
       "   '299_3048': ['YPR164W (MMS1)'],\n",
       "   '299_2908': ['YPR164W (MMS1)'],\n",
       "   '299_599': ['YLR457C (NBP1)'],\n",
       "   '299_2011': ['YLR457C (NBP1)'],\n",
       "   '299_1879': ['YKL222C (YKL222C)'],\n",
       "   '299_3596': ['YKL222C (YKL222C)'],\n",
       "   '299_3570': [],\n",
       "   '299_303': ['YPR171W (BSP1)'],\n",
       "   '299_1656': ['YPR171W (BSP1)'],\n",
       "   '299_2732': [],\n",
       "   '299_1612': [],\n",
       "   '299_197': [],\n",
       "   '299_2539': [],\n",
       "   '299_1950': [],\n",
       "   '299_3749': []},\n",
       "  80,\n",
       "  67,\n",
       "  3),\n",
       " 'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_18': ({'299_4188': ['YJR152W (DAL5)'],\n",
       "   '299_5111': ['YER175C (TMT1)', 'YHR209W (CRG1)'],\n",
       "   '299_1668': ['YKL221W (MCH2)'],\n",
       "   '299_17': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '299_1826': [],\n",
       "   '299_202': [],\n",
       "   '299_3596': ['YKL222C (YKL222C)'],\n",
       "   '299_1612': []},\n",
       "  8,\n",
       "  5,\n",
       "  1),\n",
       " 'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_3689': ({'299_3688': ['YLL058W (YLL058W)',\n",
       "    'YML082W (YML082W)',\n",
       "    'YJR130C (STR2)'],\n",
       "   '299_4587': [],\n",
       "   '299_1540': ['YLL055W (YCT1)'],\n",
       "   '299_2284': [],\n",
       "   '299_805': [],\n",
       "   '299_804': []},\n",
       "  6,\n",
       "  2,\n",
       "  2),\n",
       " 'yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519_Seq_3775': ({'218_1495': ['YJR152W (DAL5)'],\n",
       "   '218_359': ['YJR015W (YJR015W)', 'YGR197C (SNG1)'],\n",
       "   '218_3774': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '218_4195': ['YLL055W (YCT1)'],\n",
       "   '218_3690': [],\n",
       "   '218_2882': [],\n",
       "   '218_1042': []},\n",
       "  7,\n",
       "  4,\n",
       "  2),\n",
       " 'yHMPu5000034881_torulaspora_pretoriensis_160519_Seq_4597': ({'223_3317': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '223_5431': ['YLR144C (ACF2)', 'YNR067C (DSE4)'],\n",
       "   '223_3448': ['YIL166C (YIL166C)'],\n",
       "   '223_5502': ['YLL057C (JLP1)'],\n",
       "   '223_1148': [],\n",
       "   '223_4596': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '223_4136': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '223_3439': [],\n",
       "   '223_2826': []},\n",
       "  9,\n",
       "  6,\n",
       "  4),\n",
       " 'yHMPu5000026152_torulaspora_franciscae_160519_Seq_4502': ({'164_1858': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '164_1518': ['YLR144C (ACF2)', 'YNR067C (DSE4)'],\n",
       "   '164_2372': [],\n",
       "   '164_4501': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '164_2190': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '164_3906': [],\n",
       "   '164_3065': []},\n",
       "  7,\n",
       "  4,\n",
       "  3),\n",
       " 'torulaspora_delbrueckii_Seq_4236': ({'135_1052': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '135_2650': ['YLR144C (ACF2)', 'YNR067C (DSE4)'],\n",
       "   '135_2694': ['YDL178W (DLD2)', 'YEL071W (DLD3)'],\n",
       "   '135_4235': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '135_1272': ['YPL273W (SAM4)', 'YLL062C (MHT1)'],\n",
       "   '135_2478': ['YLL055W (YCT1)'],\n",
       "   '135_1470': []},\n",
       "  7,\n",
       "  6,\n",
       "  4),\n",
       " 'lachancea_cidri_Seq_3705': ({'54_510': ['YDL247W (MPH2)',\n",
       "    'YJR160C (MPH3)',\n",
       "    'YBR298C (MAL31)',\n",
       "    'YGR289C (MAL11)'],\n",
       "   '54_2788': ['YNL330C (RPD3)'],\n",
       "   '54_3704': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '54_4515': ['YOL145C (CTR9)'],\n",
       "   '54_1800': ['YNL329C (PEX6)'],\n",
       "   '54_3130': ['YOL148C (SPT20)'],\n",
       "   '54_968': ['YOL146W (PSF3)'],\n",
       "   '54_3223': ['YAR050W (FLO1)', 'YHR211W (FLO5)', 'YAL063C (FLO9)'],\n",
       "   '54_2041': ['YDR242W (AMD2)'],\n",
       "   '54_1940': ['YOL144W (NOP8)'],\n",
       "   '54_4016': ['YOL149W (DCP1)'],\n",
       "   '54_3758': [],\n",
       "   '54_1088': ['YNL328C (MDJ2)'],\n",
       "   '54_3878': [],\n",
       "   '54_486': ['YLL056C (YLL056C)'],\n",
       "   '54_1779': [],\n",
       "   '54_3414': ['YOR383C (FIT3)', 'YDR534C (FIT1)'],\n",
       "   '54_2529': ['YOR382W (FIT2)'],\n",
       "   '54_489': ['YOR384W (FRE5)'],\n",
       "   '54_1012': [],\n",
       "   '54_3059': [],\n",
       "   '54_3304': [],\n",
       "   '54_3512': [],\n",
       "   '54_5': [],\n",
       "   '54_34': []},\n",
       "  25,\n",
       "  16,\n",
       "  2),\n",
       " 'lachancea_fermentati_Seq_4096': ({'57_3374': ['YCL064C (CHA1)'],\n",
       "   '57_4095': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '57_1304': ['YIR038C (GTT1)'],\n",
       "   '57_2799': ['YCL061C (MRC1)'],\n",
       "   '57_645': ['YCL063W (VAC17)'],\n",
       "   '57_3210': [],\n",
       "   '57_1326': []},\n",
       "  7,\n",
       "  5,\n",
       "  1),\n",
       " 'yHMPu5000034862_zygotorulaspora_florentina_160519_Seq_956': ({'217_1299': [],\n",
       "   '217_955': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '217_836': ['YLL055W (YCT1)'],\n",
       "   '217_800': [],\n",
       "   '217_423': [],\n",
       "   '217_1848': [],\n",
       "   '217_2528': [],\n",
       "   '217_4748': [],\n",
       "   '217_3444': [],\n",
       "   '217_1018': []},\n",
       "  10,\n",
       "  2,\n",
       "  2),\n",
       " 'lachancea_nothofagi_Seq_4267': ({'62_4133': ['YGR260W (TNA1)'],\n",
       "   '62_4266': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '62_2037': ['YAR050W (FLO1)', 'YHR211W (FLO5)', 'YAL063C (FLO9)'],\n",
       "   '62_2761': ['YGR234W (YHB1)'],\n",
       "   '62_1466': [],\n",
       "   '62_2188': [],\n",
       "   '62_1228': []},\n",
       "  7,\n",
       "  4,\n",
       "  1),\n",
       " 'yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519_Seq_4055': ({'218_4054': ['YLL058W (YLL058W)',\n",
       "    'YML082W (YML082W)',\n",
       "    'YJR130C (STR2)'],\n",
       "   '218_2990': []},\n",
       "  2,\n",
       "  1,\n",
       "  1),\n",
       " 'zygosaccharomyces_bailii_Seq_2609': ({'341_1724': ['YBR187W (GDT1)',\n",
       "    'YBR186W (PCH2)'],\n",
       "   '341_2608': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '341_739': ['YHR029C (YHI9)'],\n",
       "   '341_2913': [],\n",
       "   '341_424': [],\n",
       "   '341_2110': []},\n",
       "  6,\n",
       "  3,\n",
       "  1),\n",
       " 'yHMPu5000034866_zygosaccharomyces_bisporus_160519_Seq_959': ({'219_332': ['YDL144C (YDL144C)'],\n",
       "   '219_958': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '219_734': ['YIL051C (MMF1)'],\n",
       "   '219_1931': ['YDR242W (AMD2)'],\n",
       "   '219_330': [],\n",
       "   '219_235': [],\n",
       "   '219_1585': []},\n",
       "  7,\n",
       "  4,\n",
       "  1),\n",
       " 'lachancea_waltii_Seq_3824': ({'65_300': ['YGR260W (TNA1)'],\n",
       "   '65_880': ['YKL215C (OXP1)'],\n",
       "   '65_1256': ['YML054C (CYB2)'],\n",
       "   '65_2008': [],\n",
       "   '65_3823': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '65_3277': [],\n",
       "   '65_4187': []},\n",
       "  7,\n",
       "  4,\n",
       "  1),\n",
       " 'lachancea_quebecensis_Seq_2817': ({'63_2816': ['YLL058W (YLL058W)',\n",
       "    'YML082W (YML082W)',\n",
       "    'YJR130C (STR2)'],\n",
       "   '63_3002': ['YGL258W (VEL1)', 'YOR387C (YOR387C)'],\n",
       "   '63_128': [],\n",
       "   '63_882': []},\n",
       "  4,\n",
       "  2,\n",
       "  1),\n",
       " 'lachancea_thermotolerans_Seq_2696': ({'64_2337': ['YGL253W (HXK2)',\n",
       "    'YFR053C (HXK1)'],\n",
       "   '64_2695': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '64_4093': ['YFR052W (RPN12)'],\n",
       "   '64_4177': ['YGL252C (RTG2)']},\n",
       "  4,\n",
       "  4,\n",
       "  1),\n",
       " 'lachancea_lanzarotensis_Seq_1470': ({'59_4245': ['YGR224W (AZR1)',\n",
       "    'YCL069W (VBA3)',\n",
       "    'YKR105C (VBA5)',\n",
       "    'YPR198W (SGE1)'],\n",
       "   '59_3255': ['YIL120W (QDR1)', 'YIL121W (QDR2)', 'YNL065W (AQR1)'],\n",
       "   '59_1469': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '59_2117': ['YFL061W (DDI2)', 'YNL335W (DDI3)'],\n",
       "   '59_1323': [],\n",
       "   '59_3492': [],\n",
       "   '59_3636': []},\n",
       "  7,\n",
       "  4,\n",
       "  1),\n",
       " 'lachancea_fantastica_Seq_5001': ({'56_950': ['YBR092C (PHO3)',\n",
       "    'YHR215W (PHO12)',\n",
       "    'YDL024C (DIA3)',\n",
       "    'YAR071W (PHO11)',\n",
       "    'YBR093C (PHO5)'],\n",
       "   '56_5000': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '56_1980': ['YJL218W (YJL218W)'],\n",
       "   '56_2942': [],\n",
       "   '56_1829': ['YMR319C (FET4)'],\n",
       "   '56_3836': ['YKL216W (URA1)'],\n",
       "   '56_4990': []},\n",
       "  7,\n",
       "  5,\n",
       "  1),\n",
       " 'lachancea_mirantina_Seq_873': ({'61_2125': ['YBR008C (FLR1)'],\n",
       "   '61_1918': ['YMR279C (YMR279C)', 'YOR378W (AMF1)', 'YML116W (ATR1)'],\n",
       "   '61_872': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '61_1389': ['YKL060C (FBA1)'],\n",
       "   '61_310': [],\n",
       "   '61_2726': [],\n",
       "   '61_3435': []},\n",
       "  7,\n",
       "  4,\n",
       "  1),\n",
       " 'lachancea_dasiensis_Seq_3537': ({'55_1984': ['YMR089C (YTA12)',\n",
       "    'YER017C (AFG3)'],\n",
       "   '55_3713': ['YJR152W (DAL5)'],\n",
       "   '55_2285': ['YKL140W (TGL1)'],\n",
       "   '55_3778': [],\n",
       "   '55_2018': ['YMR109W (MYO5)', 'YKL129C (MYO3)'],\n",
       "   '55_749': ['YMR105C (PGM2)', 'YKL127W (PGM1)'],\n",
       "   '55_3849': ['YGL202W (ARO8)'],\n",
       "   '55_2462': ['YMR108W (ILV2)'],\n",
       "   '55_3536': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '55_1226': ['YMR110C (HFD1)'],\n",
       "   '55_3313': ['YMR102C (YMR102C)', 'YKL121W (DGR2)'],\n",
       "   '55_4020': ['YMR116C (ASC1)'],\n",
       "   '55_937': ['YKL113C (RAD27)'],\n",
       "   '55_3891': ['YMR093W (UTP15)'],\n",
       "   '55_1231': ['YKL134C (OCT1)'],\n",
       "   '55_4586': ['YKL110C (KTI12)'],\n",
       "   '55_4779': ['YKL135C (APL2)'],\n",
       "   '55_4627': ['YKL117W (SBA1)'],\n",
       "   '55_1571': ['YMR099C (YMR099C)'],\n",
       "   '55_1004': ['YKL120W (OAC1)'],\n",
       "   '55_1516': ['YKL125W (RRN3)'],\n",
       "   '55_3889': ['YMR092C (AIP1)'],\n",
       "   '55_263': ['YMR097C (MTG1)'],\n",
       "   '55_2035': ['YKL138C (MRPL31)'],\n",
       "   '55_3949': ['YKL133C (YKL133C)', 'YMR115W (MGR3)'],\n",
       "   '55_4199': ['YMR100W (MUB1)'],\n",
       "   '55_1922': ['YKL119C (VPH2)'],\n",
       "   '55_3441': ['YKL116C (PRR1)'],\n",
       "   '55_2876': ['YLR026C (SED5)'],\n",
       "   '55_3256': ['YMR106C (YKU80)'],\n",
       "   '55_2708': ['YMR091C (NPL6)'],\n",
       "   '55_1618': ['YMR101C (SRT1)'],\n",
       "   '55_4459': ['YMR113W (FOL3)'],\n",
       "   '55_4378': ['YLR025W (SNF7)'],\n",
       "   '55_3293': ['YFL059W (SNZ3)', 'YNL333W (SNZ2)', 'YMR096W (SNZ1)'],\n",
       "   '55_1934': ['YKL137W (CMC1)'],\n",
       "   '55_3563': ['YMR114C (YMR114C)'],\n",
       "   '55_2624': ['YFL060C (SNO3)', 'YMR095C (SNO1)', 'YNL334C (SNO2)'],\n",
       "   '55_2679': ['YMR112C (MED11)'],\n",
       "   '55_4205': ['YKL124W (SSH4)'],\n",
       "   '55_3156': ['YKL126W (YPK1)', 'YMR104C (YPK2)'],\n",
       "   '55_489': ['YMR098C (ATP25)'],\n",
       "   '55_1879': ['YKL122C (SRP21)'],\n",
       "   '55_2409': ['YKL109W (HAP4)', 'YKL108W (SLD2)'],\n",
       "   '55_752': ['YKL130C (SHE2)'],\n",
       "   '55_1346': [],\n",
       "   '55_412': ['YMR094W (CTF13)'],\n",
       "   '55_4014': ['YMR117C (SPC24)'],\n",
       "   '55_2405': ['YMR107W (SPG4)'],\n",
       "   '55_1066': ['YMR111C (YMR111C)'],\n",
       "   '55_1931': ['YKL138C-A (HSK3)'],\n",
       "   '55_3474': [],\n",
       "   '55_1186': [],\n",
       "   '55_3425': [],\n",
       "   '55_3124': [],\n",
       "   '55_533': []},\n",
       "  56,\n",
       "  49,\n",
       "  1),\n",
       " 'lachancea_fantastica_Seq_4073': ({'56_3404': ['YBR068C (BAP2)',\n",
       "    'YDR508C (GNP1)',\n",
       "    'YPL274W (SAM3)',\n",
       "    'YDR046C (BAP3)',\n",
       "    'YKR039W (GAP1)',\n",
       "    'YGR191W (HIP1)',\n",
       "    'YBR069C (TAT1)',\n",
       "    'YLL061W (MMP1)',\n",
       "    'YCL025C (AGP1)',\n",
       "    'YOL020W (TAT2)'],\n",
       "   '56_2122': ['YGR224W (AZR1)',\n",
       "    'YCL069W (VBA3)',\n",
       "    'YKR105C (VBA5)',\n",
       "    'YPR198W (SGE1)'],\n",
       "   '56_4089': ['YDL124W (YDL124W)'],\n",
       "   '56_4554': ['YGL202W (ARO8)'],\n",
       "   '56_4072': ['YLL058W (YLL058W)', 'YML082W (YML082W)', 'YJR130C (STR2)'],\n",
       "   '56_4226': ['YBR043C (QDR3)'],\n",
       "   '56_4636': [],\n",
       "   '56_2367': [],\n",
       "   '56_2118': [],\n",
       "   '56_933': [],\n",
       "   '56_700': [],\n",
       "   '56_2391': [],\n",
       "   '56_864': [],\n",
       "   '56_827': [],\n",
       "   '56_4945': [],\n",
       "   '56_1811': [],\n",
       "   '56_4330': [],\n",
       "   '56_386': [],\n",
       "   '56_1941': [],\n",
       "   '56_3672': [],\n",
       "   '56_1061': [],\n",
       "   '56_3265': [],\n",
       "   '56_488': []},\n",
       "  23,\n",
       "  6,\n",
       "  2)}"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_synteny"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'56_488'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "goi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'C:\\\\Users\\\\heineib\\\\Google Drive\\\\Crick_LMS\\\\projects\\\\jason_met_auxotrophy\\\\'"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metproj_data_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'nodes': [[79.4218757801101,\n",
       "   244.10986328125,\n",
       "   83.4218757801101,\n",
       "   248.10986328125,\n",
       "   0,\n",
       "   None],\n",
       "  [143.69822256018895, 54.0, 147.69822256018895, 58.0, 1, None],\n",
       "  [217.52814199639514, 26.5, 221.52814199639514, 30.5, 2, None],\n",
       "  [215.59483984263412, 81.5, 219.59483984263412, 85.5, 3, None],\n",
       "  [141.53605282708412,\n",
       "   434.2197265625,\n",
       "   145.53605282708412,\n",
       "   438.2197265625,\n",
       "   4,\n",
       "   None],\n",
       "  [183.53605282708412, 225.875, 187.53605282708412, 229.875, 5, None],\n",
       "  [225.53605282708412, 164.0, 229.53605282708412, 168.0, 6, None],\n",
       "  [260.5360528270841, 136.5, 264.5360528270841, 140.5, 7, None],\n",
       "  [260.5360528270841, 191.5, 264.5360528270841, 195.5, 8, None],\n",
       "  [218.53605282708412, 287.75, 222.53605282708412, 291.75, 9, None],\n",
       "  [260.36508661066273, 246.5, 264.36508661066273, 250.5, 10, None],\n",
       "  [260.5360528270841, 329.0, 264.5360528270841, 333.0, 11, None],\n",
       "  [305.5392672625773, 301.5, 309.5392672625773, 305.5, 12, None],\n",
       "  [298.07386218784006, 356.5, 302.07386218784006, 360.5, 13, None],\n",
       "  [183.53605282708412,\n",
       "   642.564453125,\n",
       "   187.53605282708412,\n",
       "   646.564453125,\n",
       "   14,\n",
       "   None],\n",
       "  [225.53605282708412, 463.0625, 229.53605282708412, 467.0625, 15, None],\n",
       "  [288.2414208910842, 411.5, 292.2414208910842, 415.5, 16, None],\n",
       "  [267.5360528270841, 514.625, 271.5360528270841, 518.625, 17, None],\n",
       "  [321.46519724049534, 466.5, 325.46519724049534, 470.5, 18, None],\n",
       "  [309.5360528270841, 562.75, 313.5360528270841, 566.75, 19, None],\n",
       "  [353.3145323440297, 521.5, 357.3145323440297, 525.5, 20, None],\n",
       "  [350.5360528270841, 604.0, 354.5360528270841, 608.0, 21, None],\n",
       "  [392.5360528270841, 576.5, 396.5360528270841, 580.5, 22, None],\n",
       "  [396.25446656340176, 631.5, 400.25446656340176, 635.5, 23, None],\n",
       "  [225.53605282708412,\n",
       "   822.06640625,\n",
       "   229.53605282708412,\n",
       "   826.06640625,\n",
       "   24,\n",
       "   None],\n",
       "  [297.20215217102947, 686.5, 301.20215217102947, 690.5, 25, None],\n",
       "  [266.5360528270841, 957.6328125, 270.5360528270841, 961.6328125, 26, None],\n",
       "  [308.5360528270841, 769.0, 312.5360528270841, 773.0, 27, None],\n",
       "  [345.73204893681543, 741.5, 349.73204893681543, 745.5, 28, None],\n",
       "  [374.07956154391223, 796.5, 378.07956154391223, 800.5, 29, None],\n",
       "  [308.5360528270841, 1146.265625, 312.5360528270841, 1150.265625, 30, None],\n",
       "  [350.5360528270841, 940.875, 354.5360528270841, 944.875, 31, None],\n",
       "  [385.5360528270841, 879.0, 389.5360528270841, 883.0, 32, None],\n",
       "  [427.5360528270841, 851.5, 431.5360528270841, 855.5, 33, None],\n",
       "  [427.5360528270841, 906.5, 431.5360528270841, 910.5, 34, None],\n",
       "  [392.5360528270841, 1002.75, 396.5360528270841, 1006.75, 35, None],\n",
       "  [434.5360528270841, 961.5, 438.5360528270841, 965.5, 36, None],\n",
       "  [437.2403454576587, 1044.0, 441.2403454576587, 1048.0, 37, None],\n",
       "  [479.2403454576587, 1016.5, 483.2403454576587, 1020.5, 38, None],\n",
       "  [479.2403454576587, 1071.5, 483.2403454576587, 1075.5, 39, None],\n",
       "  [350.5360528270841, 1351.65625, 354.5360528270841, 1355.65625, 40, None],\n",
       "  [392.5360528270841, 1209.0, 396.5360528270841, 1213.0, 41, None],\n",
       "  [434.5360528270841, 1154.0, 438.5360528270841, 1158.0, 42, None],\n",
       "  [487.761818717455, 1126.5, 491.761818717455, 1130.5, 43, None],\n",
       "  [509.4406967837978, 1181.5, 513.4406967837979, 1185.5, 44, None],\n",
       "  [427.5360528270841, 1264.0, 431.5360528270841, 1268.0, 45, None],\n",
       "  [480.5360528270841, 1236.5, 484.5360528270841, 1240.5, 46, None],\n",
       "  [473.5360528270841, 1291.5, 477.5360528270841, 1295.5, 47, None],\n",
       "  [392.5360528270841, 1494.3125, 396.5360528270841, 1498.3125, 48, None],\n",
       "  [434.5360528270841, 1387.75, 438.5360528270841, 1391.75, 49, None],\n",
       "  [471.44428944425584, 1346.5, 475.44428944425584, 1350.5, 50, None],\n",
       "  [476.5360528270841, 1429.0, 480.5360528270841, 1433.0, 51, None],\n",
       "  [518.5360528270842, 1401.5, 522.5360528270842, 1405.5, 52, None],\n",
       "  [518.5360528270842, 1456.5, 522.5360528270842, 1460.5, 53, None],\n",
       "  [434.5360528270841, 1600.875, 438.5360528270841, 1604.875, 54, None],\n",
       "  [476.5360528270841, 1539.0, 480.5360528270841, 1543.0, 55, None],\n",
       "  [518.5360528270842, 1511.5, 522.5360528270842, 1515.5, 56, None],\n",
       "  [518.5360528270842, 1566.5, 522.5360528270842, 1570.5, 57, None],\n",
       "  [476.5360528270841, 1662.75, 480.5360528270841, 1666.75, 58, None],\n",
       "  [518.5360528270842, 1621.5, 522.5360528270842, 1625.5, 59, None],\n",
       "  [518.5360528270842, 1704.0, 522.5360528270842, 1708.0, 60, None],\n",
       "  [560.5360528270842, 1676.5, 564.5360528270842, 1680.5, 61, None],\n",
       "  [560.5360528270842, 1731.5, 564.5360528270842, 1735.5, 62, None]],\n",
       " 'faces': [[47.9218757801101,\n",
       "   230.10986328125,\n",
       "   79.9218757801101,\n",
       "   246.10986328125,\n",
       "   0,\n",
       "   '0.341'],\n",
       "  [112.19822256018895, 40.0, 144.19822256018895, 56.0, 1, '0.265'],\n",
       "  [221.02814199639514,\n",
       "   19.0,\n",
       "   456.0281419963951,\n",
       "   38.0,\n",
       "   2,\n",
       "   'lachancea_dasiensis_Seq_3537'],\n",
       "  [186.02814199639516, 12.5, 218.02814199639516, 28.5, 2, '0.306'],\n",
       "  [1001.5360528270841,\n",
       "   0.5,\n",
       "   1072.536052827084,\n",
       "   56.5,\n",
       "   2,\n",
       "   {'text': '55', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   0.5,\n",
       "   1142.536052827084,\n",
       "   56.5,\n",
       "   2,\n",
       "   {'text': '87%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   0.5,\n",
       "   1212.536052827084,\n",
       "   56.5,\n",
       "   2,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [219.09483984263412,\n",
       "   74.0,\n",
       "   456.0948398426341,\n",
       "   93.0,\n",
       "   3,\n",
       "   'lachancea_fantastica_Seq_4073'],\n",
       "  [184.0948398426341, 67.5, 216.0948398426341, 83.5, 3, '0.298'],\n",
       "  [1001.5360528270841,\n",
       "   55.5,\n",
       "   1072.536052827084,\n",
       "   111.5,\n",
       "   3,\n",
       "   {'text': '22', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   55.5,\n",
       "   1142.536052827084,\n",
       "   111.5,\n",
       "   3,\n",
       "   {'text': '23%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   55.5,\n",
       "   1212.536052827084,\n",
       "   111.5,\n",
       "   3,\n",
       "   {'text': '20%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [110.03605282708412,\n",
       "   420.2197265625,\n",
       "   142.03605282708412,\n",
       "   436.2197265625,\n",
       "   4,\n",
       "   '0.256'],\n",
       "  [145.03605282708412, 211.875, 184.03605282708412, 227.875, 5, '0.0605'],\n",
       "  [187.03605282708412, 150.0, 226.03605282708412, 166.0, 6, '0.0821'],\n",
       "  [264.0360528270841,\n",
       "   129.0,\n",
       "   523.0360528270842,\n",
       "   148.0,\n",
       "   7,\n",
       "   'lachancea_quebecensis_Seq_2817'],\n",
       "  [229.03605282708412, 122.5, 261.0360528270841, 138.5, 7, '0.123'],\n",
       "  [1001.5360528270842,\n",
       "   110.5,\n",
       "   1072.5360528270842,\n",
       "   166.5,\n",
       "   7,\n",
       "   {'text': '3', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   110.5,\n",
       "   1142.5360528270842,\n",
       "   166.5,\n",
       "   7,\n",
       "   {'text': '33%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   110.5,\n",
       "   1212.5360528270842,\n",
       "   166.5,\n",
       "   7,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [264.0360528270841,\n",
       "   184.0,\n",
       "   538.0360528270842,\n",
       "   203.0,\n",
       "   8,\n",
       "   'lachancea_thermotolerans_Seq_2696'],\n",
       "  [229.03605282708412, 177.5, 261.0360528270841, 193.5, 8, '0.126'],\n",
       "  [1001.5360528270842,\n",
       "   165.5,\n",
       "   1072.5360528270842,\n",
       "   221.5,\n",
       "   8,\n",
       "   {'text': '3', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   165.5,\n",
       "   1142.5360528270842,\n",
       "   221.5,\n",
       "   8,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   165.5,\n",
       "   1212.5360528270842,\n",
       "   221.5,\n",
       "   8,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [187.03605282708412, 273.75, 219.03605282708412, 289.75, 9, '0.137'],\n",
       "  [263.86508661066273,\n",
       "   239.0,\n",
       "   488.86508661066273,\n",
       "   258.0,\n",
       "   10,\n",
       "   'lachancea_mirantina_Seq_873'],\n",
       "  [228.86508661066273, 232.5, 260.86508661066273, 248.5, 10, '0.168'],\n",
       "  [1001.5360528270842,\n",
       "   220.5,\n",
       "   1072.5360528270842,\n",
       "   276.5,\n",
       "   10,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   220.5,\n",
       "   1142.5360528270842,\n",
       "   276.5,\n",
       "   10,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   220.5,\n",
       "   1212.5360528270842,\n",
       "   276.5,\n",
       "   10,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [222.03605282708412, 315.0, 261.0360528270841, 331.0, 11, '0.0428'],\n",
       "  [309.0392672625773,\n",
       "   294.0,\n",
       "   571.0392672625774,\n",
       "   313.0,\n",
       "   12,\n",
       "   'lachancea_lanzarotensis_Seq_1470'],\n",
       "  [274.0392672625773, 287.5, 306.0392672625773, 303.5, 12, '0.182'],\n",
       "  [1001.5360528270842,\n",
       "   275.5,\n",
       "   1072.5360528270842,\n",
       "   331.5,\n",
       "   12,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   275.5,\n",
       "   1142.5360528270842,\n",
       "   331.5,\n",
       "   12,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   275.5,\n",
       "   1212.5360528270842,\n",
       "   331.5,\n",
       "   12,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [301.57386218784006,\n",
       "   349.0,\n",
       "   538.5738621878401,\n",
       "   368.0,\n",
       "   13,\n",
       "   'lachancea_fantastica_Seq_5001'],\n",
       "  [266.57386218784006, 342.5, 298.57386218784006, 358.5, 13, '0.149'],\n",
       "  [1001.5360528270842,\n",
       "   330.5,\n",
       "   1072.5360528270842,\n",
       "   386.5,\n",
       "   13,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   330.5,\n",
       "   1142.5360528270842,\n",
       "   386.5,\n",
       "   13,\n",
       "   {'text': '67%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   330.5,\n",
       "   1212.5360528270842,\n",
       "   386.5,\n",
       "   13,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [145.03605282708412,\n",
       "   628.564453125,\n",
       "   184.03605282708412,\n",
       "   644.564453125,\n",
       "   14,\n",
       "   '0.0407'],\n",
       "  [187.03605282708412, 449.0625, 226.03605282708412, 465.0625, 15, '0.0337'],\n",
       "  [291.7414208910842,\n",
       "   404.0,\n",
       "   493.7414208910842,\n",
       "   423.0,\n",
       "   16,\n",
       "   'lachancea_waltii_Seq_3824'],\n",
       "  [256.7414208910842, 397.5, 288.7414208910842, 413.5, 16, '0.258'],\n",
       "  [1001.5360528270842,\n",
       "   385.5,\n",
       "   1072.5360528270842,\n",
       "   441.5,\n",
       "   16,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   385.5,\n",
       "   1142.5360528270842,\n",
       "   441.5,\n",
       "   16,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   385.5,\n",
       "   1212.5360528270842,\n",
       "   441.5,\n",
       "   16,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [229.03605282708412, 500.625, 268.0360528270841, 516.625, 17, '0.0153'],\n",
       "  [324.96519724049534,\n",
       "   459.0,\n",
       "   558.9651972404954,\n",
       "   478.0,\n",
       "   18,\n",
       "   'lachancea_nothofagi_Seq_4267'],\n",
       "  [296.96519724049534, 452.5, 321.96519724049534, 468.5, 18, '0.22'],\n",
       "  [1001.5360528270842,\n",
       "   440.5,\n",
       "   1072.5360528270842,\n",
       "   496.5,\n",
       "   18,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   440.5,\n",
       "   1142.5360528270842,\n",
       "   496.5,\n",
       "   18,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   440.5,\n",
       "   1212.5360528270842,\n",
       "   496.5,\n",
       "   18,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [271.0360528270841, 548.75, 310.0360528270841, 564.75, 19, '0.0814'],\n",
       "  [356.8145323440297,\n",
       "   514.0,\n",
       "   867.8145323440297,\n",
       "   533.0,\n",
       "   20,\n",
       "   'yHMPu5000034866_zygosaccharomyces_bisporus_160519_Seq_959'],\n",
       "  [321.81453234402966, 507.5, 353.81453234402966, 523.5, 20, '0.176'],\n",
       "  [1001.5360528270842,\n",
       "   495.5,\n",
       "   1072.5360528270842,\n",
       "   551.5,\n",
       "   20,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   495.5,\n",
       "   1142.5360528270842,\n",
       "   551.5,\n",
       "   20,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   495.5,\n",
       "   1212.5360528270842,\n",
       "   551.5,\n",
       "   20,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [313.0360528270841, 590.0, 351.0360528270841, 606.0, 21, '0.0117'],\n",
       "  [396.0360528270841,\n",
       "   569.0,\n",
       "   967.0360528270842,\n",
       "   588.0,\n",
       "   22,\n",
       "   'yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519_Seq_4055'],\n",
       "  [354.0360528270841, 562.5, 393.0360528270841, 578.5, 22, '0.0912'],\n",
       "  [1001.5360528270842,\n",
       "   550.5,\n",
       "   1072.5360528270842,\n",
       "   606.5,\n",
       "   22,\n",
       "   {'text': '1', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   550.5,\n",
       "   1142.5360528270842,\n",
       "   606.5,\n",
       "   22,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   550.5,\n",
       "   1212.5360528270842,\n",
       "   606.5,\n",
       "   22,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [399.75446656340176,\n",
       "   624.0,\n",
       "   675.7544665634018,\n",
       "   643.0,\n",
       "   23,\n",
       "   'zygosaccharomyces_bailii_Seq_2609'],\n",
       "  [364.75446656340176, 617.5, 396.75446656340176, 633.5, 23, '0.185'],\n",
       "  [1001.5360528270842,\n",
       "   605.5,\n",
       "   1072.5360528270842,\n",
       "   661.5,\n",
       "   23,\n",
       "   {'text': '5', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   605.5,\n",
       "   1142.5360528270842,\n",
       "   661.5,\n",
       "   23,\n",
       "   {'text': '40%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   605.5,\n",
       "   1212.5360528270842,\n",
       "   661.5,\n",
       "   23,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [187.03605282708412,\n",
       "   808.06640625,\n",
       "   226.03605282708412,\n",
       "   824.06640625,\n",
       "   24,\n",
       "   '0.0299'],\n",
       "  [300.70215217102947,\n",
       "   679.0,\n",
       "   781.7021521710294,\n",
       "   698.0,\n",
       "   25,\n",
       "   'yHMPu5000034862_zygotorulaspora_florentina_160519_Seq_956'],\n",
       "  [265.7021521710294, 672.5, 297.7021521710294, 688.5, 25, '0.297'],\n",
       "  [1001.5360528270842,\n",
       "   660.5,\n",
       "   1072.5360528270842,\n",
       "   716.5,\n",
       "   25,\n",
       "   {'text': '9', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   660.5,\n",
       "   1142.5360528270842,\n",
       "   716.5,\n",
       "   25,\n",
       "   {'text': '11%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   660.5,\n",
       "   1212.5360528270842,\n",
       "   716.5,\n",
       "   25,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [229.03605282708412,\n",
       "   943.6328125,\n",
       "   267.0360528270841,\n",
       "   959.6328125,\n",
       "   26,\n",
       "   '0.0115'],\n",
       "  [270.0360528270841, 755.0, 309.0360528270841, 771.0, 27, '0.0235'],\n",
       "  [349.23204893681543,\n",
       "   734.0,\n",
       "   548.2320489368154,\n",
       "   753.0,\n",
       "   28,\n",
       "   'lachancea_cidri_Seq_3705'],\n",
       "  [314.2320489368155, 727.5, 346.2320489368155, 743.5, 28, '0.148'],\n",
       "  [1001.5360528270842,\n",
       "   715.5,\n",
       "   1072.5360528270842,\n",
       "   771.5,\n",
       "   28,\n",
       "   {'text': '24', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   715.5,\n",
       "   1142.5360528270842,\n",
       "   771.5,\n",
       "   28,\n",
       "   {'text': '62%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   715.5,\n",
       "   1212.5360528270842,\n",
       "   771.5,\n",
       "   28,\n",
       "   {'text': '7%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [377.57956154391223,\n",
       "   789.0,\n",
       "   616.5795615439122,\n",
       "   808.0,\n",
       "   29,\n",
       "   'lachancea_fermentati_Seq_4096'],\n",
       "  [342.57956154391223, 782.5, 374.57956154391223, 798.5, 29, '0.271'],\n",
       "  [1001.5360528270842,\n",
       "   770.5,\n",
       "   1072.5360528270842,\n",
       "   826.5,\n",
       "   29,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   770.5,\n",
       "   1142.5360528270842,\n",
       "   826.5,\n",
       "   29,\n",
       "   {'text': '67%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   770.5,\n",
       "   1212.5360528270842,\n",
       "   826.5,\n",
       "   29,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [270.0360528270841,\n",
       "   1132.265625,\n",
       "   309.0360528270841,\n",
       "   1148.265625,\n",
       "   30,\n",
       "   '0.0364'],\n",
       "  [312.0360528270841, 926.875, 351.0360528270841, 942.875, 31, '0.0274'],\n",
       "  [354.0360528270841, 865.0, 386.0360528270841, 881.0, 32, '0.105'],\n",
       "  [431.0360528270841,\n",
       "   844.0,\n",
       "   935.0360528270842,\n",
       "   863.0,\n",
       "   33,\n",
       "   'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_3689'],\n",
       "  [389.0360528270841, 837.5, 428.0360528270841, 853.5, 33, '0.0999'],\n",
       "  [1001.5360528270842,\n",
       "   825.5,\n",
       "   1072.5360528270842,\n",
       "   881.5,\n",
       "   33,\n",
       "   {'text': '5', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   825.5,\n",
       "   1142.5360528270842,\n",
       "   881.5,\n",
       "   33,\n",
       "   {'text': '20%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   825.5,\n",
       "   1212.5360528270842,\n",
       "   881.5,\n",
       "   33,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [431.0360528270841,\n",
       "   899.0,\n",
       "   1002.0360528270842,\n",
       "   918.0,\n",
       "   34,\n",
       "   'yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519_Seq_3775'],\n",
       "  [389.0360528270841, 892.5, 428.0360528270841, 908.5, 34, '0.0984'],\n",
       "  [1001.5360528270842,\n",
       "   880.5,\n",
       "   1072.5360528270842,\n",
       "   936.5,\n",
       "   34,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   880.5,\n",
       "   1142.5360528270842,\n",
       "   936.5,\n",
       "   34,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   880.5,\n",
       "   1212.5360528270842,\n",
       "   936.5,\n",
       "   34,\n",
       "   {'text': '33%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [354.0360528270841, 988.75, 393.0360528270841, 1004.75, 35, '0.0812'],\n",
       "  [438.0360528270841,\n",
       "   954.0,\n",
       "   690.0360528270842,\n",
       "   973.0,\n",
       "   36,\n",
       "   'torulaspora_delbrueckii_Seq_4236'],\n",
       "  [396.0360528270841, 947.5, 435.0360528270841, 963.5, 36, '0.0816'],\n",
       "  [1001.5360528270842,\n",
       "   935.5,\n",
       "   1072.5360528270842,\n",
       "   991.5,\n",
       "   36,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   935.5,\n",
       "   1142.5360528270842,\n",
       "   991.5,\n",
       "   36,\n",
       "   {'text': '83%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   935.5,\n",
       "   1212.5360528270842,\n",
       "   991.5,\n",
       "   36,\n",
       "   {'text': '60%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [412.7403454576587, 1030.0, 437.7403454576587, 1046.0, 37, '0.18'],\n",
       "  [482.7403454576587,\n",
       "   1009.0,\n",
       "   955.7403454576587,\n",
       "   1028.0,\n",
       "   38,\n",
       "   'yHMPu5000034881_torulaspora_pretoriensis_160519_Seq_4597'],\n",
       "  [440.7403454576587, 1002.5, 479.7403454576587, 1018.5, 38, '0.0692'],\n",
       "  [1001.5360528270841,\n",
       "   990.5,\n",
       "   1072.536052827084,\n",
       "   1046.5,\n",
       "   38,\n",
       "   {'text': '8', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   990.5,\n",
       "   1142.536052827084,\n",
       "   1046.5,\n",
       "   38,\n",
       "   {'text': '62%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   990.5,\n",
       "   1212.536052827084,\n",
       "   1046.5,\n",
       "   38,\n",
       "   {'text': '60%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [482.7403454576587,\n",
       "   1064.0,\n",
       "   947.7403454576587,\n",
       "   1083.0,\n",
       "   39,\n",
       "   'yHMPu5000026152_torulaspora_franciscae_160519_Seq_4502'],\n",
       "  [440.7403454576587, 1057.5, 479.7403454576587, 1073.5, 39, '0.0549'],\n",
       "  [1001.5360528270841,\n",
       "   1045.5,\n",
       "   1072.536052827084,\n",
       "   1101.5,\n",
       "   39,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   1045.5,\n",
       "   1142.536052827084,\n",
       "   1101.5,\n",
       "   39,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   1045.5,\n",
       "   1212.536052827084,\n",
       "   1101.5,\n",
       "   39,\n",
       "   {'text': '67%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [312.0360528270841, 1337.65625, 351.0360528270841, 1353.65625, 40, '0.0132'],\n",
       "  [354.0360528270841, 1195.0, 393.0360528270841, 1211.0, 41, '0.0204'],\n",
       "  [396.0360528270841, 1140.0, 435.0360528270841, 1156.0, 42, '0.0598'],\n",
       "  [491.261818717455,\n",
       "   1119.0,\n",
       "   958.2618187174551,\n",
       "   1138.0,\n",
       "   43,\n",
       "   'yHMPu5000026256_zygotorulaspora_mrakii_161215_Seq_3510'],\n",
       "  [456.261818717455, 1112.5, 488.261818717455, 1128.5, 43, '0.217'],\n",
       "  [1001.5360528270842,\n",
       "   1100.5,\n",
       "   1072.5360528270842,\n",
       "   1156.5,\n",
       "   43,\n",
       "   {'text': '4', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1100.5,\n",
       "   1142.5360528270842,\n",
       "   1156.5,\n",
       "   43,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1100.5,\n",
       "   1212.5360528270842,\n",
       "   1156.5,\n",
       "   43,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [512.9406967837979,\n",
       "   1174.0,\n",
       "   952.9406967837979,\n",
       "   1193.0,\n",
       "   44,\n",
       "   'yHMPu5000035652_torulaspora_maleeae_160613_Seq_388'],\n",
       "  [478.9406967837978, 1167.5, 509.9406967837978, 1183.5, 44, '0.311'],\n",
       "  [1001.5360528270842,\n",
       "   1155.5,\n",
       "   1072.5360528270842,\n",
       "   1211.5,\n",
       "   44,\n",
       "   {'text': '6', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1155.5,\n",
       "   1142.5360528270842,\n",
       "   1211.5,\n",
       "   44,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1155.5,\n",
       "   1212.5360528270842,\n",
       "   1211.5,\n",
       "   44,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [396.0360528270841, 1250.0, 428.0360528270841, 1266.0, 45, '0.134'],\n",
       "  [484.0360528270841,\n",
       "   1229.0,\n",
       "   988.0360528270842,\n",
       "   1248.0,\n",
       "   46,\n",
       "   'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_2166'],\n",
       "  [431.0360528270841, 1222.5, 481.0360528270841, 1238.5, 46, '2.65e-06'],\n",
       "  [1001.5360528270842,\n",
       "   1210.5,\n",
       "   1072.5360528270842,\n",
       "   1266.5,\n",
       "   46,\n",
       "   {'text': '79', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1210.5,\n",
       "   1142.5360528270842,\n",
       "   1266.5,\n",
       "   46,\n",
       "   {'text': '84%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1210.5,\n",
       "   1212.5360528270842,\n",
       "   1266.5,\n",
       "   46,\n",
       "   {'text': '3%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [477.0360528270841,\n",
       "   1284.0,\n",
       "   963.0360528270842,\n",
       "   1303.0,\n",
       "   47,\n",
       "   'yHMPu5000035651_torulaspora_microellipsoides_160519_Seq_18'],\n",
       "  [431.0360528270841, 1277.5, 474.0360528270841, 1293.5, 47, '2.5e-06'],\n",
       "  [1001.5360528270842,\n",
       "   1265.5,\n",
       "   1072.5360528270842,\n",
       "   1321.5,\n",
       "   47,\n",
       "   {'text': '7', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1265.5,\n",
       "   1142.5360528270842,\n",
       "   1321.5,\n",
       "   47,\n",
       "   {'text': '57%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1265.5,\n",
       "   1212.5360528270842,\n",
       "   1321.5,\n",
       "   47,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [354.0360528270841, 1480.3125, 393.0360528270841, 1496.3125, 48, '0.0385'],\n",
       "  [396.0360528270841, 1373.75, 435.0360528270841, 1389.75, 49, '0.0146'],\n",
       "  [474.94428944425584,\n",
       "   1339.0,\n",
       "   964.9442894442559,\n",
       "   1358.0,\n",
       "   50,\n",
       "   'yHMPu5000034862_zygotorulaspora_florentina_160519_Seq_3219'],\n",
       "  [439.94428944425584, 1332.5, 471.94428944425584, 1348.5, 50, '0.147'],\n",
       "  [1001.5360528270842,\n",
       "   1320.5,\n",
       "   1072.5360528270842,\n",
       "   1376.5,\n",
       "   50,\n",
       "   {'text': '5', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1320.5,\n",
       "   1142.5360528270842,\n",
       "   1376.5,\n",
       "   50,\n",
       "   {'text': '20%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1320.5,\n",
       "   1212.5360528270842,\n",
       "   1376.5,\n",
       "   50,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [438.0360528270841, 1415.0, 477.0360528270841, 1431.0, 51, '0.0448'],\n",
       "  [522.0360528270842,\n",
       "   1394.0,\n",
       "   810.0360528270842,\n",
       "   1413.0,\n",
       "   52,\n",
       "   'saccharomyces_eubayanus_Seq_1484'],\n",
       "  [480.0360528270841, 1387.5, 519.0360528270842, 1403.5, 52, '0.0448'],\n",
       "  [1001.5360528270842,\n",
       "   1375.5,\n",
       "   1072.5360528270842,\n",
       "   1431.5,\n",
       "   52,\n",
       "   {'text': '5', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1375.5,\n",
       "   1142.5360528270842,\n",
       "   1431.5,\n",
       "   52,\n",
       "   {'text': '20%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1375.5,\n",
       "   1212.5360528270842,\n",
       "   1431.5,\n",
       "   52,\n",
       "   {'text': '0%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [522.0360528270842,\n",
       "   1449.0,\n",
       "   783.0360528270842,\n",
       "   1468.0,\n",
       "   53,\n",
       "   'saccharomyces_uvarum_Seq_2777'],\n",
       "  [480.0360528270841, 1442.5, 519.0360528270842, 1458.5, 53, '0.0423'],\n",
       "  [1001.5360528270842,\n",
       "   1430.5,\n",
       "   1072.5360528270842,\n",
       "   1486.5,\n",
       "   53,\n",
       "   {'text': '5', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1430.5,\n",
       "   1142.5360528270842,\n",
       "   1486.5,\n",
       "   53,\n",
       "   {'text': '40%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1430.5,\n",
       "   1212.5360528270842,\n",
       "   1486.5,\n",
       "   53,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [396.0360528270841, 1586.875, 435.0360528270841, 1602.875, 54, '0.0643'],\n",
       "  [438.0360528270841, 1525.0, 477.0360528270841, 1541.0, 55, '0.0181'],\n",
       "  [522.0360528270842,\n",
       "   1504.0,\n",
       "   796.0360528270842,\n",
       "   1523.0,\n",
       "   56,\n",
       "   'saccharomyces_kudriavzevii_Seq_59'],\n",
       "  [480.0360528270841, 1497.5, 519.0360528270842, 1513.5, 56, '0.0658'],\n",
       "  [1001.5360528270842,\n",
       "   1485.5,\n",
       "   1072.5360528270842,\n",
       "   1541.5,\n",
       "   56,\n",
       "   {'text': '13', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1485.5,\n",
       "   1142.5360528270842,\n",
       "   1541.5,\n",
       "   56,\n",
       "   {'text': '38%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1485.5,\n",
       "   1212.5360528270842,\n",
       "   1541.5,\n",
       "   56,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [522.0360528270842,\n",
       "   1559.0,\n",
       "   803.0360528270842,\n",
       "   1578.0,\n",
       "   57,\n",
       "   'saccharomyces_arboricola_Seq_1887'],\n",
       "  [480.0360528270841, 1552.5, 519.0360528270842, 1568.5, 57, '0.0706'],\n",
       "  [1001.5360528270842,\n",
       "   1540.5,\n",
       "   1072.5360528270842,\n",
       "   1596.5,\n",
       "   57,\n",
       "   {'text': '8', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1540.5,\n",
       "   1142.5360528270842,\n",
       "   1596.5,\n",
       "   57,\n",
       "   {'text': '50%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1540.5,\n",
       "   1212.5360528270842,\n",
       "   1596.5,\n",
       "   57,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [438.0360528270841, 1648.75, 477.0360528270841, 1664.75, 58, '0.0222'],\n",
       "  [522.0360528270842,\n",
       "   1614.0,\n",
       "   786.0360528270842,\n",
       "   1633.0,\n",
       "   59,\n",
       "   'saccharomyces_mikatae_Seq_3298'],\n",
       "  [480.0360528270841, 1607.5, 519.0360528270842, 1623.5, 59, '0.0754'],\n",
       "  [1001.5360528270842,\n",
       "   1595.5,\n",
       "   1072.5360528270842,\n",
       "   1651.5,\n",
       "   59,\n",
       "   {'text': '33', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.5360528270842,\n",
       "   1595.5,\n",
       "   1142.5360528270842,\n",
       "   1651.5,\n",
       "   59,\n",
       "   {'text': '94%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.5360528270842,\n",
       "   1595.5,\n",
       "   1212.5360528270842,\n",
       "   1651.5,\n",
       "   59,\n",
       "   {'text': '23%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [480.0360528270841, 1690.0, 519.0360528270842, 1706.0, 60, '0.0336'],\n",
       "  [564.0360528270842,\n",
       "   1669.0,\n",
       "   845.0360528270842,\n",
       "   1688.0,\n",
       "   61,\n",
       "   'saccharomyces_cerevisiae_YLL058W'],\n",
       "  [522.0360528270842, 1662.5, 561.0360528270842, 1678.5, 61, '0.0622'],\n",
       "  [1001.5360528270841,\n",
       "   1650.5,\n",
       "   1072.536052827084,\n",
       "   1706.5,\n",
       "   61,\n",
       "   {'text': '12', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   1650.5,\n",
       "   1142.536052827084,\n",
       "   1706.5,\n",
       "   61,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   1650.5,\n",
       "   1212.536052827084,\n",
       "   1706.5,\n",
       "   61,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [564.0360528270842,\n",
       "   1724.0,\n",
       "   847.0360528270842,\n",
       "   1743.0,\n",
       "   62,\n",
       "   'saccharomyces_paradoxus_Seq_3252'],\n",
       "  [522.0360528270842, 1717.5, 561.0360528270842, 1733.5, 62, '0.0367'],\n",
       "  [1001.5360528270841,\n",
       "   1705.5,\n",
       "   1072.536052827084,\n",
       "   1761.5,\n",
       "   62,\n",
       "   {'text': '8', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1071.536052827084,\n",
       "   1705.5,\n",
       "   1142.536052827084,\n",
       "   1761.5,\n",
       "   62,\n",
       "   {'text': '88%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}],\n",
       "  [1141.536052827084,\n",
       "   1705.5,\n",
       "   1212.536052827084,\n",
       "   1761.5,\n",
       "   62,\n",
       "   {'text': '100%', 'color': 'black', 'fontsize': 16, 'font': 'Verdana'}]],\n",
       " 'node_areas': {0: [1.0, 1.0, 1002.0360528270841, 1761.0],\n",
       "  1: [82.9218757801101, 1.0, 456.0948398426341, 111.0],\n",
       "  2: [147.19822256018895, 1.0, 456.0281419963951, 56.0],\n",
       "  3: [147.19822256018895, 56.0, 456.0948398426341, 111.0],\n",
       "  4: [82.9218757801101, 111.0, 1002.0360528270841, 1761.0],\n",
       "  5: [145.03605282708412, 111.0, 571.0392672625774, 386.0],\n",
       "  6: [187.03605282708412, 111.0, 538.0360528270842, 221.0],\n",
       "  7: [229.03605282708412, 111.0, 523.0360528270842, 166.0],\n",
       "  8: [229.03605282708412, 166.0, 538.0360528270842, 221.0],\n",
       "  9: [187.03605282708412, 221.0, 571.0392672625774, 386.0],\n",
       "  10: [222.03605282708412, 221.0, 488.86508661066273, 276.0],\n",
       "  11: [222.03605282708412, 276.0, 571.0392672625774, 386.0],\n",
       "  12: [264.0360528270841, 276.0, 571.0392672625774, 331.0],\n",
       "  13: [264.0360528270841, 331.0, 538.5738621878401, 386.0],\n",
       "  14: [145.03605282708412, 386.0, 1002.0360528270842, 1761.0],\n",
       "  15: [187.03605282708412, 386.0, 967.0360528270842, 661.0],\n",
       "  16: [229.03605282708412, 386.0, 493.7414208910842, 441.0],\n",
       "  17: [229.03605282708412, 441.0, 967.0360528270842, 661.0],\n",
       "  18: [271.0360528270841, 441.0, 558.9651972404954, 496.0],\n",
       "  19: [271.0360528270841, 496.0, 967.0360528270842, 661.0],\n",
       "  20: [313.0360528270841, 496.0, 867.8145323440297, 551.0],\n",
       "  21: [313.0360528270841, 551.0, 967.0360528270842, 661.0],\n",
       "  22: [354.0360528270841, 551.0, 967.0360528270842, 606.0],\n",
       "  23: [354.0360528270841, 606.0, 675.7544665634018, 661.0],\n",
       "  24: [187.03605282708412, 661.0, 1002.0360528270842, 1761.0],\n",
       "  25: [229.03605282708412, 661.0, 781.7021521710294, 716.0],\n",
       "  26: [229.03605282708412, 716.0, 1002.0360528270842, 1761.0],\n",
       "  27: [270.0360528270841, 716.0, 616.5795615439122, 826.0],\n",
       "  28: [312.0360528270841, 716.0, 548.2320489368154, 771.0],\n",
       "  29: [312.0360528270841, 771.0, 616.5795615439122, 826.0],\n",
       "  30: [270.0360528270841, 826.0, 1002.0360528270842, 1761.0],\n",
       "  31: [312.0360528270841, 826.0, 1002.0360528270842, 1101.0],\n",
       "  32: [354.0360528270841, 826.0, 1002.0360528270842, 936.0],\n",
       "  33: [389.0360528270841, 826.0, 935.0360528270842, 881.0],\n",
       "  34: [389.0360528270841, 881.0, 1002.0360528270842, 936.0],\n",
       "  35: [354.0360528270841, 936.0, 955.7403454576586, 1101.0],\n",
       "  36: [396.0360528270841, 936.0, 690.0360528270842, 991.0],\n",
       "  37: [396.0360528270841, 991.0, 955.7403454576586, 1101.0],\n",
       "  38: [440.7403454576587, 991.0, 955.7403454576587, 1046.0],\n",
       "  39: [440.7403454576587, 1046.0, 947.7403454576587, 1101.0],\n",
       "  40: [312.0360528270841, 1101.0, 988.0360528270842, 1761.0],\n",
       "  41: [354.0360528270841, 1101.0, 988.0360528270842, 1321.0],\n",
       "  42: [396.0360528270841, 1101.0, 958.2618187174551, 1211.0],\n",
       "  43: [438.0360528270841, 1101.0, 958.2618187174551, 1156.0],\n",
       "  44: [438.0360528270841, 1156.0, 952.9406967837979, 1211.0],\n",
       "  45: [396.0360528270841, 1211.0, 988.0360528270842, 1321.0],\n",
       "  46: [431.0360528270841, 1211.0, 988.0360528270842, 1266.0],\n",
       "  47: [431.0360528270841, 1266.0, 963.0360528270842, 1321.0],\n",
       "  48: [354.0360528270841, 1321.0, 964.9442894442559, 1761.0],\n",
       "  49: [396.0360528270841, 1321.0, 964.9442894442559, 1486.0],\n",
       "  50: [438.0360528270841, 1321.0, 964.9442894442559, 1376.0],\n",
       "  51: [438.0360528270841, 1376.0, 810.0360528270842, 1486.0],\n",
       "  52: [480.0360528270841, 1376.0, 810.0360528270842, 1431.0],\n",
       "  53: [480.0360528270841, 1431.0, 783.0360528270842, 1486.0],\n",
       "  54: [396.0360528270841, 1486.0, 847.0360528270842, 1761.0],\n",
       "  55: [438.0360528270841, 1486.0, 803.0360528270842, 1596.0],\n",
       "  56: [480.0360528270841, 1486.0, 796.0360528270842, 1541.0],\n",
       "  57: [480.0360528270841, 1541.0, 803.0360528270842, 1596.0],\n",
       "  58: [438.0360528270841, 1596.0, 847.0360528270842, 1761.0],\n",
       "  59: [480.0360528270841, 1596.0, 786.0360528270842, 1651.0],\n",
       "  60: [480.0360528270841, 1651.0, 847.0360528270842, 1761.0],\n",
       "  61: [522.0360528270842, 1651.0, 845.0360528270842, 1706.0],\n",
       "  62: [522.0360528270842, 1706.0, 847.0360528270842, 1761.0]}}"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#def plot_tree_proms_exp_data(goi_pair, prom_phyls, t, y1000_species_subset, proms, motif_names, branch_labels, goi_exp_data, exp_subset_goi):\n",
    "\n",
    "#makes tree ready to render for a goi pair and given promoters. \n",
    "#Includes visualization of gene expression data\n",
    "#\n",
    "#branch_labels can be \n",
    "# 'all':  puts branch length on top, bootstrap/alrt on the bottom or \n",
    "# 'bootstrap': Just puts bootstrap on top\n",
    "\n",
    "#Color Node by species: \n",
    "sacc_colors = {'KLE': \"#deb9f6\", #e4cee4\",#\"#C6AFE9\", \n",
    "               'ZT': \"YellowGreen\",\n",
    "               'Post_WGH': \"LightYellow\" #White\" # \"LightYellow\"\n",
    "              }\n",
    "\n",
    "# post_WGH_colors = {'low':  '#8cc3f6', # '#d3d3fe', #'#3192ff',#'#7eeaf7', ##2DD7ED',      #'#e6fcff', \n",
    "#                    'high': '#fcbba1'} #'#59E3EB'}  #'#ffebe6'}\n",
    "\n",
    "\n",
    "#fname_tree = os.path.normpath(y1000plus_dir + 'proteins_og/' + goi_pair + '_' + og + '/tree/run' + N_tree_dict[goi_pair][0] + '/' + goi_pair + '_' + og + '_aln_' + N_tree_dict[goi_pair][1] + 'trimmed.fasta.treefile')\n",
    "goi = 'STR2_YLL058W'\n",
    "fname_tree = y1000plus_dir + os.path.normpath(\"y1000plus_tools_data/y1000plus/proteins_og/\" + goi + '/'+ goi + '_aln_trimmed.fasta.treefile')\n",
    "t = Tree(fname_tree, format=1) \n",
    "hsu1_mrca = t.get_common_ancestor(['saccharomyces_cerevisiae_YLL058W','lachancea_fantastica_Seq_4073'])\n",
    "hsu1_mrca.ladderize()\n",
    "\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = True #False\n",
    "ts.show_branch_length=True\n",
    "\n",
    "\n",
    "\n",
    "# motif_colors = {'PDS': 'yellow', 'TATA': 'blue', 'STRE': 'red'}\n",
    "# motif_lengths = {'PDS': 3*6, 'TATA': 3*8, 'STRE': 3*5 }  #They are triple the size\n",
    "\n",
    "#box params:\n",
    "width_box = 70\n",
    "height_box = 55\n",
    "\n",
    "cmap_counts = cm.get_cmap('Greys')\n",
    "vmin = 0\n",
    "vmax = 80\n",
    "cmap_counts_norm = colors.Normalize(vmin=vmin, vmax=vmax)\n",
    "\n",
    "cmap_pct = cm.get_cmap('Reds')\n",
    "# vmin = 0\n",
    "# vmax = 57\n",
    "# cmap_counts_norm = colors.Normalize(vmin=vmin, vmax=vmax)\n",
    "\n",
    "\n",
    "# cmap_TATA = cm.get_cmap(\"Blues\")\n",
    "\n",
    "# cmap_exp = cm.get_cmap('coolwarm')\n",
    "# vmin = -7.5\n",
    "# vmax = 7.5\n",
    "# cmap_exp_norm = colors.Normalize(vmin=vmin, vmax=vmax)\n",
    "# nan_color = '#808080'\n",
    "\n",
    "# conds = list(goi_exp_data.columns)\n",
    "\n",
    "\n",
    "# # To get rid of a set of species for the visualization\n",
    "# # if less_nonsacc: \n",
    "# #     nonsacc_visualization_subset = pd.read_csv(y1000plus_dir + 'species_visualization_subset.csv')\n",
    "# #     species_subset = ( set(nonsacc_visualization_subset['original_genome_id']) | \\\n",
    "# #                        set(y1000_species[y1000_species['Major clade']=='Saccharomycetaceae']['original_genome_id']) ) # | \\\n",
    "# #                        #set(y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'])\n",
    "# #                      #)\n",
    "# #     y1000_species_subset = y1000_species[(y1000_species['original_genome_id'].isin(species_subset))]\n",
    "# #     node_subset = []\n",
    "# #     #For each node in the tree:\n",
    "# #     for node in t.iter_leaves():  \n",
    "# #         #Get the promoter sequence with motif info, make it into a motif list\n",
    "# #         if 'saccharomyces_cerevisiae' in node.name:\n",
    "# #             species='saccharomyces_cerevisiae'\n",
    "# #         else: \n",
    "# #             species = '_'.join(node.name.split('_')[:-2])\n",
    "# #         if species in species_subset: \n",
    "# #             node_subset.append(node.name)\n",
    "\n",
    "# #     t.prune(node_subset)\n",
    "# #     t.ladderize()\n",
    "\n",
    "# # for node in t.iter_leaves():  \n",
    "# #     print(node.name)\n",
    "\n",
    "# #For each node in the tree:\n",
    "for node in hsu1_mrca.traverse(): \n",
    "    #name_face = AttrFace(\"name\",fsize=15)\n",
    "    #node.add_face(name_face, column=0, position=\"branch-right\")     \n",
    "    if node.is_leaf():#Get the promoter sequence with motif info, make it into a motif list\n",
    "        \n",
    "        if 'saccharomyces_cerevisiae' in node.name:       \n",
    "            species='saccharomyces_cerevisiae'\n",
    "            gene_id = species + '@' + node.name.split(species +'_')[1]\n",
    "        elif 'candida_albicans' in node.name: \n",
    "            species = 'candida_albicans'\n",
    "            gene_id = species + '@' + node.name.split(species +'_')[1]\n",
    "        else: \n",
    "            species = '_'.join(node.name.split('_')[:-2])\n",
    "            gene_id = species + '@' + '_'.join(node.name.split('_')[-2:])\n",
    "\n",
    "        \n",
    "        if ('zygosaccharomyces' in node.name) | ('zygotorulaspora' in node.name) | ('torulaspora' in node.name):\n",
    "            spec_group = 'ZT'\n",
    "        elif 'lachancea' in node.name : \n",
    "            spec_group = 'KLE'\n",
    "        elif 'saccharomyces' in node.name: \n",
    "            spec_group = 'Post_WGH'\n",
    "        else :\n",
    "            raise ValueError('No spec_group determined for {}'.format(species))\n",
    "\n",
    "\n",
    "        \n",
    "        nstyle = NodeStyle()\n",
    "        node_color = sacc_colors[spec_group]\n",
    "        nstyle['bgcolor'] = node_color\n",
    "        node.set_style(nstyle)\n",
    "\n",
    "        if species=='saccharomyces_cerevisiae':\n",
    "            N_total_genes = 13\n",
    "            N_sc_orths = 13\n",
    "            N_cluster_orths = 13\n",
    "        else: \n",
    "            (sc_genes_dict, N_total_genes, N_sc_orths, N_cluster_orths ) = gene_synteny[node.name] \n",
    "        \n",
    "        #Add face for Number of genes within the bp range\n",
    "        rgb = colors.to_hex(cmap_counts(cmap_counts_norm(N_total_genes-1)))\n",
    "\n",
    "        rectFace_tot = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb, \n",
    "                            label= {\"text\": str(N_total_genes-1), \n",
    "                                    \"color\": 'black', \n",
    "                                    \"fontsize\": 16, \n",
    "                                    \"font\": 'Verdana'\n",
    "                                   }\n",
    "                           )\n",
    "        node.add_face(rectFace_tot, column = 1, position= \"aligned\")\n",
    "\n",
    "        \n",
    "        #Add face for pct of genes that have Sc orthologs\n",
    "        \n",
    "        pct_sc = (N_sc_orths-1)/(N_total_genes-1)\n",
    "        \n",
    "        rgb = colors.to_hex(cmap_pct(pct_sc))\n",
    "        \n",
    "        rectFace_pct_sc = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb, \n",
    "                            label= {\"text\": '{:0.0f}%'.format(pct_sc*100), \n",
    "                                    \"color\": 'black', \n",
    "                                    \"fontsize\": 16, \n",
    "                                    \"font\": 'Verdana'\n",
    "                                   }\n",
    "                           )\n",
    "        node.add_face(rectFace_pct_sc, column = 2, position= \"aligned\")\n",
    "        \n",
    "        \n",
    "        #Add face for pct of sc orthologs that are in the cluster       \n",
    "        \n",
    "        if N_sc_orths == 1: \n",
    "            pct_cluster = 0\n",
    "        else: \n",
    "            pct_cluster = (N_cluster_orths-1)/(N_sc_orths-1)\n",
    "        \n",
    "        rgb = colors.to_hex(cmap_pct(pct_cluster))\n",
    "        \n",
    "        rectFace_pct_cluster = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb, \n",
    "                            label= {\"text\": '{:0.0f}%'.format(pct_cluster*100), \n",
    "                                    \"color\": 'black', \n",
    "                                    \"fontsize\": 16, \n",
    "                                    \"font\": 'Verdana'\n",
    "                                   }\n",
    "                           )\n",
    "        node.add_face(rectFace_pct_cluster, column = 3, position= \"aligned\")\n",
    "        \n",
    "\n",
    "        \n",
    "        \n",
    "\n",
    "\n",
    "#hsu1_mrca.render('%%inline', tree_style=ts)\n",
    "hsu1_mrca.render(metproj_data_dir + 'hsu1_synteny_summary.svg', tree_style=ts)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#         if 'saccharomyces_cerevisiae' in node.name:       \n",
    "#             species='saccharomyces_cerevisiae'\n",
    "#             gene_id = species + '@' + node.name.split(species +'_')[1]\n",
    "#         elif 'candida_albicans' in node.name: \n",
    "#             species = 'candida_albicans'\n",
    "#             gene_id = species + '@' + node.name.split(species +'_')[1]\n",
    "#         else: \n",
    "#             species = '_'.join(node.name.split('_')[:-2])\n",
    "#             gene_id = species + '@' + '_'.join(node.name.split('_')[-2:])\n",
    "\n",
    "\n",
    "    #Color node based on species group\n",
    "\n",
    "#         spec_group = exp_subset_goi[node.name][1]\n",
    "\n",
    "#         if spec_group in {'high', 'low'}:\n",
    "#             node_color = post_WGH_colors[spec_group]\n",
    "#         else:\n",
    "#             node_color = sacc_colors[spec_group]\n",
    "\n",
    "\n",
    "#         prom_results = proms.loc[gene_id]\n",
    "\n",
    "#     #     simple_motifs = [\n",
    "#     #             # seq.start, seq.end, shape, width, height, fgcolor, bgcolor\n",
    "#     #             [10, 60, \">\", None, 10, \"black\", \"red\", None],\n",
    "#     #             [120, 150, \"<\", None, 10, \"black\", \"red\", None]\n",
    "#     #     ]\n",
    "#         motifs = []\n",
    "\n",
    "#         for motif_name in motif_names: #, 'PDS']:   #Leaving out PDS\n",
    "#             motif_len = motif_lengths[motif_name]\n",
    "#             motif_color = motif_colors[motif_name]\n",
    "#             if prom_results[motif_name + '_count'] >0:\n",
    "\n",
    "#                 for motif in prom_results[motif_name + '_full_features']:\n",
    "#                     loc = motif[0]\n",
    "#                     if loc <= L_prom:\n",
    "#                         direction = motif[1]\n",
    "#                         shape = '>'\n",
    "#                         start = L_prom-loc\n",
    "#                         stop = L_prom-loc + motif_len\n",
    "#                         if direction == 'rev':\n",
    "#                             shape = '<'\n",
    "#                             start = L_prom-loc-motif_len\n",
    "#                             stop = L_prom -loc\n",
    "#                         motifs.append([start,stop,shape,None, height, \"black\", motif_color, None])\n",
    "\n",
    "#         seqFace = SeqMotifFace(seq, motifs=motifs, seq_format=\"-\")\n",
    "#         node.add_face(seqFace, column=0, position=\"aligned\")\n",
    "\n",
    "\n",
    "          #Add face \n",
    "\n",
    "#         motif_name = 'STRE'\n",
    "#         L_STRE_count = 700\n",
    "\n",
    "#         N_STRE = 0\n",
    "#         for result in prom_results['STRE_full_features']:\n",
    "#             if result[0]<L_STRE_count:\n",
    "#                 N_STRE = N_STRE+1\n",
    "\n",
    "#         rgb = colors.to_hex(cmap_STRE(cmap_STRE_norm(N_STRE)))\n",
    "\n",
    "#         rectFace_STRE = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb, \n",
    "#                             label= {\"text\": str(N_STRE), \n",
    "#                                     \"color\": 'black', \n",
    "#                                     \"fontsize\": 20, \n",
    "#                                     \"font\": 'Verdana'\n",
    "#                                    }\n",
    "#                            )\n",
    "#         node.add_face(rectFace_STRE, column = 1, position= \"aligned\")\n",
    "\n",
    "#         #Add face for TATA box within 700bp\n",
    "#         motif_name = 'TATA'\n",
    "#         L_TATA = 300\n",
    "\n",
    "#         TATA = 0.0\n",
    "#         for result in prom_results['TATA_full_features']:\n",
    "#             if result[0]<L_TATA:\n",
    "#                 TATA = 1.0\n",
    "\n",
    "#         rgb = colors.to_hex(cmap_TATA(TATA))\n",
    "\n",
    "#         rectFace_TATA = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb)\n",
    "\n",
    "#         node.add_face(rectFace_TATA, column=2, position=\"aligned\")\n",
    "\n",
    "\n",
    "#         #add faces for expression data\n",
    "\n",
    "#         for jj, cond in enumerate(conds):\n",
    "#             data_val = goi_exp_data.loc[exp_subset_goi[node.name][0],cond]\n",
    "\n",
    "#             if np.isnan(data_val): \n",
    "#                 rgb = nan_color\n",
    "#             else: \n",
    "#                 rgb = colors.to_hex(cmap_exp(cmap_exp_norm(data_val)))\n",
    "\n",
    "#             rectFace_exp_cond = RectFace(width=width_box, height=height_box, fgcolor='black', bgcolor=rgb)\n",
    "#             node.add_face(rectFace_exp_cond, column = 3+jj, position= \"aligned\")\n",
    "\n",
    "\n",
    "#     else:  # If node is not a leaf, add the support label\n",
    "#         if branch_labels=='all': \n",
    "#             node_label = TextFace(node.name)\n",
    "#             node.add_face(node_label, column=1, position = \"branch-bottom\")\n",
    "#         elif branch_labels =='bootstrap':\n",
    "#             node_label = TextFace(node.name.split('/')[0])\n",
    "#             node.add_face(node_label, column=1, position = \"branch-top\")\n",
    "#         else: \n",
    "#             raise ValueError('invalid value for branch_labels: {}'.format(branch_labels))\n",
    "\n",
    "\n",
    "# return t, ts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'64_2337': ['YGL253W (HXK2)', 'YFR053C (HXK1)'],\n",
       " '64_2695': ['YJR130C (STR2)', 'YML082W (YML082W)', 'YLL058W (YLL058W)'],\n",
       " '64_4093': ['YFR052W (RPN12)'],\n",
       " '64_4177': ['YGL252C (RTG2)']}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check extent of scaffolds.\n",
    "\n",
    "#For the ZT Branch that has some conservation see what the other proteins (that don't group with orthologs) are similar to\n",
    "\n",
    "#Figure out why Z. rouxii ortholog didn't show up\n",
    "\n",
    "#Verify that the orthologs do map most closely to the YLL proteins. \n",
    "\n",
    "#Repeat the analysis for neighboring proteins.  \n",
    "\n",
    "#Repeat the analysis for the other Sulfur operon. \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Print out order on top of phylogenetic tree\n",
    "\n",
    "#C:\\Users\\heineib\\Documents\\GitHub\\y1000plus_tools\\data\\y1000plus_tools_data\\y1000plus\\gffutils_dbs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Find genes in S. cer related to surrounding genes\n",
    "\n",
    "goi_og_lookup_surr, og_genes_lookup_surr = y1000plus_tools.make_og_genes_lookup(surrounding_CDS_df['y1000_id'], y1000_species_subset)\n",
    "\n",
    "scer_prefix = '110'\n",
    "\n",
    "sc_genes_dict = {}\n",
    "for goi, og in goi_og_lookup_surr.items():\n",
    "    og_genes = og_genes_lookup_surr[og]\n",
    "    sc_genes = []\n",
    "    for og_gene in list(og_genes): \n",
    "        if (og_gene.split('_')[0]==scer_prefix): \n",
    "            sc_genes.append(scer_y1000_id_gene_id_lookup[og_gene])\n",
    "    sc_genes_dict[goi] = sc_genes \n",
    "    \n",
    "sc_genes_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "check_gene = '111_1158'\n",
    "for sc_goi, og in goi_og_lookup.items():\n",
    "    og_genes = og_genes_lookup[og]\n",
    "    if check_gene in og_genes:\n",
    "        print('{} is an ortholog of {}'.format(check_gene, sc_goi))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_spec_synteny(synteny_window_new_cols, focus_spec, focus_anc, genome_name_lookup, y1000_species_subset, anc_genes_ogs_flat, og_genes_lookup, bp_range):\n",
    "    #Given a species, find genes surrounding the ortholog of the goi pair.\n",
    "\n",
    "\n",
    "    #Get focus species genes that are present in focus ogs\n",
    "    spec_og_id_lookup = dict(zip( y1000_species_subset['original_genome_id'], y1000_species_subset['spec_og_id']))\n",
    "    spec_og_id = spec_og_id_lookup[focus_spec]\n",
    "    focus_ogs = anc_genes_ogs_flat[focus_anc]\n",
    "\n",
    "    og_genes = set()\n",
    "    for og in focus_ogs: \n",
    "        og_genes = og_genes | og_genes_lookup[og]\n",
    "\n",
    "    focus_spec_genes = []\n",
    "    for og_gene in list(og_genes): \n",
    "        spec_og_id_test = int(og_gene.split('_')[0])\n",
    "        if spec_og_id == spec_og_id_test:\n",
    "            focus_spec_genes.append(og_gene)\n",
    "\n",
    "    #Make full_gene lookup for focus spec\n",
    "    focus_spec_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/' + focus_spec + '.csv')\n",
    "    focus_spec_lookup = pd.read_csv(focus_spec_lookup_fname, index_col=0)\n",
    "    y1000_id_gene_full_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup['gene_full']))        \n",
    "    gene_full_y1000_id_lookup = dict(zip(focus_spec_lookup['gene_full'],focus_spec_lookup['y1000_id']))    \n",
    "    y1000_id_gene_id_lookup = dict(zip(focus_spec_lookup['y1000_id'],focus_spec_lookup.index))    \n",
    "\n",
    "    # Load GTF for given sequence: \n",
    "    gtf_dir = y1000plus_dir + \"0_332yeast_genomes/332_genome_annotations/gtf/\"\n",
    "    db_fname = gtf_dir + 'gffutils_dbs/' + focus_spec + '.db'\n",
    "\n",
    "    gtf_db = gffutils.FeatureDB(db_fname)\n",
    "\n",
    "\n",
    "    # Make a dictionary of scaffold sizes\n",
    "    genome_assembly_fname = y1000plus_dir + os.path.normpath('0_332yeast_genomes/332_genome_assemblies/' + focus_spec + '.fas')\n",
    "\n",
    "    scaffold_sizes = {}\n",
    "    for seq in SeqIO.parse(genome_assembly_fname, \"fasta\"):\n",
    "        scaffold_sizes[seq.id] = len(seq.seq)\n",
    "\n",
    "    for og_gene in focus_spec_genes: \n",
    "        #print(og_gene)\n",
    "        focus_gene_column = y1000_find_surrounding_genes(og_gene, y1000_id_gene_id_lookup, y1000_id_gene_full_lookup, gene_full_y1000_id_lookup, og_genes_lookup, gtf_db, bp_range, scaffold_sizes, anc_genes_ogs_flat, focus_spec_genes, focus_spec)\n",
    "        synteny_window_new_cols = synteny_window_new_cols.merge(focus_gene_column, left_on='Ancestor', right_index = True, how='left')\n",
    "\n",
    "    return synteny_window_new_cols\n",
    "\n",
    "\n",
    "def y1000_find_surrounding_genes(og_gene, y1000_id_gene_id_lookup, y1000_id_gene_full_lookup, gene_full_y1000_id_lookup, og_genes_lookup, gtf_db, bp_range, scaffold_sizes, anc_genes_ogs_flat, focus_spec_genes, focus_spec):\n",
    "    #for a given gene find the surrounding genes \n",
    "    \n",
    "    focus_gene_id = y1000_id_gene_id_lookup[og_gene]\n",
    "\n",
    "    gene_full = y1000_id_gene_full_lookup[og_gene]\n",
    "\n",
    "    cursor = gtf_db.execute('select * from features where attributes like \"%' + gene_full + '%\"')\n",
    "    all_features = cursor.fetchall()\n",
    "    if len(all_features) == 0:\n",
    "        print('No features found ' + gene_full + ' ' + focus_spec)  \n",
    "\n",
    "    CDS_list = [feature for feature in all_features if feature['featuretype']=='CDS']\n",
    "    if len(CDS_list)==1: \n",
    "        start = CDS_list[0]['start']\n",
    "        end = CDS_list[0]['end']\n",
    "    elif len(CDS_list)>1: \n",
    "        print('More than one CDS - introns?.  N of CDS: '+ str(len(CDS_list)))\n",
    "        start = min([CDS['start'] for CDS in CDS_list])\n",
    "        end = max([CDS['end'] for CDS in CDS_list])\n",
    "    else: \n",
    "        raise AssertionError('No CDS found' + og_gene)\n",
    "\n",
    "    CDS = CDS_list[0]\n",
    "    strand = CDS['strand']\n",
    "    scaffold = CDS['seqid']\n",
    "\n",
    "    # if strand=='+': \n",
    "    #     lower = start\n",
    "    #     higher = end\n",
    "    # elif strand == '-': \n",
    "    #     lower = end\n",
    "    #     higher = start\n",
    "\n",
    "    low = max(0,start-bp_range/2)\n",
    "    high = min(scaffold_sizes[scaffold], end + bp_range/2)\n",
    "\n",
    "    surrounding_CDS = list(gtf_db.region(region=(scaffold,low,high), completely_within=False, featuretype='CDS'))\n",
    "\n",
    "    #surrounding_CDS info: (gene_full, gene_id, strand, start, end)\n",
    "    surrounding_CDS_data = []\n",
    "    for CDS_surr in surrounding_CDS:\n",
    "        surrounding_CDS_data.append((CDS_surr.attributes['gene_id'][0],CDS_surr.id,CDS_surr.strand,CDS_surr.start,CDS_surr.end ))\n",
    "\n",
    "    #Find OG and line that each gene is in\n",
    "    anc_focus_spec_genes = OrderedDict([(anc,[]) for anc in anc_genes_ogs_flat.keys()])\n",
    "    for (gene_full, gene_id, strand, start, end) in surrounding_CDS_data:\n",
    "        y1000_id = gene_full_y1000_id_lookup[gene_full]\n",
    "        for anc,og_list in anc_genes_ogs_flat.items():\n",
    "            for og in og_list: \n",
    "                og_genes = og_genes_lookup[og]\n",
    "                if y1000_id in og_genes:  #could put in a test to make sure this only happens once\n",
    "                    anc_focus_spec_genes[anc].append(y1000_id_gene_id_lookup[y1000_id])\n",
    "                    #could make triple dashes if len(anc_focus_spec_genes[anc]) == 0:\n",
    "            #remove redundancies created by genes with introns\n",
    "            anc_focus_spec_genes[anc]=list(set(anc_focus_spec_genes[anc]))\n",
    "    #Merge into synteny window dataframe\n",
    "\n",
    "    focus_gene_column = pd.DataFrame(pd.Series(anc_focus_spec_genes), columns = [focus_spec + '_' + focus_gene_id])\n",
    "\n",
    "\n",
    "\n",
    "    return focus_gene_column"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
