{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Code from 20180201_promoter_motif_phylogeny\n",
    "\n",
    "#This is super rough - would need a lot of work to get it up to speed. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Intergenic regions for the four sensu stricu species is here: \n",
    "#http://www.saccharomycessensustricto.org/cgi-bin/s3.cgi?data=Intergenics&version=current\n",
    "\n",
    "#genolevures has the other missing species. \n",
    "#NOt to hard to get the promoter from the web interface here - \n",
    "#wish there was an API though\n",
    "#gene_search_url = 'http://gryc.inra.fr/index.php?page=locus&seqid=SAKL0B08096g'\n",
    "#have to add it to cart, check nucleotide and then download after adding 1000bases to the 5' site."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use YGOB and Trevors intergenics to make orthologous promoters ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#For a given gene: \n",
    "\n",
    "#Given a species list from the YGOB Pillars\n",
    "\n",
    "    #get set of orthologous genes\n",
    "\n",
    "#make fasta of orthologous promoters. \n",
    "#export file. \n",
    "\n",
    "\n",
    "#count no of STREs, PDS's.  Output surrounding 2 bases and locations\n",
    "#Align and get all conserved regions\n",
    "\n",
    "#visualize list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make a dataframe based on the YGOB pillars file\n",
    "\n",
    "fname_pillars = os.path.normpath(data_processing_dir + 'ortholog_files_YGOB\\YGOB_pillars_bmh.txt')\n",
    "YGOB_pillars = pd.read_table(fname_pillars, header = None)\n",
    "\n",
    "post_wgd = ['Vpol', 'Tpha', 'Tbla', 'Ndai', 'Ncas', 'Knag', 'Kafr', 'Cgla', 'Suva', 'Skud', 'Smik','Scer']\n",
    "post_wgd_p1 = [spec + '_p1' for spec in post_wgd]\n",
    "post_wgd_p2 = [spec + '_p2' for spec in post_wgd]\n",
    "post_wgd_p2.reverse()\n",
    "pre_wgd = ['Zrou', 'Tdel','Klac','Egos','Ecym','Lklu','Lthe','Lwal']\n",
    "columns = post_wgd_p1 + ['Anc'] + pre_wgd + post_wgd_p2\n",
    "\n",
    "YGOB_pillars.columns = columns\n",
    "\n",
    "\n",
    "#    0    V. polyspora Position 1\n",
    "#    1    T. phaffii Position 1\n",
    "#    2    T. blattae Position 1\n",
    "#    3    N. dairenensis Position 1\n",
    "#    4    N. castellii Position 1\n",
    "#    5    K. naganishii Position 1\n",
    "#    6    K. africana Position 1\n",
    "#    7    C. glabrata Position 1\n",
    "#    8    S. bayanus var. uvarum Position 1\n",
    "#    9    S. kudriavzevii Position 1\n",
    "#    10   S. mikatae Position 1\n",
    "#    11   S. cerevisiae Position 1\n",
    "#    12   Ancestral Gene Order\n",
    "#    13   Z. rouxii\n",
    "#    14   T. delbrueckii\n",
    "#    15   K. lactis\n",
    "#    16   E. gossypii \n",
    "#    17   E. cymbalariae\n",
    "#    18   L. kluyveri\n",
    "#    19   L. thermotolerans\n",
    "#    20   L. waltii\n",
    "#    21   S. cerevisiae Position 2\n",
    "#    22   S. mikatae Position 2\n",
    "#    23   S. kudriavzevii Position 2\n",
    "#    24   S. bayanus var. uvarum Position 2\n",
    "#    25   C. glabrata Position 2\n",
    "#    26   K. africana Position 2\n",
    "#    27   K. naganishii Position 2\n",
    "#    28   N. castellii Position 2\n",
    "#    29   N. dairenensis Position 2\n",
    "#    30   T. blattae Position 2\n",
    "#    31   T. phaffii Position 2\n",
    "#    32   V. polyspora Position 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gois_gr1_m24 = pd.read_csv(data_processing_dir + '20170817_klscpka/20180327_msn24_gis1_rph1_overlap.csv')\n",
    "genes = io_library.SC_orf_lookup_by_name(gois_gr1_m24['SC_common_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for a given list of genes:\n",
    "#Searches for exact matches of defined motifs in various sequences\n",
    "motif_dict = {'STRE':'AGGGG',\n",
    "             'Msn2_pref': 'CGGGG', \n",
    "              #'PDS1': 'TTAGGGAT',\n",
    "              #'PDS2': 'TAAGGGAT',\n",
    "              'PDS core': 'AGGGA'\n",
    "             }\n",
    "output_format = 'full'\n",
    "sequence_context = 2\n",
    "\n",
    "output_dict = {}\n",
    "gene_names = io_library.SC_common_name_lookup(genes) #list(gois_gr1_m24['SC_common_name'])\n",
    "for ind, gene in enumerate(genes): \n",
    "    gene_name = gene_names[ind]\n",
    "    print(gene_name)\n",
    "    #Lookup orthologs based on the YGOB Pillars dataframe\n",
    "    gene_pos = ''\n",
    "    for pos in ['p1','p2']: \n",
    "        Scer_pos = YGOB_pillars['Scer_' + pos]\n",
    "        Scer_pos_find = Scer_pos[Scer_pos==gene]\n",
    "        if len(Scer_pos_find)==1:\n",
    "            if gene_pos != '':\n",
    "                print('Warning: found gene in two pillars')\n",
    "            gene_pos = pos\n",
    "            gene_index = Scer_pos_find.index[0]\n",
    "        elif len(Scer_pos_find)>1: \n",
    "            print('Warning: found gene twice in same pillar')\n",
    "\n",
    "    #extract intergenic regions from trevor's file\n",
    "\n",
    "    specs = ['Kafr','Cgla','Suva', 'Skud', 'Smik','Scer']\n",
    "    ##to include pre wgd species would need to update\n",
    "    specs_pillar = [spec + '_' + gene_pos for spec in specs]\n",
    "\n",
    "    orthologs = YGOB_pillars.loc[gene_index, specs_pillar]\n",
    "\n",
    "\n",
    "    seqs = {}\n",
    "\n",
    "    for spec in specs: \n",
    "        fname_intergenics = os.path.normpath(data_processing_dir + 'promoter_phylogenies/intergenics/' + spec + '_intergenics.faa' )\n",
    "        record_iterator = SeqIO.parse(fname_intergenics, \"fasta\")\n",
    "\n",
    "        for seq in record_iterator: \n",
    "            if seq.id == orthologs[spec + '_' + gene_pos]:\n",
    "                seqs[spec] = seq\n",
    "\n",
    "    #initial dataframe has species for index, but reindex on seq id before outputting\n",
    "    seq_ids = [seq.id for seq in seqs.values()]\n",
    "    output_data_frame = pd.DataFrame(index = seq_ids)\n",
    "    output_data_frame['spec'] = seqs.keys()\n",
    "\n",
    "    for motif_name, motif in motif_dict.items():\n",
    "        L_motif = len(motif)\n",
    "        output_motif = []\n",
    "\n",
    "        for spec, seq in seqs.items(): \n",
    "\n",
    "            prom_seq = seq.seq\n",
    "            if len(prom_seq)<no_prom_thresh: \n",
    "                output_motif.append('no_promoter')\n",
    "            else: \n",
    "                output_motif_seq = io_library.exact_promoter_scan(motif, prom_seq, output_format, sequence_context)\n",
    "                output_motif.append(output_motif_seq)\n",
    "        output_data_frame[motif_name] = output_motif\n",
    "\n",
    "    output_dict[gene_name] = output_data_frame\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load my data and Msn2/4 delete data\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_output = pd.DataFrame()\n",
    "category_dict = dict(zip(gois_gr1_m24['SC_common_name'],gois_gr1_m24['category']))\n",
    "kl_sc_PKA_data_solis = pd.read_csv(data_processing_dir + os.path.normpath('20170817_klscpka/kl_sc_PKA_data_solis.csv'))\n",
    "kl_genename_dict = dict(zip(kl_sc_PKA_data_solis['SC_common_name'],kl_sc_PKA_data_solis['kl_genename']))\n",
    "\n",
    "for gene_name in gois_gr1_m24['SC_common_name']: \n",
    "       \n",
    "    new_output = output_dict[gene_name]\n",
    "    new_output['SC_common_name'] = gene_name\n",
    "    new_output['category'] = category_dict[gene_name]\n",
    "    new_output['kl_genename'] = kl_genename_dict[gene_name]\n",
    "\n",
    "    full_output = pd.concat([full_output, new_output])\n",
    "\n",
    "full_output.to_csv(data_processing_dir + 'promoter_phylogenies/gis1_rph1_STRE_PDS_hits.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_output = pd.read_csv(data_processing_dir + 'promoter_phylogenies/gis1_rph1_STRE_PDS_hits.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_output\n",
    "full_output = pd.DataFrame()\n",
    "pd.concat([full_output, new_output])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "motif_dict = {'STRE':'AGGGG',\n",
    "             'Msn2_pref': 'CGGGG', \n",
    "              'PDS1': 'TTAGGGAT',\n",
    "              'PDS2': 'TAAGGGAT',\n",
    "              'PDS core': 'AGGGA'\n",
    "             }\n",
    "output_format = 'full'\n",
    "sequence_context = 2\n",
    "no_prom_thresh = 20 #if length of intergenic region is less than this, call no promoter\n",
    "\n",
    "gene_name = 'SOL4'\n",
    "gene = 'YGR248W'\n",
    "\n",
    "#Lookup orthologs based on the YGOB Pillars dataframe\n",
    "gene_pos = ''\n",
    "for pos in ['p1','p2']: \n",
    "    Scer_pos = YGOB_pillars['Scer_' + pos]\n",
    "    Scer_pos_find = Scer_pos[Scer_pos==gene]\n",
    "    if len(Scer_pos_find)==1:\n",
    "        if gene_pos != '':\n",
    "            print('Warning: found gene in two pillars')\n",
    "        gene_pos = pos\n",
    "        gene_index = Scer_pos_find.index[0]\n",
    "    elif len(Scer_pos_find)>1: \n",
    "        print('Warning: found gene twice in same pillar')\n",
    "\n",
    "#extract intergenic regions from trevor's file\n",
    "\n",
    "specs = ['Kafr','Cgla','Suva', 'Skud', 'Smik','Scer']\n",
    "##to include pre wgd species would need to update\n",
    "specs_pillar = [spec + '_' + gene_pos for spec in specs]\n",
    "\n",
    "orthologs = YGOB_pillars.loc[gene_index, specs_pillar]\n",
    "\n",
    "\n",
    "seqs = {}\n",
    "\n",
    "for spec in specs: \n",
    "    fname_intergenics = os.path.normpath(data_processing_dir + 'promoter_phylogenies/intergenics/' + spec + '_intergenics.faa' )\n",
    "    record_iterator = SeqIO.parse(fname_intergenics, \"fasta\")\n",
    "\n",
    "    for seq in record_iterator: \n",
    "        if seq.id == orthologs[spec + '_' + gene_pos]:\n",
    "            seqs[spec] = seq\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#promoter_counts = io_library.exact_promoter_scan(gene_list, motif_dict, promoter_database, output_format, sequence_context)\n",
    "\n",
    "#initial dataframe has species for index, but reindex on seq id before outputting\n",
    "seq_ids = [seq.id for seq in seqs.values()]\n",
    "output_data_frame = pd.DataFrame(index = seq_ids)\n",
    "output_data_frame['spec'] = seqs.keys()\n",
    "\n",
    "for motif_name, motif in motif_dict.items():\n",
    "    L_motif = len(motif)\n",
    "    output_motif = []\n",
    "\n",
    "    for spec, seq in seqs.items(): \n",
    "\n",
    "        prom_seq = seq.seq\n",
    "        if len(prom_seq)<no_prom_thresh: \n",
    "            output_motif.append('no_promoter')\n",
    "        else: \n",
    "            output_motif_seq = io_library.exact_promoter_scan(motif, prom_seq, output_format, sequence_context)\n",
    "            output_motif.append(output_motif_seq)\n",
    "    output_data_frame[motif_name] = output_motif\n",
    "\n",
    "output_data_frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name = 'SOL3'\n",
    "gene = 'YHR163W'\n",
    "\n",
    "## Flag if no gene comes up\n",
    "#Lookup orthologs based on the YGOB Pillars dataframe\n",
    "gene_pos = ''\n",
    "for pos in ['p1','p2']: \n",
    "    Scer_pos = YGOB_pillars['Scer_' + pos]\n",
    "    Scer_pos_find = Scer_pos[Scer_pos==gene]\n",
    "    if len(Scer_pos_find)==1:\n",
    "        if gene_pos != '':\n",
    "            print('Warning: found gene in two pillars')\n",
    "        gene_pos = pos\n",
    "        gene_index = Scer_pos_find.index[0]\n",
    "    elif len(Scer_pos_find)>1: \n",
    "        print('Warning: found gene twice in same pillar')\n",
    "\n",
    "#extract intergenic regions from trevor's file\n",
    "\n",
    "specs = ['Kafr','Cgla','Suva', 'Skud', 'Smik','Scer']\n",
    "##to include pre wgd species would need to update\n",
    "specs_pillar = [spec + '_' + gene_pos for spec in specs]\n",
    "\n",
    "orthologs = YGOB_pillars.loc[gene_index, specs_pillar]\n",
    "\n",
    "\n",
    "seqs = {}\n",
    "\n",
    "for spec in specs: \n",
    "    fname_intergenics = os.path.normpath(data_processing_dir + 'promoter_phylogenies/intergenics/' + spec + '_intergenics.faa' )\n",
    "    record_iterator = SeqIO.parse(fname_intergenics, \"fasta\")\n",
    "\n",
    "    for seq in record_iterator: \n",
    "        if seq.id == orthologs[spec + '_' + gene_pos]:\n",
    "            seqs[spec] = seq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "motif = 'AGGGG'\n",
    "sequence_context = 2\n",
    "output_format = 'full'\n",
    "prom_seq = seq.seq\n",
    "\n",
    "#exact_promoter_scan(motif, prom_seq, output_format='count', sequence_context=0):\n",
    "#this function is for a single sequence - same name used to be for one that searched a list of genes \n",
    "# now called exact_promoter_scan_genelist\n",
    "# \n",
    "# scans a sequence (biopython object) for a motif in both forward and backward directions \n",
    "# If output format is 'count', then just lists the number of hits. \n",
    "# If output format is 'full', gives the location, direction (relative to the gene), and\n",
    "# sequence context (number of bases on either side of the hit)output_format\n",
    "\n",
    "\n",
    "L_motif = len(motif)\n",
    "\n",
    "#fwd search\n",
    "prom_seq_fwd = str(prom_seq)\n",
    "L_prom = len(prom_seq_fwd)\n",
    "motif_sites_fwd = [m.start() for m in re.finditer(motif, prom_seq_fwd)]\n",
    "#to find overlapping motifs\n",
    "#[m.start() fr m in re.finditer('(?=' + motif + ')', prom_seq_rev)]\n",
    "\n",
    "#reverse search\n",
    "#rev search\n",
    "prom_seq_rev = str(prom_seq.reverse_complement())\n",
    "#location is in reference to the fwd seq\n",
    "motif_sites_rev = [len(prom_seq_rev) - m.start() - L_motif for m in re.finditer(motif, prom_seq_rev)]\n",
    "\n",
    "if output_format == 'count':  \n",
    "    n_motifs_gene = len(motif_sites_fwd) + len(motif_sites_rev)\n",
    "    output = n_motifs_gene\n",
    "elif output_format == 'full': \n",
    "    loc_hitseq_gene = []\n",
    "    for motif_site in motif_sites_fwd: \n",
    "        loc = L_prom-1-motif_site\n",
    "        hitseq = prom_seq_fwd[motif_site-sequence_context:motif_site+L_motif+sequence_context]  \n",
    "        hitdir = 'fwd'\n",
    "        loc_hitseq_gene.append((loc,hitdir,hitseq))\n",
    "    for motif_site in motif_sites_rev: \n",
    "        loc = L_prom - motif_site+L_motif\n",
    "        hitseq = prom_seq_rev[(L_prom-motif_site-L_motif-sequence_context):(L_prom-motif_site+sequence_context)]\n",
    "        hitdir = 'rev'\n",
    "        loc_hitseq_gene.append((loc,hitdir,hitseq))\n",
    "\n",
    "    if len(loc_hitseq_gene)==0: \n",
    "        output = None\n",
    "    else:\n",
    "        output = loc_hitseq_gene\n",
    "else: \n",
    "    print(\"choose output_format: 'count' or 'full' \")\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#suspect Smik gnd2 is actually Smik gnd1 - check intergenics\n",
    "#RGI1 overlaps with dubious orf.  \n",
    "#Smik 4.730 is missing from intergenics.  I think that intergenics is not as good as 700bp upstream  \n",
    "#YER067W overlaps with dubious orf so isn't in trevor's file.  If you take the promoter as the 1kb upstream then it there are several STREs and one PDS. \n",
    "\n",
    "\n",
    "gene_names"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
