{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus data dir is /home/heineike/genomes/y1000plus/\n",
      "Added /home/heineike/github/y1000plus_tools/ to path\n",
      "Importing y1000plus_tools.py\n",
      "home directory is \n",
      "y1000plus_tools dir is \n",
      "y1000plus data dir is \n",
      "Added /home/heineike/github/expression_broad_data/core/ to path\n",
      "importing io_library.py\n",
      "Importing io_library.  If autoreload, may need to reset base_dir and data_processing dir \n",
      "  io_library.base_dir=base_dir \n",
      " io_library.data_processing_dir = data_processing_dir\n",
      "sys.path : \n",
      "\n",
      "['/home/heineike/github/y1000plus_tools/scripts', '', '/home/lab/envs/seqanalysis/lib/python35.zip', '/home/lab/envs/seqanalysis/lib/python3.5', '/home/lab/envs/seqanalysis/lib/python3.5/plat-linux', '/home/lab/envs/seqanalysis/lib/python3.5/lib-dynload', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/argh-0.26.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/glob2-0.4.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/mmtf_python-1.0.2-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/IPython/extensions', '/home/heineike/.ipython', '/home/heineike/github/y1000plus_tools/', '/home/heineike/github/expression_broad_data/core/']\n"
     ]
    }
   ],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "#not including outgroup species\n",
    "#y1000_species_subset = y1000_species\n",
    "y1000_species_subset = y1000_species[y1000_species['Major clade']!='outgroup']\n",
    "\n",
    "# #Select only clades of interest: \n",
    "# clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "# #clades_of_interest = {'Saccharomycetaceae'}\n",
    "# y1000_species_subset = y1000_species[y1000_species['Major clade'].isin(clades_of_interest)]\n",
    "\n",
    "#outgroup = 'Lipomyces starkeyi'\n",
    "#outgroup_orig_genome = y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'].values[0]\n",
    "#'Yarrowia lipolytica'\n",
    "# #outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "# y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "gois = ['YKR054C']\n",
    "\n",
    "#Make promoter files for various genes of interest\n",
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "# target_y1000_id = scer_lookup.loc[goi,'y1000_id']\n",
    "\n",
    "y1000_ids = dict(scer_lookup.loc[gois,'y1000_id'])\n",
    "y1000_id_list = [y1000_ids[goi] for goi in gois]\n",
    "\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'110_1706': 'OG2473'}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "goi_og_lookup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "komagataella_pastoris\n",
      "metschnikowia_similis\n",
      "saccharomyces_cerevisiae\n",
      "yHMPu5000034974_deakozyma_indianensis_160519\n",
      "yarrowia_lipolytica\n",
      "yHMPu5000034624_pichia_nakasei_160519\n",
      "zygosaccharomyces_rouxii\n",
      "babjeviella_inositovora\n",
      "yHAB159_kazachstania_solicola_160519\n",
      "yHMPu5000034632_candida_athensensis_160519\n",
      "yHMPu5000041713_debaryomyces_maramus_160928\n",
      "candida_dubliniensis\n",
      "nakazawaea_peltata\n",
      "spathaspora_girioi\n",
      "yHMPu5000041862_candida_golubevii_160928\n",
      "metschnikowia_arizonensis\n",
      "metschnikowia_bicuspidata\n",
      "yHMPu5000034612_saturnispora_saitoi_160519\n",
      "metschnikowia_continentalis\n",
      "yHMPu5000034748_lipomyces_oligophaga_160519\n",
      "yHMPu5000034660_diddensiella_caesifluorescens_160519\n",
      "yHMPu5000034988_candida_fructus_160519\n",
      "yHAB166_kazachstania_yakushimaensis_160519\n",
      "yHMPu5000035047_barnettozyma_pratensis_160519\n",
      "lachancea_fermentati\n",
      "yHMPu5000035673_candida_orba_160613\n",
      "lodderomyces_elongisporus\n",
      "candida_infanticola\n",
      "yHMPu5000034892_ogataea_pilisensis_160519\n",
      "yHMPu5000034673_blastobotrys_nivea_160519\n",
      "metschnikowia_santaceciliae\n",
      "yHMPu5000034661_dipodascus_albidus_160519\n",
      "lipomyces_starkeyi\n",
      "scheffersomyces_lignosus\n",
      "candida_tenuis\n",
      "yHMPu5000034655_botryozyma_nematodophila_160519\n",
      "nakaseomyces_nivariensis\n",
      "kluyveromyces_lactis\n",
      "yHMPu5000035691_candida_salmanticensis_160519\n",
      "yHAB132_kazachstania_martiniae_160519\n",
      "ambrosiozyma_monospora\n",
      "yHMPu5000035675_kregervanrija_fluxuum_160613\n",
      "yHMPu5000035658_starmera_amethionina_160613\n",
      "saccharomycopsis_malanga\n",
      "yHMPu5000035670_phaffomyces_opuntiae_160613\n",
      "pachysolen_tannophilus\n",
      "wickerhamomyces_anomalus\n",
      "yHMPu5000035032_candida_gatunensis_160519\n",
      "yHMPu5000041829_debaryomyces_fabryi_160928\n",
      "yHMPu5000041822_dipodascus_geniculatus_160928\n",
      "metschnikowia_mauinuiana\n",
      "yHMPu5000035037_candida_montana_160519\n",
      "candida_succiphila\n",
      "yHMPu5000034625_pichia_kudriavzevii_160519\n",
      "metschnikowia_hawaiiensis\n",
      "yHMPu5000034866_zygosaccharomyces_bisporus_160519\n",
      "yHMPu5000034614_saturnispora_silvae_160519\n",
      "yHMPu5000034876_tetrapisispora_iriomotensis_160519\n",
      "yHMPu5000035667_kurtzmaniella_cleridarum_160928\n",
      "yHMPu5000034611_saturnispora_mendoncae_160519\n",
      "wickerhamiella_domercqiae\n",
      "yHMPu5000041840_candida_wancherniae_160928\n",
      "yHMPu5000041833_candida_tammaniensis_160928\n",
      "meyerozyma_guilliermondii\n",
      "yHMPu5000034978_cyberlindnera_mrakii_160519\n",
      "yHAB143_kazachstania_siamensis_160519\n",
      "yHMPu5000035318_hyphopichia_heimii_160928\n",
      "yHMPu5000035274_wickerhamomyces_alni_160928\n",
      "yHMPu5000034757_lipomyces_doorenjongii_160519\n",
      "yHMPu5000035045_barnettozyma_hawaiiensis_160613\n",
      "yHMPu5000041818_magnusiomyces_tetrasperma_160928\n",
      "yHMPu5000034903_ogataea_naganishii_160519\n",
      "yHMPu5000035686_cyberlindnera_saturnus_160519\n",
      "yHMPu5000035044_barnettozyma_californica_160519\n",
      "saccharomyces_uvarum\n",
      "yHMPu5000035041_ambrosiozyma_pseudovanderkliftii_160519\n",
      "yHMPu5000035659_saturnispora_dispora_160519\n",
      "metschnikowia_bowlesiae\n",
      "yHMPu5000035282_trigonopsis_vinaria_160928\n",
      "debaryomyces_hansenii\n",
      "sugiyamaella_lignohabitans\n",
      "yHMPu5000035688_cyberlindnera_xylosilytica_160519\n",
      "candida_apicola\n",
      "yHMPu5000034886_ogataea_trehalophila_160519\n",
      "yHMPu5000034883_peterozyma_xylosa_160519\n",
      "candida_auris\n",
      "yHMPu5000035633_candida_hispaniensis_160519\n",
      "yHMPu5000035033_candida_cretensis_160519\n",
      "arxula_adeninivorans\n",
      "yHMPu5000035040_ambrosiozyma_maleeae_160519\n",
      "yHMPu5000034607_saccharomycopsis_capsularis_160519\n",
      "yHMPu5000034631_martiniozyma_abiesophila_161215\n",
      "saccharomyces_eubayanus\n",
      "yHMPu5000035674_kregervanrija_delftensis_160613\n",
      "yHMPu5000034760_lipomyces_kononenkoae_160519\n",
      "ogataea_methanolica\n",
      "sporopachydermia_quercuum\n",
      "yHAB133_kazachstania_unispora_160519\n",
      "yHMPu5000035243_zygoascus_meyerae_160928\n",
      "yHMPu5000034610_saturnispora_hagleri_160519\n",
      "yHMPu5000035662_meyerozyma_caribbica_160613\n",
      "metschnikowia_proteae\n",
      "yHMPu5000035242_zygoascus_ofunaensis_160928\n",
      "yHMPu5000035046_barnettozyma_populi_160613\n",
      "lachancea_cidri\n",
      "yHMPu5000035707_candida_heveicola_160519\n",
      "yHMPu5000035271_wickerhamomyces_bovis_160928\n",
      "yHAB158_kazachstania_viticola_160519\n",
      "yHMPu5000034904_ogataea_nonfermentans_160519\n",
      "metschnikowia_hibisci\n",
      "yHMPu5000035665_middelhovenomyces_tepae_160613\n",
      "geotrichum_candidum\n",
      "yHMPu5000035286_candida_azyma_160928\n",
      "yHMPu5000034709_kluyveromyces_aestuarii_160519\n",
      "yHMPu5000035679_candida_rhagii_160613\n",
      "yHAB155_kazachstania_spencerorum_160519\n",
      "yHMPu5000035018_candida_canberraensis_160519\n",
      "yHMPu5000034998_cephaloascus_albidus_160519\n",
      "yHMPu5000034627_pichia_heedii_160519\n",
      "yHMPu5000035681_candida_gotoi_160519\n",
      "candida_sojae\n",
      "yHMPu5000034758_lipomyces_japonicus_160519\n",
      "eremothecium_coryli\n",
      "yHMPu5000035651_torulaspora_microellipsoides_160519\n",
      "yHMPu5000034875_tetrapisispora_fleetii_160519\n",
      "yHMPu5000035022_candida_emberorum_160519\n",
      "clavispora_lusitaniae\n",
      "starmerella_bombicola_JCM9596\n",
      "yHMPu5000034671_blastobotrys_peoriensis_160519\n",
      "yHMPu5000034594_starmera_quercuum_160519\n",
      "yHMPu5000035643_yarrowia_bubula_160519\n",
      "kluyveromyces_marxianus\n",
      "metschnikowia_ipomoeae\n",
      "lachancea_kluyveri\n",
      "candida_intermedia\n",
      "yarrowia_keelungensis\n",
      "yHMPu5000026124_ogataea_henricii_160519\n",
      "yHMPu5000035640_yamadazyma_philogaea_160519\n",
      "yHMPu5000034897_ogataea_glucozyma_160519\n",
      "yHMPu5000041678_debaryomyces_prosopidis_160928\n",
      "metschnikowia_drakensbergensis\n",
      "yHMPu5000035252_yamadazyma_nakazawae_160928\n",
      "yHMPu5000041863_candida_gorgasii_160928\n",
      "zygosaccharomyces_bailii\n",
      "metschnikowia_matae\n",
      "kazachstania_naganishii\n",
      "yHMPu5000035639_wickerhamomyces_canadensis_160613\n",
      "metschnikowia_aberdeeniae\n",
      "yHMPu5000035645_yarrowia_divulgata_160519\n",
      "yHMPu5000034990_candida_corydali_160519\n",
      "ascoidea_asiatica\n",
      "candida_albicans\n",
      "eremothecium_gossypii\n",
      "spathaspora_passalidarum\n",
      "candida_tropicalis\n",
      "yHMPu5000034710_kluyveromyces_dobzhanskii_160519\n",
      "alloascoidea_hylecoeti\n",
      "yHMPu5000034654_aciculoconidium_aculeatum_160519\n",
      "yHMPu5000035687_cyberlindnera_suaveolens_160613\n",
      "wickerhamia_fluorescens\n",
      "yHMPu5000034932_kuraishia_ogatae_160519\n",
      "metschnikowia_shivogae\n",
      "candida_orthopsilosis\n",
      "yHMPu5000026274_komagataella_populi_160519\n",
      "yHMPu5000034881_torulaspora_pretoriensis_160519\n",
      "yHMPu5000034979_cyberlindnera_misumaiensis_160519\n",
      "yHMPu5000034636_ogataea_nitratoaversa_160519\n",
      "lachancea_nothofagi\n",
      "yHMPu5000034947_ambrosiozyma_philentoma_160519\n",
      "lachancea_dasiensis\n",
      "yHMPu5000035641_yamadazyma_scolyti_160613\n",
      "yHMPu5000034885_ogataea_zsoltii_160519\n",
      "scheffersomyces_stipitis\n",
      "eremothecium_sinecaudum\n",
      "wickerhamomyces_ciferrii\n",
      "saccharomyces_paradoxus\n",
      "millerozyma_acaciae\n",
      "yHMPu5000035279_tortispora_starmeri_160928\n",
      "brettanomyces_anomalus\n",
      "saccharomyces_arboricola\n",
      "yHMPu5000034891_ogataea_pini_160519\n",
      "priceomyces_haplophilus\n",
      "ambrosiozyma_kashinagacola\n",
      "yHMPu5000035302_candida_fragi_160928\n",
      "cyberlindnera_fabianii_JCM3601\n",
      "yHMPu5000034597_candida_stellimalicola_160519\n",
      "yHMPu5000034643_candida_schatavii_160519\n",
      "metschnikowia_lockheadii\n",
      "yHAB153_kazachstania_rosinii_160519\n",
      "yHMPu5000034742_lipomyces_suomiensis_160519\n",
      "yHMPu5000034613_saturnispora_serradocipensis_160519\n",
      "yHMPu5000026142_citeromyces_matritensis_160519\n",
      "yHMPu5000026197_brettanomyces_custersianus_160519\n",
      "metschnikowia_kamakouana\n",
      "metschnikowia_hamakuensis\n",
      "yHMPu5000034967_candida_freyschussii_160519\n",
      "yHMPu5000026137_ambrosiozyma_ambrosiae_160519\n",
      "yHMPu5000034635_nadsonia_fulvescens_160519\n",
      "ogataea_polymorpha\n",
      "candida_versatilis\n",
      "yHMPu5000034952_citeromyces_siamensis_160519\n",
      "eremothecium_cymbalariae\n",
      "yHMPu5000035335_candida_blattae_160928\n",
      "lachancea_lanzarotensis\n",
      "yHMPu5000034615_saturnispora_zaruensis_160519\n",
      "yHMPu5000041824_debaryomyces_subglobosus_160928\n",
      "yHMPu5000041693_debaryomyces_nepalensis_160928\n",
      "yHMPu5000034637_ogataea_populiabae_160519\n",
      "naumovozyma_castellii\n",
      "candida_homilentoma\n",
      "yHMPu5000035677_kodamaea_ohmeri_160519\n",
      "yHMPu5000034629_pichia_exigua_160519\n",
      "nadsonia_fulvescens_var_elongata\n",
      "kazachstania_africana\n",
      "spathaspora_hagerdaliae\n",
      "yHMPu5000034946_ambrosiozyma_oregonensis_160519\n",
      "yHMPu5000034648_candida_restingae_160519\n",
      "lachancea_thermotolerans\n",
      "yHMPu5000035048_barnettozyma_salicaria_160519\n",
      "yHMPu5000026152_torulaspora_franciscae_160519\n",
      "yHMPu5000034918_nakazawaea_holstii_160519\n",
      "saccharomyces_mikatae\n",
      "yHMPu5000035297_priceomyces_castillae_160928\n",
      "yHMPu5000034605_spencermartinsiella_europaea_160519\n",
      "saprochaete_clavata\n",
      "tetrapisispora_phaffii\n",
      "ascoidea_rubescens\n",
      "yHMPu5000035650_trigonopsis_variabilis_160613\n",
      "yHAB154_kazachstania_transvaalensis_160519\n",
      "lachancea_fantastica\n",
      "pichia_membranifaciens\n",
      "yHMPu5000034933_kuraishia_molischiana_160519\n",
      "yHMPu5000035261_candida_ponderosae_160928\n",
      "yHMPu5000035652_torulaspora_maleeae_160613\n",
      "yHMPu5000035690_candida_vartiovaarae_160613\n",
      "spathaspora_gorwiae\n",
      "candida_carpophila\n",
      "yHMPu5000034877_tetrapisispora_namnaonensis_160519\n",
      "yHMPu5000035031_candida_kruisii_160519\n",
      "yHMPu5000034939_komagataella_pseudopastoris_160519\n",
      "yarrowia_deformans\n",
      "kuraishia_capsulata\n",
      "lachancea_waltii\n",
      "metschnikowia_matae_maris\n",
      "torulaspora_delbrueckii\n",
      "yHMPu5000034950_citeromyces_hawaiiensis_160519\n",
      "yHMPu5000034999_cephaloascus_fragrans_160519\n",
      "yHMPu5000034675_blastobotrys_mokoenaii_160519\n",
      "nakaseomyces_bracarensis\n",
      "yHMPu5000035296_priceomyces_carsonii_160928\n",
      "yHMPu5000035325_cyberlindnera_petersonii_160928\n",
      "tortispora_caseinolytica\n",
      "metschnikowia_dekortum\n",
      "yHMPu5000041743_candida_hawaiiana_160928\n",
      "yHMPu5000034604_sporopachydermia_lactativora_160519\n",
      "ogataea_parapolymorpha\n",
      "candida_glabrata\n",
      "lachancea_quebecensis\n",
      "yHMPu5000035244_candida_incommunis_160928\n",
      "yHMPu5000035699_cyberlindnera_maclurae_160613\n",
      "yHMPu5000035268_wickerhamomyces_hampshirensis_160928\n",
      "metschnikowia_kipukae\n",
      "yHMPu5000034623_pichia_norvegensis_160519\n",
      "yHMPu5000034606_priceomyces_medius_160519\n",
      "yHMPu5000034667_blastobotrys_serpentis_160519\n",
      "naumovozyma_dairenensis\n",
      "metschnikowia_cerradonensis\n",
      "yHMPu5000034646_wickerhamiella_cacticola_160519\n",
      "candida_parapsilosis\n",
      "yHMPu5000034986_candida_oregonensis_160519\n",
      "yHMPu5000041855_candida_ascalaphidarum_160928\n",
      "yHMPu5000034884_peterozyma_toletana_160519\n",
      "saccharomyces_kudriavzevii\n",
      "candida_sorboxylosa\n",
      "yHMPu5000034665_kodamaea_laetipori_160519\n",
      "dekkera_bruxellensis\n",
      "yHAB164_kazachstania_aerobia_160519\n",
      "yHMPu5000035654_tortispora_ganteri_160519\n",
      "candida_arabinofermentans\n",
      "yHMPu5000035689_candida_mycetangii_160613\n",
      "yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519\n",
      "yHMPu5000034749_lipomyces_mesembrius_160519\n",
      "yHMPu5000035672_phaffomyces_thermotolerans_160613\n",
      "yHAB147_kazachstania_taianensis_160519\n",
      "metschnikowia_borealis\n",
      "spathaspora_arborariae\n",
      "yHMPu5000034901_ogataea_methylivora_160519\n",
      "tetrapisispora_blattae\n",
      "yHMPu5000034973_danielozyma_ontarioensis_160519\n",
      "yHMPu5000035703_cyberlindnera_americana_160613\n",
      "yHMPu5000026145_ambrosiozyma_vanderkliftii_160519\n",
      "yHMPu5000035011_candida_pyralidae_160519\n",
      "yHAB136_kazachstania_bromeliacearum_160519\n",
      "candida_tanzawaensis\n",
      "yHMPu5000034890_ogataea_ramenticola_160519\n",
      "yHMPu5000034669_blastobotrys_raffinofermentans_160519\n",
      "yHMPu5000034761_lipomyces_lipofer_160519\n",
      "yHMPu5000035301_pichia_terricola_160928\n",
      "yHMPu5000034887_ogataea_trehaloabstinens_160519\n",
      "yHMPu5000034670_blastobotrys_proliferans_160519\n",
      "yHMPu5000035671_phaffomyces_antillensis_160613\n",
      "yHMPu5000034899_ogataea_kodamae_160519\n",
      "yHMPu5000034681_blastobotrys_americana_160519\n",
      "candida_boidinii_JCM9604\n",
      "ashbya_aceri\n",
      "cyberlindnera_jadinii\n",
      "yHMPu5000034674_blastobotrys_muscicola_160519\n",
      "hyphopichia_burtonii\n",
      "yHMPu5000034754_lipomyces_arxii_160519\n",
      "yHMPu5000034622_pichia_occidentalis_160519\n",
      "lachancea_meyersii\n",
      "DYN1 protein sequence extraction complete\n"
     ]
    }
   ],
   "source": [
    "goi_common_name = 'DYN1'\n",
    "y1000_id = '110_1706'\n",
    "og = goi_og_lookup[y1000_id]\n",
    "og_genes = og_genes_lookup[og]\n",
    "protein_fname = goi_common_name + '_' + og\n",
    "y1000plus_tools.extract_protein_seqs(og_genes, protein_fname, y1000_species_subset)\n",
    "print(goi_common_name + ' protein sequence extraction complete')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "outgroup = 'Yarrowia lipolytica'#'Lipomyces starkeyi'\n",
    "outgroup_orig_genome = y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'].values[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mafft --genafpair --maxiterate 1000 /home/heineike/genomes/y1000plus/proteins_og/DYN1_OG2473.fasta > /home/heineike/genomes/y1000plus/proteins_og/DYN1_OG2473_aln.fasta\n",
      "iqtree -s /home/heineike/genomes/y1000plus/proteins_og/DYN1_OG2473_aln_trimmed.fasta -nt AUTO -o yarrowia_lipolytica_Seq_1861\n"
     ]
    }
   ],
   "source": [
    "# Mafft command \n",
    "\n",
    "#Make tree: \n",
    "\n",
    "#Use Mafft to construct an MSA\n",
    "\n",
    "#goi_common_name and og already set above\n",
    "\n",
    "protein_fname_base = y1000plus_dir + os.path.normpath('proteins_og/' + goi_common_name + '_' + og)\n",
    "\n",
    "\n",
    "#Command: \n",
    "mafft_command = ['mafft', '--genafpair', '--maxiterate', '1000', \n",
    "                protein_fname_base + '.fasta', \n",
    "                '>',\n",
    "                protein_fname_base + '_aln.fasta']\n",
    "\n",
    "#Running it from this notebook seems slow\n",
    "#!{\" \".join(mafft_command)}\n",
    "\n",
    "print(\" \".join(mafft_command))\n",
    "\n",
    "#This is the E-ENS-i algorithm which is suitable for conserved regions separated by long unaligned gaps. https://mafft.cbrc.jp/alignment/software/algorithms/algorithms.html\n",
    "#tree command: (designated outgroup)\n",
    "\n",
    "#\n",
    "outgroup_gene = None\n",
    "prot_seqs = SeqIO.parse(protein_fname_base + '.fasta', \"fasta\")\n",
    "for seq in prot_seqs: \n",
    "    if seq.id.split('Seq')[0]== outgroup_orig_genome+'_':\n",
    "    #if seq.id.split('Seq')[0]=='hanseniaspora_vinae_':\n",
    "        outgroup_gene = seq.id\n",
    "        \n",
    "if outgroup_gene == None: \n",
    "    print('No yarrowia_lipolytica ortholog - need to pick a different outgroup')\n",
    "\n",
    "    \n",
    "    \n",
    "#Note: for iqtree to run you need to activate the seqanalysis python environment: \n",
    "#. activate /home/lab/envs/seqanalysis\n",
    "\n",
    "iqtree_command = [\"iqtree\", \n",
    "                  \"-s\" , protein_fname_base + '_aln_trimmed.fasta',\n",
    "                  \"-nt\", \"AUTO\",  #automatically determines number of threads \n",
    "                  \"-o\", outgroup_gene]\n",
    "print(\" \".join(iqtree_command))\n",
    "\n",
    "#Current trimming protocol - open up on Jalview and trim by eye. Save as _trimmed.fasta, then find and replace\n",
    "# all \"/\" at the end of each gene name with a space. \n",
    "\n",
    "\n",
    "# #Use the MSA to build a tree\n",
    "\n",
    "# #from that tree for all post WGD species, \n",
    "#     #see if gene is closer to \"low\" or \"high\" goi.  Assign pre_wgh, low, or high label\n",
    "#     #could verify that outgroup doesn't appear between target and S.Cer low or high gene as well. \n",
    "    \n",
    "# #Order search results based on the tree"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
