{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = otherb\n",
      "home directory is /home/heineike/\n",
      "y1000plus_tools dir is /home/heineike/github/y1000plus_tools/\n",
      "y1000plus data dir is /home/heineike/genomes/y1000plus/\n",
      "Added /home/heineike/github/y1000plus_tools/ to path\n",
      "Importing y1000plus_tools.py\n",
      "home directory is \n",
      "y1000plus_tools dir is \n",
      "y1000plus data dir is \n",
      "Added /home/heineike/github/expression_broad_data/core/ to path\n",
      "importing io_library.py\n",
      "Importing io_library.  If autoreload, may need to reset base_dir and data_processing dir \n",
      "  io_library.base_dir=base_dir \n",
      " io_library.data_processing_dir = data_processing_dir\n",
      "sys.path : \n",
      "\n",
      "['/home/heineike/github/y1000plus_tools/scripts', '', '/home/lab/envs/seqanalysis/lib/python35.zip', '/home/lab/envs/seqanalysis/lib/python3.5', '/home/lab/envs/seqanalysis/lib/python3.5/plat-linux', '/home/lab/envs/seqanalysis/lib/python3.5/lib-dynload', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/argh-0.26.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/glob2-0.4.1-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/mmtf_python-1.0.2-py3.5.egg', '/home/lab/envs/seqanalysis/lib/python3.5/site-packages/IPython/extensions', '/home/heineike/.ipython', '/home/heineike/github/y1000plus_tools/', '/home/heineike/github/expression_broad_data/core/']\n"
     ]
    }
   ],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "%matplotlib inline\n",
    "#%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_new_gene_name(gene_name, spec): \n",
    "    if spec == 'Scer':\n",
    "        new_gene_name = gene_name + '_' + spec\n",
    "    elif spec == 'Calb': \n",
    "        new_gene_name = gene_name.split(',')[0] + '_' + spec\n",
    "    elif spec == 'Ylip': \n",
    "        new_gene_name = gene_name.split('p')[0] + 'g_' + spec\n",
    "    else:\n",
    "        raise(ValueError, 'Spec ' + spec + ' has no rule for making new_gene_name')\n",
    "    return new_gene_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Scer</th>\n",
       "      <th>Tpha</th>\n",
       "      <th>Ncas</th>\n",
       "      <th>Zrou</th>\n",
       "      <th>Klac</th>\n",
       "      <th>Lthe</th>\n",
       "      <th>Calb</th>\n",
       "      <th>Cdub</th>\n",
       "      <th>Ctro</th>\n",
       "      <th>Cpar</th>\n",
       "      <th>Cgui</th>\n",
       "      <th>Clus</th>\n",
       "      <th>Ylip</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SC_common_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Lys1</th>\n",
       "      <td>YIR034C_Scer</td>\n",
       "      <td>TPHA0N01860_Tpha</td>\n",
       "      <td>NCAS0A08650_Ncas</td>\n",
       "      <td>ZYRO0D00594g_Zrou</td>\n",
       "      <td>KLLA0E07987g_Klac</td>\n",
       "      <td>KLTH0C00594g_Lthe</td>\n",
       "      <td>orf19.1789.1_Calb</td>\n",
       "      <td>CD36_44980_Cdub</td>\n",
       "      <td>CTRG_03875_Ctro</td>\n",
       "      <td>CPAR2_500810_Cpar</td>\n",
       "      <td>PGUG_05365_Cgui</td>\n",
       "      <td>CLUG_04507_Clus</td>\n",
       "      <td>YALI0B15444g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys9</th>\n",
       "      <td>YNR050C_Scer</td>\n",
       "      <td>TPHA0M02120_Tpha</td>\n",
       "      <td>NCAS0A05670_Ncas</td>\n",
       "      <td>ZYRO0D17578g_Zrou</td>\n",
       "      <td>KLLA0C18744g_Klac</td>\n",
       "      <td>KLTH0A07590g_Lthe</td>\n",
       "      <td>orf19.7448_Calb</td>\n",
       "      <td>CD36_86570_Cdub</td>\n",
       "      <td>CTRG_05710_Ctro</td>\n",
       "      <td>CPAR2_808420_Cpar</td>\n",
       "      <td>PGUG_04239_Cgui</td>\n",
       "      <td>CLUG_04206_Clus</td>\n",
       "      <td>YALI0D22891g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys5</th>\n",
       "      <td>YGL154C_Scer</td>\n",
       "      <td>TPHA0C03550_Tpha</td>\n",
       "      <td>NCAS0D00740_Ncas</td>\n",
       "      <td>ZYRO0A02552g_Zrou</td>\n",
       "      <td>KLLA0E08933g_Klac</td>\n",
       "      <td>KLTH0G11066g_Lthe</td>\n",
       "      <td>orf19.6304_Calb</td>\n",
       "      <td>CD36_30280_Cdub</td>\n",
       "      <td>CTRG_00546_Ctro</td>\n",
       "      <td>CPAR2_203950_Cpar</td>\n",
       "      <td>PGUG_00079_Cgui</td>\n",
       "      <td>CLUG_03191_Clus</td>\n",
       "      <td>YALI0E06457g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys2</th>\n",
       "      <td>YBR115C_Scer</td>\n",
       "      <td>TPHA0A03520_Tpha</td>\n",
       "      <td>NCAS0I01040_Ncas</td>\n",
       "      <td>ZYRO0C16566g_Zrou</td>\n",
       "      <td>KLLA0B09218g_Klac</td>\n",
       "      <td>KLTH0F10384g_Lthe</td>\n",
       "      <td>orf19.2970_Calb</td>\n",
       "      <td>CD36_02620_Cdub</td>\n",
       "      <td>CTRG_04682_Ctro</td>\n",
       "      <td>CPAR2_108400_Cpar</td>\n",
       "      <td>PGUG_04759_Cgui</td>\n",
       "      <td>CLUG_04446_Clus</td>\n",
       "      <td>YALI0E09306g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys12</th>\n",
       "      <td>YIL094C_Scer</td>\n",
       "      <td>TPHA0C03410_Tpha</td>\n",
       "      <td>NCAS0B06250_Ncas</td>\n",
       "      <td>ZYRO0A01914g_Zrou</td>\n",
       "      <td>KLLA0E08339g_Klac</td>\n",
       "      <td>KLTH0G10384g_Lthe</td>\n",
       "      <td>orf19.2525_Calb</td>\n",
       "      <td>CD36_26410_Cdub</td>\n",
       "      <td>CTRG_00945_Ctro</td>\n",
       "      <td>CPAR2_802940_Cpar</td>\n",
       "      <td>PGUG_03876_Cgui</td>\n",
       "      <td>NONE</td>\n",
       "      <td>YALI0D10593g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys4</th>\n",
       "      <td>YDR234W_Scer</td>\n",
       "      <td>TPHA0L01150_Tpha</td>\n",
       "      <td>NCAS0B02660_Ncas</td>\n",
       "      <td>ZYRO0A10076g_Zrou</td>\n",
       "      <td>KLLA0C15125g_Klac</td>\n",
       "      <td>KLTH0E10582g_Lthe</td>\n",
       "      <td>orf19.3846_Calb</td>\n",
       "      <td>CD36_44080_Cdub</td>\n",
       "      <td>CTRG_03911_Ctro</td>\n",
       "      <td>CPAR2_500690_Cpar</td>\n",
       "      <td>PGUG_01420_Cgui</td>\n",
       "      <td>CLUG_00676_Clus</td>\n",
       "      <td>YALI0E02728g_Ylip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys20</th>\n",
       "      <td>YDL182W_Scer</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NCAS0E03640_Ncas</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys21</th>\n",
       "      <td>YDL131W_Scer</td>\n",
       "      <td>TPHA0P00840_Tpha</td>\n",
       "      <td>NCAS0A13800_Ncas</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lys20/21</th>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>ZYRO0C09416g_Zrou</td>\n",
       "      <td>KLLA0F05489g_Klac</td>\n",
       "      <td>KLTH0H02486g_Lthe</td>\n",
       "      <td>orf19.7188_Calb</td>\n",
       "      <td>CD36_73600_Cdub</td>\n",
       "      <td>CTRG_05192_Ctro</td>\n",
       "      <td>CPAR2_702440_Cpar</td>\n",
       "      <td>PGUG_04057_Cgui</td>\n",
       "      <td>CLUG_05370_Clus</td>\n",
       "      <td>YALI0F31075g_Ylip</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Scer              Tpha              Ncas  \\\n",
       "SC_common_name                                                     \n",
       "Lys1            YIR034C_Scer  TPHA0N01860_Tpha  NCAS0A08650_Ncas   \n",
       "Lys9            YNR050C_Scer  TPHA0M02120_Tpha  NCAS0A05670_Ncas   \n",
       "Lys5            YGL154C_Scer  TPHA0C03550_Tpha  NCAS0D00740_Ncas   \n",
       "Lys2            YBR115C_Scer  TPHA0A03520_Tpha  NCAS0I01040_Ncas   \n",
       "Lys12           YIL094C_Scer  TPHA0C03410_Tpha  NCAS0B06250_Ncas   \n",
       "Lys4            YDR234W_Scer  TPHA0L01150_Tpha  NCAS0B02660_Ncas   \n",
       "Lys20           YDL182W_Scer              NONE  NCAS0E03640_Ncas   \n",
       "Lys21           YDL131W_Scer  TPHA0P00840_Tpha  NCAS0A13800_Ncas   \n",
       "Lys20/21                NONE              NONE              NONE   \n",
       "\n",
       "                             Zrou               Klac               Lthe  \\\n",
       "SC_common_name                                                            \n",
       "Lys1            ZYRO0D00594g_Zrou  KLLA0E07987g_Klac  KLTH0C00594g_Lthe   \n",
       "Lys9            ZYRO0D17578g_Zrou  KLLA0C18744g_Klac  KLTH0A07590g_Lthe   \n",
       "Lys5            ZYRO0A02552g_Zrou  KLLA0E08933g_Klac  KLTH0G11066g_Lthe   \n",
       "Lys2            ZYRO0C16566g_Zrou  KLLA0B09218g_Klac  KLTH0F10384g_Lthe   \n",
       "Lys12           ZYRO0A01914g_Zrou  KLLA0E08339g_Klac  KLTH0G10384g_Lthe   \n",
       "Lys4            ZYRO0A10076g_Zrou  KLLA0C15125g_Klac  KLTH0E10582g_Lthe   \n",
       "Lys20                        NONE               NONE               NONE   \n",
       "Lys21                        NONE               NONE               NONE   \n",
       "Lys20/21        ZYRO0C09416g_Zrou  KLLA0F05489g_Klac  KLTH0H02486g_Lthe   \n",
       "\n",
       "                             Calb             Cdub             Ctro  \\\n",
       "SC_common_name                                                        \n",
       "Lys1            orf19.1789.1_Calb  CD36_44980_Cdub  CTRG_03875_Ctro   \n",
       "Lys9              orf19.7448_Calb  CD36_86570_Cdub  CTRG_05710_Ctro   \n",
       "Lys5              orf19.6304_Calb  CD36_30280_Cdub  CTRG_00546_Ctro   \n",
       "Lys2              orf19.2970_Calb  CD36_02620_Cdub  CTRG_04682_Ctro   \n",
       "Lys12             orf19.2525_Calb  CD36_26410_Cdub  CTRG_00945_Ctro   \n",
       "Lys4              orf19.3846_Calb  CD36_44080_Cdub  CTRG_03911_Ctro   \n",
       "Lys20                        NONE             NONE             NONE   \n",
       "Lys21                        NONE             NONE             NONE   \n",
       "Lys20/21          orf19.7188_Calb  CD36_73600_Cdub  CTRG_05192_Ctro   \n",
       "\n",
       "                             Cpar             Cgui             Clus  \\\n",
       "SC_common_name                                                        \n",
       "Lys1            CPAR2_500810_Cpar  PGUG_05365_Cgui  CLUG_04507_Clus   \n",
       "Lys9            CPAR2_808420_Cpar  PGUG_04239_Cgui  CLUG_04206_Clus   \n",
       "Lys5            CPAR2_203950_Cpar  PGUG_00079_Cgui  CLUG_03191_Clus   \n",
       "Lys2            CPAR2_108400_Cpar  PGUG_04759_Cgui  CLUG_04446_Clus   \n",
       "Lys12           CPAR2_802940_Cpar  PGUG_03876_Cgui             NONE   \n",
       "Lys4            CPAR2_500690_Cpar  PGUG_01420_Cgui  CLUG_00676_Clus   \n",
       "Lys20                        NONE             NONE             NONE   \n",
       "Lys21                        NONE             NONE             NONE   \n",
       "Lys20/21        CPAR2_702440_Cpar  PGUG_04057_Cgui  CLUG_05370_Clus   \n",
       "\n",
       "                             Ylip  \n",
       "SC_common_name                     \n",
       "Lys1            YALI0B15444g_Ylip  \n",
       "Lys9            YALI0D22891g_Ylip  \n",
       "Lys5            YALI0E06457g_Ylip  \n",
       "Lys2            YALI0E09306g_Ylip  \n",
       "Lys12           YALI0D10593g_Ylip  \n",
       "Lys4            YALI0E02728g_Ylip  \n",
       "Lys20                        NONE  \n",
       "Lys21                        NONE  \n",
       "Lys20/21        YALI0F31075g_Ylip  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Identify all Lys genes \n",
    "lys_gene_table = pd.read_table(base_dir + os.path.normpath('data/gcn4_liron/Lys_orth_names.tsv'), index_col=0 )\n",
    "lys_gene_table_raw = lys_gene_table.copy()\n",
    "\n",
    "\n",
    "specs = ['Scer','Calb','Ylip']\n",
    "for spec in specs: \n",
    "    gene_name_list = []\n",
    "    for gene_name in lys_gene_table[spec]:\n",
    "        if gene_name != \"NONE\":\n",
    "            new_gene_name = get_new_gene_name(gene_name,spec)\n",
    "        else:\n",
    "            new_gene_name = gene_name\n",
    "        gene_name_list.append(new_gene_name)\n",
    "    lys_gene_table[spec] = gene_name_list\n",
    "            \n",
    "lys_gene_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Scer', 'Tpha', 'Ncas', 'Zrou', 'Klac', 'Lthe', 'Calb', 'Cdub', 'Ctro',\n",
       "       'Cpar', 'Cgui', 'Clus', 'Ylip'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Extract just 700 base pairs\n",
    "N = 700\n",
    "\n",
    "#YGOB: 'Tpha','Ncas','Zrou','Lthe' also could do 'Scer', 'Klac'\n",
    "#Mine: 'Scer', 'Klac'\n",
    "#spec = 'Scer', 'Klac'\n",
    "#Calb data sources: 'Calb', 'Cdub','Ctro', 'Cpar','Cgui', 'Clus'\n",
    "#'Ylip'?? Where do we get that\n",
    "\n",
    "#for spec in lys_gene_table.columns: \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Extract Intergenic regions.  From Trevor's Data\n",
    "Scer, \n",
    "Tpha,\n",
    "Ncas,\n",
    "Zrou,\n",
    "Lthe,\n",
    "Klac,\n",
    "Calb, \n",
    "Cdub,\n",
    "Ctro,\n",
    "Cpar, \n",
    "Cgui,\n",
    "Clus,\n",
    "Ylip\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'YBR115C': 'Lys2',\n",
       " 'YDL131W': 'Lys21',\n",
       " 'YDL182W': 'Lys20',\n",
       " 'YDR234W': 'Lys4',\n",
       " 'YGL154C': 'Lys5',\n",
       " 'YIL094C': 'Lys12',\n",
       " 'YIR034C': 'Lys1',\n",
       " 'YNR050C': 'Lys9'}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spec = 'Scer'\n",
    "genename_to_lys = {}\n",
    "for lys_gene, gene_spec in lys_gene_table[spec].items():\n",
    "    if gene_spec != 'NONE':\n",
    "        genename = gene_spec.split('_')[0]\n",
    "        genename_to_lys[genename]=lys_gene\n",
    "genename_to_lys\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scer\n",
      "Tpha\n",
      "Ncas\n",
      "Zrou\n",
      "Klac\n",
      "Lthe\n",
      "Calb\n",
      "Cdub\n",
      "Ctro\n",
      "Cpar\n",
      "Cgui\n",
      "Clus\n",
      "Ylip\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(array([  7.,  12.,  11.,  13.,   4.,   2.,   1.,   4.,   3.,   1.]),\n",
       " array([    0. ,   244.2,   488.4,   732.6,   976.8,  1221. ,  1465.2,\n",
       "         1709.4,  1953.6,  2197.8,  2442. ]),\n",
       " <a list of 10 Patch objects>)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXUAAAD8CAYAAACINTRsAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAADNdJREFUeJzt3W+MZfVdx/H3RyhVKIZFpg0C64AhJKQxQiYGxaApllJopCY+gFhdW5J9YpUajS7hQfuQ+qdRo6lZCxaVwANKUyKtZYNtiEmL7tItLN1S/hTbLSu7DaatmkixXx/M2WQy3d2ZuefO3t7vvF/J5J77mzP3fL/3zP3kzDn3/iZVhSSphx+adQGSpOkx1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkho5/VRu7LzzzqvFxcVTuUlJmnv79u37ZlUtrGfdUxrqi4uL7N2791RuUpLmXpJ/X++6nn6RpEYMdUlqxFCXpEYMdUlqxFCXpEYMdUlqxFCXpEYMdUlqxFCXpEZO6SdKNR8Wdz08s22/eOeNM9u21IFH6pLUiKEuSY0Y6pLUiKEuSY0Y6pLUiKEuSY0Y6pLUiKEuSY0Y6pLUiKEuSY0Y6pLUyJqhnuTuJEeSHFgx9sdJvpzkySQfT3LO5pYpSVqP9RypfxS4ftXYHuDNVfVTwFeA26dclyRpAmuGelU9BryyauyRqnptuPt54MJNqE2StEHTOKf+HuBTU3gcSdJIo+ZTT3IH8Bpw70nW2QnsBNi+ffuYzc3MrOYXd25xSRs18ZF6kh3AO4Bfq6o60XpVtbuqlqpqaWFhYdLNSZLWYaIj9STXA38I/EJV/c90S5IkTWo9b2m8D/gccFmSQ0luBf4SOBvYk2R/kr/e5DolSeuw5pF6Vd1ynOG7NqEWSdJIfqJUkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhoZNZ+6Ntes5nGXNL88UpekRgx1SWrEUJekRgx1SWrEUJekRgx1SWrEUJekRgx1SWrEUJekRgx1SWrEUJekRtYM9SR3JzmS5MCKsXOT7Eny7HC7bXPLlCStx3qO1D8KXL9qbBfwaFVdCjw63JckzdiaoV5VjwGvrBq+CbhnWL4HeOeU65IkTWDSc+pvqqrDAMPtG6dXkiRpUpt+oTTJziR7k+w9evToZm9Okra0SUP95STnAwy3R060YlXtrqqlqlpaWFiYcHOSpPWYNNQfAnYMyzuAT0ynHEnSGOt5S+N9wOeAy5IcSnIrcCfw1iTPAm8d7kuSZmzN/1FaVbec4FvXTrkWSdJIfqJUkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhox1CWpEUNdkhoZFepJfjfJ00kOJLkvyQ9PqzBJ0sZNHOpJLgB+B1iqqjcDpwE3T6swSdLGjT39cjrwI0lOB84EXhpfkiRpUhOHelV9A/gT4GvAYeBbVfXI6vWS7EyyN8neo0ePTl6pJGlNY06/bANuAi4Gfhw4K8m7Vq9XVburaqmqlhYWFiavVJK0pjGnX34J+GpVHa2q7wIPAj83nbIkSZMYE+pfA65KcmaSANcCB6dTliRpEmPOqT8OPAA8ATw1PNbuKdUlSZrA6WN+uKreD7x/SrVIkkbyE6WS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNjJom4FRa3PXwrEuQpB94HqlLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1MirUk5yT5IEkX05yMMnPTqswSdLGjZ2l8c+Bf6qqX01yBnDmFGqSJE1o4lBP8qPANcBvAlTVq8Cr0ylLkjSJMadfLgGOAn+b5AtJPpLkrNUrJdmZZG+SvUePHh2xOUnSWsaE+unAlcCHq+oK4L+BXatXqqrdVbVUVUsLCwsjNidJWsuYUD8EHKqqx4f7D7Ac8pKkGZk41KvqP4CvJ7lsGLoW+NJUqpIkTWTsu19+G7h3eOfLC8C7x5ckSZrUqFCvqv3A0pRqkSSN5CdKJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGjHUJakRQ12SGhkd6klOS/KFJP84jYIkSZObxpH6bcDBKTyOJGmkUaGe5ELgRuAj0ylHkjTG2CP1PwP+APjeFGqRJI10+qQ/mOQdwJGq2pfkF0+y3k5gJ8D27dsn3Zy2iMVdD89kuy/eeeNMtjtLs3quYWs+36fKmCP1q4FfTvIicD/wliT/sHqlqtpdVUtVtbSwsDBic5KktUwc6lV1e1VdWFWLwM3AP1fVu6ZWmSRpw3yfuiQ1MvE59ZWq6rPAZ6fxWJKkyXmkLkmNGOqS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNGOqS1IihLkmNTGXuF0naCOfN3zweqUtSI4a6JDViqEtSI4a6JDViqEtSI4a6JDViqEtSI4a6JDViqEtSI4a6JDViqEtSIxOHepKLknwmycEkTye5bZqFSZI2bsyEXq8Bv1dVTyQ5G9iXZE9VfWlKtUmSNmjiI/WqOlxVTwzL3wEOAhdMqzBJ0sZN5Zx6kkXgCuDxaTyeJGkyo+dTT/IG4GPA+6rq28f5/k5gJ8D27dvHbk7aFLOa3xu2xhzfOnVGHakneR3LgX5vVT14vHWqandVLVXV0sLCwpjNSZLWMObdLwHuAg5W1YemV5IkaVJjjtSvBn4deEuS/cPXDVOqS5I0gYnPqVfVvwCZYi2SpJH8RKkkNWKoS1IjhrokNWKoS1IjhrokNWKoS1IjhrokNWKoS1IjhrokNWKoS1IjhrokNTJ6PnVJ48xyLvetZivMm++RuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1YqhLUiOGuiQ1MirUk1yf5JkkzyXZNa2iJEmTmTjUk5wG/BXwduBy4JYkl0+rMEnSxo05Uv8Z4LmqeqGqXgXuB26aTlmSpEmMCfULgK+vuH9oGJMkzciY+dRznLH6vpWSncDO4e5/JXlmwu2dB3xzwp+dZ1u1b9i6vdt3Q/ngSb+9Vu8/sd7tjAn1Q8BFK+5fCLy0eqWq2g3sHrEdAJLsraqlsY8zb7Zq37B1e7fvrWeavY85/fJvwKVJLk5yBnAz8NA0ipIkTWbiI/Wqei3Je4FPA6cBd1fV01OrTJK0YaP+R2lVfRL45JRqWcvoUzhzaqv2DVu3d/veeqbWe6q+79qmJGlOOU2AJDUyF6HefTqCJC8meSrJ/iR7h7Fzk+xJ8uxwu20YT5K/GJ6LJ5NcOdvq1y/J3UmOJDmwYmzDfSbZMaz/bJIds+hlo07Q+weSfGPY7/uT3LDie7cPvT+T5G0rxufqtZDkoiSfSXIwydNJbhvGW+/3k/S9+fu8qn6gv1i+CPs8cAlwBvBF4PJZ1zXlHl8Ezls19kfArmF5F/DBYfkG4FMsf07gKuDxWde/gT6vAa4EDkzaJ3Au8MJwu21Y3jbr3ibs/QPA7x9n3cuH3/PXAxcPv/+nzeNrATgfuHJYPhv4ytBf6/1+kr43fZ/Pw5H6Vp2O4CbgnmH5HuCdK8b/rpZ9HjgnyfmzKHCjquox4JVVwxvt823Anqp6par+E9gDXL/51Y9zgt5P5Cbg/qr636r6KvAcy6+DuXstVNXhqnpiWP4OcJDlT5633u8n6ftEprbP5yHUt8J0BAU8kmTf8AlcgDdV1WFY/gUB3jiMd3s+Ntpnt/7fO5xmuPvYKQia9p5kEbgCeJwttN9X9Q2bvM/nIdTXNR3BnLu6qq5kecbL30pyzUnW3QrPB5y4z079fxj4SeCngcPAnw7j7XpP8gbgY8D7qurbJ1v1OGNz2/tx+t70fT4Pob6u6QjmWVW9NNweAT7O8p9cLx87rTLcHhlW7/Z8bLTPNv1X1ctV9X9V9T3gb1je79Cs9ySvYznY7q2qB4fh9vv9eH2fin0+D6HeejqCJGclOfvYMnAdcIDlHo9d4d8BfGJYfgj4jeFdAlcB3zr2Z+yc2mifnwauS7Jt+NP1umFs7qy6FvIrLO93WO795iSvT3IxcCnwr8zhayFJgLuAg1X1oRXfar3fT9T3Kdnns75KvM4ryTewfPX4eeCOWdcz5d4uYfmK9heBp4/1B/wY8Cjw7HB77jAelv85yfPAU8DSrHvYQK/3sfwn53dZPgK5dZI+gfewfCHpOeDds+5rRO9/P/T25PBCPX/F+ncMvT8DvH3F+Fy9FoCfZ/l0wZPA/uHrhu77/SR9b/o+9xOlktTIPJx+kSStk6EuSY0Y6pLUiKEuSY0Y6pLUiKEuSY0Y6pLUiKEuSY38P2YAN3ZK/OX3AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#From Trevor's file, extract intergenic regions for each of the proteins of interest: \n",
    "\n",
    "specs = list(lys_gene_table.columns)\n",
    "\n",
    "intergenics_dir = home_dir + os.path.normpath('genomes/intergenics') + os.sep\n",
    "fname_lys_intergenics = base_dir + os.path.normpath('data/gcn4_liron/Lys_ortholog_intergenics.fasta')\n",
    "\n",
    "prom_length = {}\n",
    "with open(fname_lys_intergenics,'w') as f: \n",
    "\n",
    "    for spec in specs: \n",
    "        print(spec)\n",
    "        #Make dictionary from gene name to lys ortholog common name in S.Cer\n",
    "        genename_to_lys = {}\n",
    "        for lys_gene, gene_spec in lys_gene_table[spec].items():\n",
    "            if gene_spec != 'NONE':\n",
    "                genename = gene_spec.split('_')[0]\n",
    "                genename_to_lys[genename]=lys_gene\n",
    "\n",
    "\n",
    "        fname_intergenics = intergenics_dir + spec + '_intergenics.faa' \n",
    "        record_iterator = SeqIO.parse(fname_intergenics, \"fasta\")\n",
    "\n",
    "\n",
    "        for seq in record_iterator: \n",
    "            if seq.id in set(genename_to_lys.keys()):\n",
    "                f.write('>' + genename_to_lys[seq.id] + '_' + seq.id + '_' + spec + '\\n')\n",
    "                f.write(str(seq.seq) + '\\n')\n",
    "                prom_length[ genename_to_lys[seq.id] + '_' + seq.id + '_' + spec] = len(seq.seq)\n",
    "\n",
    "\n",
    "\n",
    "plt.hist(prom_length.values())\n",
    "\n",
    "# #initial dataframe has species for index, but reindex on seq id before outputting\n",
    "# seq_ids = [seq.id for seq in seqs.values()]\n",
    "# output_data_frame = pd.DataFrame(index = seq_ids)\n",
    "# output_data_frame['spec'] = seqs.keys()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "272"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(seq.seq)\n",
    "#str(seq.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for a given list of genes:\n",
    "#Searches for exact matches of defined motifs in various sequences\n",
    "motif_dict = {'STRE':'AGGGG',\n",
    "             'Msn2_pref': 'CGGGG', \n",
    "              #'PDS1': 'TTAGGGAT',\n",
    "              #'PDS2': 'TAAGGGAT',\n",
    "              'PDS core': 'AGGGA'\n",
    "             }\n",
    "output_format = 'full'\n",
    "sequence_context = 2\n",
    "\n",
    "output_dict = {}\n",
    "gene_names = io_library.SC_common_name_lookup(genes) #list(gois_gr1_m24['SC_common_name'])\n",
    "for ind, gene in enumerate(genes): \n",
    "    gene_name = gene_names[ind]\n",
    "    print(gene_name)\n",
    "    #Lookup orthologs based on the YGOB Pillars dataframe\n",
    "    gene_pos = ''\n",
    "    for pos in ['p1','p2']: \n",
    "        Scer_pos = YGOB_pillars['Scer_' + pos]\n",
    "        Scer_pos_find = Scer_pos[Scer_pos==gene]\n",
    "        if len(Scer_pos_find)==1:\n",
    "            if gene_pos != '':\n",
    "                print('Warning: found gene in two pillars')\n",
    "            gene_pos = pos\n",
    "            gene_index = Scer_pos_find.index[0]\n",
    "        elif len(Scer_pos_find)>1: \n",
    "            print('Warning: found gene twice in same pillar')\n",
    "\n",
    "    #extract intergenic regions from trevor's file\n",
    "\n",
    "    specs = ['Kafr','Cgla','Suva', 'Skud', 'Smik','Scer']\n",
    "    ##to include pre wgd species would need to update\n",
    "    specs_pillar = [spec + '_' + gene_pos for spec in specs]\n",
    "\n",
    "    orthologs = YGOB_pillars.loc[gene_index, specs_pillar]\n",
    "\n",
    "\n",
    "    seqs = {}\n",
    "\n",
    "    for spec in specs: \n",
    "        fname_intergenics = os.path.normpath(data_processing_dir + 'promoter_phylogenies/intergenics/' + spec + '_intergenics.faa' )\n",
    "        record_iterator = SeqIO.parse(fname_intergenics, \"fasta\")\n",
    "\n",
    "        for seq in record_iterator: \n",
    "            if seq.id == orthologs[spec + '_' + gene_pos]:\n",
    "                seqs[spec] = seq\n",
    "\n",
    "    #initial dataframe has species for index, but reindex on seq id before outputting\n",
    "    seq_ids = [seq.id for seq in seqs.values()]\n",
    "    output_data_frame = pd.DataFrame(index = seq_ids)\n",
    "    output_data_frame['spec'] = seqs.keys()\n",
    "\n",
    "    for motif_name, motif in motif_dict.items():\n",
    "        L_motif = len(motif)\n",
    "        output_motif = []\n",
    "\n",
    "        for spec, seq in seqs.items(): \n",
    "\n",
    "            prom_seq = seq.seq\n",
    "            if len(prom_seq)<no_prom_thresh: \n",
    "                output_motif.append('no_promoter')\n",
    "            else: \n",
    "                output_motif_seq = io_library.exact_promoter_scan(motif, prom_seq, output_format, sequence_context)\n",
    "                output_motif.append(output_motif_seq)\n",
    "        output_data_frame[motif_name] = output_motif\n",
    "\n",
    "    output_dict[gene_name] = output_data_frame\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "seqanalysis",
   "language": "python",
   "name": "seqanalysis"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
