{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "what computer are you on? a = Ben's laptop, b = gpucluster, c = Ben's desktop, d = othera\n",
      "home directory is C:\\Users\\BMH_work\\\n",
      "y1000plus_tools dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\\n",
      "y1000plus data dir is C:\\Users\\BMH_work\\github\\y1000plus_tools\\genomes\\y1000plus\\\n",
      "Importing y1000plus_tools.py\n",
      "sys.path : \n",
      "\n",
      "['C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\scripts', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\python37.zip', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\DLLs', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env', '', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\win32\\\\lib', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\Pythonwin', 'C:\\\\Users\\\\BMH_work\\\\Anaconda3\\\\envs\\\\y1000_plus_env\\\\lib\\\\site-packages\\\\IPython\\\\extensions', 'C:\\\\Users\\\\BMH_work\\\\.ipython', 'C:\\\\Users\\\\BMH_work\\\\github\\\\y1000plus_tools\\\\', 'C:\\\\Users\\\\BMH_work\\\\github\\\\yeast_esr_expression_analysis\\\\']\n",
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# %load notebook_setup.py\n",
    "#Use %load notebook_setup.py to load these commands at the top of your\n",
    "#ipython notebook.  Includes running the std_libraries.py file. \n",
    "\n",
    "%run std_libraries.py\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "# %matplotlib \n",
    "# plt.ioff()\n",
    "#%matplotlib inline\n",
    "%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "y1000_species = pd.read_csv(y1000plus_dir + \"y1000_species_table.csv\", index_col=0)\n",
    "\n",
    "#Select only clades of interest: \n",
    "#clades_of_interest = {'Saccharomycetaceae','Saccharomycodaceae','Phaffomycetaceae','CUG-Ser2','CUG-Ser1','Pichiaceae','CUG-Ala'}\n",
    "#clades_of_interest = {'Saccharomycetaceae', 'Saccharomycodaceae'}\n",
    "#outgroups = [ 'Wickerhamomyces anomalus','Cyberlindnera jadinii', 'Ascoidea rubescens','Komagataella pastoris','Debaryomyces hansenii', 'Candida albicans', 'Yarrowia lipolytica']\n",
    "#outgroup = 'Yarrowia lipolytica'\n",
    "\n",
    "#y1000_species_subset = y1000_species[(y1000_species['Major clade'].isin(clades_of_interest))  |  (y1000_species['Species name'].isin(outgroups))]\n",
    "\n",
    "\n",
    "#outgroup = 'Hanseniaspora vineae'  #For EGO2/4 - only Sacch so use different outgroup\n",
    "#y1000_species_subset = pd.concat([y1000_species_subset, y1000_species[y1000_species['Species name']==outgroup]])\n",
    "\n",
    "y1000_species_subset = y1000_species[y1000_species['Major clade']!='outgroup']\n",
    "#y1000_species_subset = y1000_species\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load S.Cer lookup table: \n",
    "scer_lookup_fname = y1000plus_dir + os.path.normpath('id_lookups/saccharomyces_cerevisiae.csv')\n",
    "\n",
    "scer_lookup = pd.read_csv(scer_lookup_fname, index_col=0)\n",
    "target_y1000_id = scer_lookup.loc['YHR084W','y1000_id']\n",
    "\n",
    "y1000_id_list = [target_y1000_id]\n",
    "goi_og_lookup, og_genes_lookup = y1000plus_tools.make_og_genes_lookup(y1000_id_list, y1000_species_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yHMPu5000041818_magnusiomyces_tetrasperma_160928\n",
      "yarrowia_keelungensis\n",
      "yHMPu5000034918_nakazawaea_holstii_160519\n",
      "saccharomyces_kudriavzevii\n",
      "yHMPu5000034667_blastobotrys_serpentis_160519\n",
      "metschnikowia_bicuspidata\n",
      "yHMPu5000034875_tetrapisispora_fleetii_160519\n",
      "yHMPu5000035672_phaffomyces_thermotolerans_160613\n",
      "ascoidea_rubescens\n",
      "lachancea_quebecensis\n",
      "scheffersomyces_stipitis\n",
      "metschnikowia_ipomoeae\n",
      "yHAB143_kazachstania_siamensis_160519\n",
      "yHMPu5000034886_ogataea_trehalophila_160519\n",
      "candida_tanzawaensis\n",
      "yHMPu5000034881_torulaspora_pretoriensis_160519\n",
      "yHMPu5000034709_kluyveromyces_aestuarii_160519\n",
      "candida_boidinii_JCM9604\n",
      "eremothecium_gossypii\n",
      "yHMPu5000041862_candida_golubevii_160928\n",
      "yHMPu5000034607_saccharomycopsis_capsularis_160519\n",
      "yHMPu5000035645_yarrowia_divulgata_160519\n",
      "starmerella_bombicola_JCM9596\n",
      "eremothecium_cymbalariae\n",
      "yHMPu5000035687_cyberlindnera_suaveolens_160613\n",
      "metschnikowia_hawaiiensis\n",
      "lachancea_meyersii\n",
      "spathaspora_passalidarum\n",
      "yHMPu5000026124_ogataea_henricii_160519\n",
      "ogataea_methanolica\n",
      "yHMPu5000034632_candida_athensensis_160519\n",
      "candida_sorboxylosa\n",
      "kuraishia_capsulata\n",
      "candida_carpophila\n",
      "yHMPu5000035665_middelhovenomyces_tepae_160613\n",
      "yHMPu5000035677_kodamaea_ohmeri_160519\n",
      "yHMPu5000035011_candida_pyralidae_160519\n",
      "ambrosiozyma_monospora\n",
      "yHMPu5000035652_torulaspora_maleeae_160613\n",
      "tetrapisispora_phaffii\n",
      "yHMPu5000041678_debaryomyces_prosopidis_160928\n",
      "ogataea_parapolymorpha\n",
      "hanseniaspora_uvarum\n",
      "nakaseomyces_delphensis\n",
      "alloascoidea_hylecoeti\n",
      "yHMPu5000035671_phaffomyces_antillensis_160613\n",
      "lachancea_fermentati\n",
      "yHMPu5000035040_ambrosiozyma_maleeae_160519\n",
      "yHMPu5000034669_blastobotrys_raffinofermentans_160519\n",
      "yHMPu5000035242_zygoascus_ofunaensis_160928\n",
      "yHMPu5000035037_candida_montana_160519\n",
      "yHMPu5000034903_ogataea_naganishii_160519\n",
      "yHMPu5000034660_diddensiella_caesifluorescens_160519\n",
      "metschnikowia_aberdeeniae\n",
      "yHMPu5000035699_cyberlindnera_maclurae_160613\n",
      "candida_parapsilosis\n",
      "metschnikowia_arizonensis\n",
      "yHMPu5000035032_candida_gatunensis_160519\n",
      "yHMPu5000034999_cephaloascus_fragrans_160519\n",
      "millerozyma_acaciae\n",
      "kluyveromyces_lactis\n",
      "yHMPu5000034627_pichia_heedii_160519\n",
      "yHMPu5000035695_hanseniaspora_pseudoguilliermondii_160519\n",
      "lodderomyces_elongisporus\n",
      "yHMPu5000034979_cyberlindnera_misumaiensis_160519\n",
      "kazachstania_naganishii\n",
      "yHMPu5000035679_candida_rhagii_160613\n",
      "yHAB148_kazachstania_intestinalis_160519\n",
      "yHMPu5000035707_candida_heveicola_160519\n",
      "yHMPu5000035282_trigonopsis_vinaria_160928\n",
      "metschnikowia_similis\n",
      "arxula_adeninivorans\n",
      "yHMPu5000034654_aciculoconidium_aculeatum_160519\n",
      "saccharomyces_uvarum\n",
      "zygosaccharomyces_rouxii\n",
      "wickerhamomyces_ciferrii\n",
      "candida_tropicalis\n",
      "yHMPu5000035318_hyphopichia_heimii_160928\n",
      "yHMPu5000026145_ambrosiozyma_vanderkliftii_160519\n",
      "yHMPu5000034636_ogataea_nitratoaversa_160519\n",
      "yHMPu5000034892_ogataea_pilisensis_160519\n",
      "yHMPu5000035703_cyberlindnera_americana_160613\n",
      "yHMPu5000034890_ogataea_ramenticola_160519\n",
      "yHMPu5000026137_ambrosiozyma_ambrosiae_160519\n",
      "lachancea_nothofagi\n",
      "yHMPu5000034952_citeromyces_siamensis_160519\n",
      "geotrichum_candidum\n",
      "candida_glabrata\n",
      "metschnikowia_borealis\n",
      "yHMPu5000034891_ogataea_pini_160519\n",
      "yHMPu5000034597_candida_stellimalicola_160519\n",
      "yHMPu5000034862_zygotorulaspora_florentina_160519\n",
      "yHMPu5000035031_candida_kruisii_160519\n",
      "nakaseomyces_bracarensis\n",
      "dekkera_bruxellensis\n",
      "yHMPu5000041693_debaryomyces_nepalensis_160928\n",
      "yHMPu5000026142_citeromyces_matritensis_160519\n",
      "yHMPu5000035243_zygoascus_meyerae_160928\n",
      "scheffersomyces_lignosus\n",
      "yHMPu5000034990_candida_corydali_160519\n",
      "yHMPu5000026197_brettanomyces_custersianus_160519\n",
      "yHMPu5000035667_kurtzmaniella_cleridarum_160928\n",
      "yHMPu5000035261_candida_ponderosae_160928\n",
      "yHMPu5000034883_peterozyma_xylosa_160519\n",
      "yHMPu5000034614_saturnispora_silvae_160519\n",
      "torulaspora_delbrueckii\n",
      "yHMPu5000034675_blastobotrys_mokoenaii_160519\n",
      "yHMPu5000034893_ogataea_philodendra_160519\n",
      "saccharomyces_mikatae\n",
      "sporopachydermia_quercuum\n",
      "yHMPu5000034957_hanseniaspora_osmophila_160519\n",
      "yHMPu5000034629_pichia_exigua_160519\n",
      "kazachstania_africana\n",
      "saccharomycopsis_malanga\n",
      "yHMPu5000034876_tetrapisispora_iriomotensis_160519\n",
      "cyberlindnera_jadinii\n",
      "yHMPu5000041840_candida_wancherniae_160928\n",
      "eremothecium_coryli\n",
      "naumovozyma_castellii\n",
      "yHMPu5000034950_citeromyces_hawaiiensis_160519\n",
      "yHMPu5000035643_yarrowia_bubula_160519\n",
      "yHMPu5000035633_candida_hispaniensis_160519\n",
      "nakaseomyces_nivariensis\n",
      "yHAB132_kazachstania_martiniae_160519\n",
      "candida_sojae\n",
      "wickerhamia_fluorescens\n",
      "yHMPu5000034623_pichia_norvegensis_160519\n",
      "yHMPu5000041713_debaryomyces_maramus_160928\n",
      "yHMPu5000034946_ambrosiozyma_oregonensis_160519\n",
      "lachancea_waltii\n",
      "spathaspora_girioi\n",
      "brettanomyces_anomalus\n",
      "yHMPu5000034973_danielozyma_ontarioensis_160519\n",
      "ashbya_aceri\n",
      "spathaspora_hagerdaliae\n",
      "yHMPu5000034637_ogataea_populiabae_160519\n",
      "yHMPu5000035686_cyberlindnera_saturnus_160519\n",
      "yHMPu5000034604_sporopachydermia_lactativora_160519\n",
      "yHMPu5000034655_botryozyma_nematodophila_160519\n",
      "metschnikowia_continentalis\n",
      "naumovozyma_dairenensis\n",
      "candida_albicans\n",
      "yHMPu5000035681_candida_gotoi_160519\n",
      "yHAB133_kazachstania_unispora_160519\n",
      "lachancea_mirantina\n",
      "yHMPu5000034622_pichia_occidentalis_160519\n",
      "priceomyces_haplophilus\n",
      "yHMPu5000034902_ogataea_minuta_160519\n",
      "yHMPu5000034988_candida_fructus_160519\n",
      "nakaseomyces_bacillisporus\n",
      "eremothecium_sinecaudum\n",
      "yHMPu5000034877_tetrapisispora_namnaonensis_160519\n",
      "candida_tenuis\n",
      "candida_arabinofermentans\n",
      "nakaseomyces_castellii\n",
      "yHMPu5000035651_torulaspora_microellipsoides_160519\n",
      "yHMPu5000035041_ambrosiozyma_pseudovanderkliftii_160519\n",
      "nadsonia_fulvescens_var_elongata\n",
      "yHMPu5000035696_hanseniaspora_singularis_160519\n",
      "yHMPu5000035302_candida_fragi_160928\n",
      "yHMPu5000034671_blastobotrys_peoriensis_160519\n",
      "yHMPu5000035044_barnettozyma_californica_160519\n",
      "meyerozyma_guilliermondii\n",
      "yHMPu5000034673_blastobotrys_nivea_160519\n",
      "yHMPu5000034885_ogataea_zsoltii_160519\n",
      "yHMPu5000041824_debaryomyces_subglobosus_160928\n",
      "lachancea_cidri\n",
      "metschnikowia_dekortum\n",
      "metschnikowia_bowlesiae\n",
      "tetrapisispora_blattae\n",
      "saprochaete_clavata\n",
      "yHAB166_kazachstania_yakushimaensis_160519\n",
      "metschnikowia_kamakouana\n",
      "saccharomyces_cerevisiae\n",
      "metschnikowia_drakensbergensis\n",
      "yHMPu5000035301_pichia_terricola_160928\n",
      "yHAB159_kazachstania_solicola_160519\n",
      "metschnikowia_shivogae\n",
      "candida_auris\n",
      "yHMPu5000034605_spencermartinsiella_europaea_160519\n",
      "yHMPu5000034899_ogataea_kodamae_160519\n",
      "yHMPu5000035047_barnettozyma_pratensis_160519\n",
      "lachancea_lanzarotensis\n",
      "candida_orthopsilosis\n",
      "yHMPu5000034963_hanseniaspora_clermontiae_160519\n",
      "yHMPu5000035654_tortispora_ganteri_160519\n",
      "metschnikowia_lockheadii\n",
      "yHMPu5000034648_candida_restingae_160519\n",
      "candida_succiphila\n",
      "metschnikowia_kipukae\n",
      "pachysolen_tannophilus\n",
      "yHMPu5000034674_blastobotrys_muscicola_160519\n",
      "yHMPu5000035674_kregervanrija_delftensis_160613\n",
      "yHMPu5000034635_nadsonia_fulvescens_160519\n",
      "saccharomyces_arboricola\n",
      "yHMPu5000034661_dipodascus_albidus_160519\n",
      "spathaspora_arborariae\n",
      "lachancea_fantastica\n",
      "yHMPu5000034901_ogataea_methylivora_160519\n",
      "debaryomyces_hansenii\n",
      "ambrosiozyma_kashinagacola\n",
      "yHMPu5000034624_pichia_nakasei_160519\n",
      "candida_intermedia\n",
      "metschnikowia_matae_maris\n",
      "yHMPu5000035688_cyberlindnera_xylosilytica_160519\n",
      "yHMPu5000035268_wickerhamomyces_hampshirensis_160928\n",
      "yHMPu5000034665_kodamaea_laetipori_160519\n",
      "yHMPu5000034978_cyberlindnera_mrakii_160519\n",
      "nakazawaea_peltata\n",
      "clavispora_lusitaniae\n",
      "yHMPu5000034625_pichia_kudriavzevii_160519\n",
      "candida_infanticola\n",
      "yHMPu5000035286_candida_azyma_160928\n",
      "lachancea_kluyveri\n",
      "vanderwaltozyma_polyspora\n",
      "spathaspora_gorwiae\n",
      "yHMPu5000041863_candida_gorgasii_160928\n",
      "yHMPu5000034884_peterozyma_toletana_160519\n",
      "yHMPu5000034947_ambrosiozyma_philentoma_160519\n",
      "yHMPu5000026256_zygotorulaspora_mrakii_161215\n",
      "yHMPu5000041833_candida_tammaniensis_160928\n",
      "yHMPu5000041855_candida_ascalaphidarum_160928\n",
      "yHMPu5000035691_candida_salmanticensis_160519\n",
      "yHMPu5000035639_wickerhamomyces_canadensis_160613\n",
      "yHMPu5000035297_priceomyces_castillae_160928\n",
      "metschnikowia_matae\n",
      "yHMPu5000034866_zygosaccharomyces_bisporus_160519\n",
      "yHMPu5000034670_blastobotrys_proliferans_160519\n",
      "metschnikowia_hibisci\n",
      "yHMPu5000035252_yamadazyma_nakazawae_160928\n",
      "yHMPu5000034646_wickerhamiella_cacticola_160519\n",
      "yHMPu5000034974_deakozyma_indianensis_160519\n",
      "cyberlindnera_fabianii_JCM3601\n",
      "metschnikowia_proteae\n",
      "candida_apicola\n",
      "metschnikowia_hamakuensis\n",
      "yHMPu5000035689_candida_mycetangii_160613\n",
      "komagataella_pastoris\n",
      "yHMPu5000035274_wickerhamomyces_alni_160928\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yHAB164_kazachstania_aerobia_160519\n",
      "yHMPu5000041829_debaryomyces_fabryi_160928\n",
      "yHMPu5000035658_starmera_amethionina_160613\n",
      "yHAB158_kazachstania_viticola_160519\n",
      "hyphopichia_burtonii\n",
      "yHMPu5000034594_starmera_quercuum_160519\n",
      "saccharomyces_paradoxus\n",
      "yHMPu5000034681_blastobotrys_americana_160519\n",
      "yHMPu5000035662_meyerozyma_caribbica_160613\n",
      "yHMPu5000034939_komagataella_pseudopastoris_160519\n",
      "yHMPu5000034897_ogataea_glucozyma_160519\n",
      "metschnikowia_cerradonensis\n",
      "yHAB134_kazachstania_turicensis_160519\n",
      "hanseniaspora_vinae\n",
      "yHMPu5000041822_dipodascus_geniculatus_160928\n",
      "yHMPu5000034933_kuraishia_molischiana_160519\n",
      "yHMPu5000035675_kregervanrija_fluxuum_160613\n",
      "lachancea_thermotolerans\n",
      "yHMPu5000035048_barnettozyma_salicaria_160519\n",
      "hanseniaspora_valbyensis\n",
      "kluyveromyces_marxianus\n",
      "yHMPu5000035325_cyberlindnera_petersonii_160928\n",
      "ogataea_polymorpha\n",
      "yHMPu5000035640_yamadazyma_philogaea_160519\n",
      "yHMPu5000034986_candida_oregonensis_160519\n",
      "ascoidea_asiatica\n",
      "yarrowia_deformans\n",
      "yHMPu5000035296_priceomyces_carsonii_160928\n",
      "yHMPu5000026274_komagataella_populi_160519\n",
      "yHMPu5000034710_kluyveromyces_dobzhanskii_160519\n",
      "yHMPu5000035022_candida_emberorum_160519\n",
      "saccharomyces_eubayanus\n",
      "lachancea_dasiensis\n",
      "candida_versatilis\n",
      "yHMPu5000035033_candida_cretensis_160519\n",
      "yHMPu5000034606_priceomyces_medius_160519\n",
      "yHMPu5000035279_tortispora_starmeri_160928\n",
      "yarrowia_lipolytica\n",
      "tortispora_caseinolytica\n",
      "yHMPu5000035244_candida_incommunis_160928\n",
      "yHMPu5000034865_zygosaccharomyces_kombuchaensis_160519\n",
      "yHMPu5000034643_candida_schatavii_160519\n",
      "yHMPu5000035673_candida_orba_160613\n",
      "yHMPu5000035684_kloeckera_hatyaiensis_160519\n",
      "yHMPu5000035046_barnettozyma_populi_160613\n",
      "metschnikowia_santaceciliae\n",
      "yHMPu5000034631_martiniozyma_abiesophila_161215\n",
      "wickerhamomyces_anomalus\n",
      "metschnikowia_mauinuiana\n",
      "yHMPu5000041743_candida_hawaiiana_160928\n",
      "yHMPu5000035670_phaffomyces_opuntiae_160613\n",
      "candida_dubliniensis\n",
      "yHAB153_kazachstania_rosinii_160519\n",
      "yHMPu5000035629_yueomyces_sinensis_160519\n",
      "yHMPu5000034904_ogataea_nonfermentans_160519\n",
      "yHMPu5000035271_wickerhamomyces_bovis_160928\n",
      "sugiyamaella_lignohabitans\n",
      "yHMPu5000034887_ogataea_trehaloabstinens_160519\n",
      "yHMPu5000034932_kuraishia_ogatae_160519\n",
      "babjeviella_inositovora\n",
      "wickerhamiella_domercqiae\n",
      "yHMPu5000035018_candida_canberraensis_160519\n",
      "yHMPu5000034967_candida_freyschussii_160519\n",
      "zygosaccharomyces_bailii\n",
      "candida_homilentoma\n",
      "yHMPu5000035335_candida_blattae_160928\n",
      "yHMPu5000035045_barnettozyma_hawaiiensis_160613\n",
      "yHMPu5000035690_candida_vartiovaarae_160613\n",
      "STE12 protein sequence extraction complete\n"
     ]
    }
   ],
   "source": [
    "protein_fname = 'STE12'\n",
    "og_genes = og_genes_lookup[goi_og_lookup[target_y1000_id]]\n",
    "y1000plus_tools.extract_protein_seqs(og_genes, protein_fname, y1000_species_subset)\n",
    "print(protein_fname + ' protein sequence extraction complete')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Alloascoideaceae',\n",
       " 'CUG-Ala',\n",
       " 'CUG-Ser1',\n",
       " 'CUG-Ser2',\n",
       " 'Dipodascaceae/Trichomonascaceae',\n",
       " 'Phaffomycetaceae',\n",
       " 'Pichiaceae',\n",
       " 'Saccharomycetaceae',\n",
       " 'Saccharomycodaceae',\n",
       " 'Sporopachydermia clade',\n",
       " 'Trigonopsidaceae'}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check to see which major clades / species are present in the orthogroup to identify outgroup\n",
    "#there is no good outgroup\n",
    "specs_in_og = []\n",
    "for gene in og_genes: \n",
    "    specs_in_og.append(gene.split('_')[0])\n",
    "\n",
    "set(y1000_species_subset[y1000_species_subset['spec_og_id'].isin(set(specs_in_og))]['Major clade'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>original_genome_id</th>\n",
       "      <th>old_species_names</th>\n",
       "      <th>species_names_fig2</th>\n",
       "      <th>hex</th>\n",
       "      <th>id_table S1</th>\n",
       "      <th>Species name</th>\n",
       "      <th>Major clade</th>\n",
       "      <th>clade_order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>CUG usage</th>\n",
       "      <th>spec_og_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>yHMPu5000034655_botryozyma_nematodophila_160519</td>\n",
       "      <td>yHMPu5000034655_Botryozyma_nematodophila</td>\n",
       "      <td>Botryozyma_nematodophila</td>\n",
       "      <td>#FF0090</td>\n",
       "      <td>15</td>\n",
       "      <td>Botryozyma nematodophila</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>2</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Botryozyma</td>\n",
       "      <td>Leu</td>\n",
       "      <td>195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>tortispora_caseinolytica</td>\n",
       "      <td>Tortispora_caseinolytica</td>\n",
       "      <td>Tortispora_caseinolytica</td>\n",
       "      <td>#FF0090</td>\n",
       "      <td>10</td>\n",
       "      <td>Tortispora caseinolytica</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>2</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Tortispora</td>\n",
       "      <td>Leu</td>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>yHMPu5000035654_tortispora_ganteri_160519</td>\n",
       "      <td>yHMPu5000035654_Tortispora_ganteri</td>\n",
       "      <td>Tortispora_ganteri</td>\n",
       "      <td>#FF0090</td>\n",
       "      <td>11</td>\n",
       "      <td>Tortispora ganteri</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>2</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Tortispora</td>\n",
       "      <td>Leu</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>yHMPu5000035279_tortispora_starmeri_160928</td>\n",
       "      <td>yHMPu5000035279_Tortispora_starmeri</td>\n",
       "      <td>Tortispora_starmeri</td>\n",
       "      <td>#FF0090</td>\n",
       "      <td>14</td>\n",
       "      <td>Tortispora starmeri</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>2</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Tortispora</td>\n",
       "      <td>Leu</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>yHMPu5000035282_trigonopsis_vinaria_160928</td>\n",
       "      <td>yHMPu5000035282_Trigonopsis_vinaria</td>\n",
       "      <td>Trigonopsis_vinaria</td>\n",
       "      <td>#FF0090</td>\n",
       "      <td>13</td>\n",
       "      <td>Trigonopsis vinaria</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>2</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsis</td>\n",
       "      <td>Leu</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  original_genome_id  \\\n",
       "ID                                                     \n",
       "205  yHMPu5000034655_botryozyma_nematodophila_160519   \n",
       "206                         tortispora_caseinolytica   \n",
       "207        yHMPu5000035654_tortispora_ganteri_160519   \n",
       "208       yHMPu5000035279_tortispora_starmeri_160928   \n",
       "210       yHMPu5000035282_trigonopsis_vinaria_160928   \n",
       "\n",
       "                            old_species_names        species_names_fig2  \\\n",
       "ID                                                                        \n",
       "205  yHMPu5000034655_Botryozyma_nematodophila  Botryozyma_nematodophila   \n",
       "206                  Tortispora_caseinolytica  Tortispora_caseinolytica   \n",
       "207        yHMPu5000035654_Tortispora_ganteri        Tortispora_ganteri   \n",
       "208       yHMPu5000035279_Tortispora_starmeri       Tortispora_starmeri   \n",
       "210       yHMPu5000035282_Trigonopsis_vinaria       Trigonopsis_vinaria   \n",
       "\n",
       "         hex  id_table S1              Species name       Major clade  \\\n",
       "ID                                                                      \n",
       "205  #FF0090           15  Botryozyma nematodophila  Trigonopsidaceae   \n",
       "206  #FF0090           10  Tortispora caseinolytica  Trigonopsidaceae   \n",
       "207  #FF0090           11        Tortispora ganteri  Trigonopsidaceae   \n",
       "208  #FF0090           14       Tortispora starmeri  Trigonopsidaceae   \n",
       "210  #FF0090           13       Trigonopsis vinaria  Trigonopsidaceae   \n",
       "\n",
       "     clade_order            Family        Genus CUG usage  spec_og_id  \n",
       "ID                                                                     \n",
       "205            2  Trigonopsidaceae   Botryozyma       Leu         195  \n",
       "206            2  Trigonopsidaceae   Tortispora       Leu         134  \n",
       "207            2  Trigonopsidaceae   Tortispora       Leu         301  \n",
       "208            2  Trigonopsidaceae   Tortispora       Leu         281  \n",
       "210            2  Trigonopsidaceae  Trigonopsis       Leu         282  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Farthest group out is the Trignopsidaceae\n",
    "\n",
    "#Check which species are present in that clade\n",
    "\n",
    "specs_in_og_table = y1000_species_subset[y1000_species_subset['spec_og_id'].isin(set(specs_in_og))]\n",
    "specs_in_og_table[specs_in_og_table['Major clade']=='Trigonopsidaceae']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outgroup gene = 134_3314\n"
     ]
    }
   ],
   "source": [
    "#Choose a species to make the outgroup.  \n",
    "\n",
    "#chose tortispora_caseinolytica because it looks like it might have been previously sequenced\n",
    "\n",
    "#choose outgroup species (important for making protein trees from multiple sequence alignments) \n",
    "outgroup = 'Tortispora caseinolytica' #'Hanseniaspora vineae'#'Yarrowia lipolytica' #'Lipomyces starkeyi' # 'Tortispora starmeri' #'Lipomyces starkeyi'   #Lipomyces starkeyi is a good outgroup for all the species\n",
    "\n",
    "outgroup_orig_genome = y1000_species[y1000_species['Species name']==outgroup]['original_genome_id'].values[0]\n",
    "\n",
    "#Test to see if orthogroup has something from outgroup: \n",
    "\n",
    "y1000_id_outgroup = y1000_species_subset[y1000_species_subset['Species name']==outgroup]['spec_og_id'].values[0]\n",
    "\n",
    "outgroup_gene = ''\n",
    "for gene in og_genes:#og_genes: \n",
    "    y1000_spec_id = gene.split('_')[0]\n",
    "    if y1000_spec_id == str(y1000_id_outgroup):\n",
    "        outgroup_gene = gene\n",
    "        \n",
    "\n",
    "if outgroup_gene =='': \n",
    "    print(outgroup + ' has no proteins in og ' + ogs + ' for ' + goi_pair + '. Pick another outgroup')\n",
    "else: \n",
    "    print('outgroup gene = ' + outgroup_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Manually obtained this by looking for the species in the alignment.  \n",
    "#can also find the name automaticalle with code from 20190219_STRE_examples\n",
    "outgroup_dict = {'134_3314':'tortispora_caseinolytica_Seq_3315'}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mafft --genafpair --maxiterate 1000 /mnt/c/Users/BMH_work/github/y1000plus_tools/genomes/y1000plus/proteins_og/STE12.fasta > /mnt/c/Users/BMH_work/github/y1000plus_tools/genomes/y1000plus/proteins_og/STE12_aln.fasta\n",
      "/home/heineike/bioinformatics/trimAl/source/trimal -in /mnt/c/Users/BMH_work/github/y1000plus_tools/genomes/y1000plus/proteins_og/STE12_aln.fasta -out /mnt/c/Users/BMH_work/github/y1000plus_tools/genomes/y1000plus/proteins_og/STE12_aln_trimmed.fasta -gappyout\n",
      "/home/heineike/bioinformatics/iqtree/iqtree-1.6.12-Linux/bin/iqtree -s /mnt/c/Users/BMH_work/github/y1000plus_tools/genomes/y1000plus/proteins_og/STE12_aln_trimmed.fasta -nt AUTO -o tortispora_caseinolytica_Seq_3315\n"
     ]
    }
   ],
   "source": [
    "protein_fname_base = '/mnt/c' + \"/\".join(y1000plus_dir.split(':')[1].split('\\\\')) + 'proteins_og/STE12'\n",
    "\n",
    "#Command: \n",
    "\n",
    "#This is the E-ENS-i algorithm which is suitable for conserved regions separated by long unaligned gaps. https://mafft.cbrc.jp/alignment/software/algorithms/algorithms.html\n",
    "\n",
    "#Mafft 7.310-1\n",
    "mafft_command = ['mafft', '--genafpair', '--maxiterate', '1000', \n",
    "                protein_fname_base + '.fasta', \n",
    "                '>',\n",
    "                protein_fname_base + '_aln.fasta']\n",
    "\n",
    "\n",
    "print(\" \".join(mafft_command))\n",
    "\n",
    "\n",
    "#trimal 1.2r59\n",
    "trimal_command = ['/home/heineike/bioinformatics/trimAl/source/trimal',\n",
    "                  '-in', protein_fname_base + '_aln.fasta', \n",
    "                  '-out', protein_fname_base + '_aln_trimmed.fasta', \n",
    "                  '-gappyout']\n",
    "\n",
    "print(\" \".join(trimal_command))\n",
    "\n",
    "\n",
    "\n",
    "#iqtree-1.6.12 linux 64 bit\n",
    "iqtree_command = [\"/home/heineike/bioinformatics/iqtree/iqtree-1.6.12-Linux/bin/iqtree\", \n",
    "                  \"-s\" , protein_fname_base + '_aln_trimmed.fasta',\n",
    "                  \"-nt\", \"AUTO\",  #automatically determines number of threads \n",
    "                  \"-o\", outgroup_dict[outgroup_gene]]\n",
    "print(\" \".join(iqtree_command))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
